---
title: "SCRIPT Flow DAA"
output:
  html_document:
    df_print: paged
---

# Setup   
## Load packages   
```{r setup}
library(ggplot2)
library(ggplotify)
library(Rmisc)
library(ggalluvial)
library(xtable)
library(later)
library(Cairo)
library(ggsci)
library(ggsignif)
library(googlesheets4)
library(googledrive)
library(gtools)
library(tidyverse)
library(summarytools)
library(qwraps2)
library(car)
library(Hmisc)
library(yardstick)
library(readxl)
library(ggfortify)
library(lubridate)
library(reshape2)
library(pheatmap)
library(RColorBrewer)
library(ggwordcloud)
library(tm)
library(factoextra)
library(uwot)
library(igraph)
library(patchwork)
library(ragg)
source("~/Documents/GitHub/COVID19_BAL_flow/rgrant/200919_correct_endpoints.R")
source("~/Documents/GitHub/COVID19_BAL_flow/rgrant/200919_code_pathogens.R")

mean_sd = function(x){
  return(mean_sdl(x, mult = 1))
}
#actually the pal for all plots
pal = pal_npg("nrc")(9)

set.seed(12345)
```

## Google login   
```{r}
drive_auth(use_oob = T, cache = T, email = "xxxxxx") # have to run in console :(
gs4_auth(token = drive_token(), use_oob = T, cache = T, email = "xxxxxx")
```


## Import Sasha's data analysis results   
```{r message=FALSE, warning=FALSE}
flow_data = read_excel("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/20200707_COVID_SCRIPT_5L_Aria.xlsx",
                        trim_ws = T,
                        .name_repair = "universal",
                        sheet = "cleaned",
                  col_types = c("text", "text", rep("numeric", 12))) %>% 
  as.data.frame() %>% 
  dplyr::mutate(panel = "new")
prior_data = read_excel("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/20200701_Pre-COVID_SCRIPT_6L_Aria.xlsx",
                        trim_ws = T,
                        .name_repair = "universal",
                        sheet = "cleaned",
                  col_types = c("text", "text", rep("numeric", 12))) %>% 
  as.data.frame() %>% 
  dplyr::mutate(panel = "old")
flow_data = smartbind(flow_data, prior_data)

#remove in-progress entries
flow_data = subset(flow_data, !is.na(Sample))

#mark neutrophilic patients
flow_data$neutrophilic = factor(ifelse(flow_data$percent_neutrophils >= 50,
                                  yes = "Neutrophilic",
                                  no = "Non-Neutrophilic"))

#adjust types
flow_data$ID = as.character(flow_data$ID)
flow_data$panel = factor(flow_data$panel)
```

## Import RNA extraction metadata   
```{r}
rna_metadata = read_sheet("xxxxxx",
                          trim_ws = T,
                          .name_repair = "universal", 
                          sheet = "SCRIPT_01 RNAseq Metadata",
                          na = c("", "NA"))
#clean up colnames
colnames(rna_metadata) = gsub("\\.", "_", colnames(rna_metadata)) #I like underscores
colnames(rna_metadata) = gsub("_+$", "", colnames(rna_metadata)) # remove trailing
colnames(rna_metadata) = gsub("_+", "_", colnames(rna_metadata)) #remove dup underscores
#tube is randomly read in as a list. fix this.
rna_metadata$tubeid_macrophRNA = as.character(rna_metadata$tubeid_macrophRNA)
rna_metadata$tubeid_macrophRNA[rna_metadata$tubeid_macrophRNA == "NULL"] = NA
#take out problem cols (added back)
rna_metadata = rna_metadata %>% 
  dplyr::select(-c(RNAseq_batch, Batch_sample))
#remove preceding zeros
rna_metadata = rna_metadata %>% 
  mutate_at("tubeid_macrophRNA", function(x){ gsub("^0", "", x) })
#fix dates
rna_metadata$sample_process_dt = as.Date(rna_metadata$sample_process_dt)

#merge with flow data
flow_data$BAL_sample = substring(flow_data$Sample, 10, 20)
flow_data = left_join(flow_data, 
                 rna_metadata, 
                 by = c("BAL_sample" = "tc_pt_study_id"),
                 na_matches = "never")
```


## Import clinical metadata   
```{r}
edw_endpoints = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 SCRIPT Basic Endpoints.csv",
                         strip.white = T,
                         check.names = T,
                         na.strings = c("", "NA"))
date_cols = colnames(edw_endpoints)[grepl("date|_dt|_intub_", colnames(edw_endpoints), ignore.case = T)]
edw_endpoints = edw_endpoints %>%
  mutate_at(.vars = date_cols,
            .funs = function(x){
              as.Date(x, format = "%m/%d/%Y %H:%M", tz = "CST") }) %>% 
  mutate(pt_study_id = as.character(pt_study_id)) %>% 
  dplyr::mutate(binned_outcome = case_when(discharge_disposition_name == "Expired" ~ "Deceased",
                                             grepl("^home|Elopement|Shelter|Custodial", 
                                                   discharge_disposition_name, 
                                                   ignore.case = T) ~ "Home",
                                             discharge_disposition_name %in% 
                                             c("Acute Care Hospital",
                                               "Inpatient Hospice",
                                               "Skilled Nursing Facility or Subacute Rehab Care",
                                               "Acute Inpatient Rehabilitation",
                                               "Planned Readmission â€“ DC/transferred to acute inpatient rehab") ~ 
                                               "Inpatient Facility",
                                             discharge_disposition_name == "Long-Term Acute Care Hospital (LTAC)" ~ "LTAC",
                                             TRUE ~ "Other")) %>% 
  select(-c(full_name, consent_dt)) %>% 
  unique()

#add EDW molecular data and give script IDs
edw_molecular = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 SCRIPT and COVID BAL Results.csv",
                         na.strings = c("", "NA"),
                         strip.white = T,
                         skip = 2,
                         check.names = T) %>% 
  select(-Name)
  
mrn_conv = edw_molecular %>% 
  dplyr::select(ir_id, MRN)
#make numeric values numeric   
numeric_cols = c("day_of_intubation", "day_of_hospitalization", "RBC_BODY_FLUID", "WBC__BODY_FLUID", "NEUTROPHILS__BODY_FLUID",
                 "Absolute_Neutrophils", "TOXIC_GRANULATION", "MACROPHAGE_BF", "MONOCYTE_BF", "LYMPH_BF", 
                 "ABSOLUTE_LYMPHOCYTES", "EOSINOPHILS__BODY_FLUID", "ABSOLUTE_EOSINOPHILS", "PLASMA_CELL_BF",
                 "OTHER_CELLS__BODY_FLUID", "AMYLASE_BF", "WHITE_BLOOD_CELL_COUNT", 
                 "C_Reactive_Protein", "LDH", "CREATINE_KINASE__TOTAL", "PROCALCITONIN", "FERRITIN", "TROPONIN_I",
                 "Creatinine", "AST__SGOT_", "D_DIMER", "max_daily_temp")
edw_molecular = edw_molecular %>% mutate_at(.vars = numeric_cols, .funs = function(x){
  x = gsub(">", "", x)
  x = gsub("<.+", "0", x)
  x = gsub(",", "", x)
  x = as.numeric(x)
  return(x)})

#remove one strange duplicate entry
edw_molecular = subset(edw_molecular, !(duplicated(edw_molecular$order_accession_num)))

#clean up pathogen data reliably
edw_molecular = edw_molecular %>% 
  rename(ASPERGILLUS_GALACTOMANNAN_ANTIGEN_NMH_LFH = ASPERGILLUS_GALACTOMANNAN_ANTIGEN_NMH_LFH_) #differences with MS2
edw_molecular = code_pathogens(edw_molecular)
edw_molecular$order_accession_num = as.character(edw_molecular$order_accession_num)

#fix dates
date_cols = colnames(edw_molecular)[grepl("date", colnames(edw_molecular), ignore.case = T)]
edw_molecular = edw_molecular %>% 
  mutate_at(.vars = date_cols, .funs = function(x){
    as.Date(x, format = "%m/%d/%Y", tz = "CST")} )

#reformat order timestamps for merging with blood gas
edw_molecular$BAL_order_timestamp = as.POSIXct(edw_molecular$BAL_order_timestamp,
                                               format = "%m/%d/%Y %X %p", 
                                               tz = "America/Chicago")

#import known COVID patients and all patient IDs from Lorenzo (NEJM MS)
covid_cases = readRDS("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/190914 pat.list.tbl1") %>% 
  dplyr::select(-c(admission.datetime, discharge.datetime)) %>% 
  dplyr::mutate(covid_confirmed = TRUE)  #all of these patients are positive

#import all patient data for script
all_patients = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 STU00204868_subjects_09_18_2020.csv",
                        na.strings = c("", "NA"),
                        strip.white = T,
                        check.names = T, 
                        colClasses = rep("character", 10)) %>% 
  separate_rows(case.number, sep = ", ")  %>% #uncollapse ID column
  dplyr::select(-diagnosis)

#import conversion from SCRIPT BAL ID to NMH accession number and add to molecular
accession_conv = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/201014_U19SCRIPTVersion20-AccessionNumberConve_DATA_2020-10-14_2138.csv",
                          na.strings = c("", "NA", "n/a", "N/A", "na"), 
                          strip.white = T) %>% 
  na.omit() %>% 
  dplyr::mutate(accession_num = gsub("-", "", accession_num)) %>%  #match destination formatting
  dplyr::mutate(study_id = substring(bal_barcode, 1, 4),
                bal_dt = as.Date(bal_dt)) %>% 
  dplyr::filter(grepl("\\d{4}-BAL-\\d{2}", bal_barcode)) %>% #remove bad entries 
  dplyr::select(bal_barcode, accession_num, study_id, bal_dt) %>% 
  unique()
colnames(accession_conv) = c("tc_pt_study_id", "order_accession_num", "study_id", "bal_date")
accession_conv$order_accession_num = trimws(accession_conv$order_accession_num) #not clear why this is necessary, but it is

#pull in manual edits for entry errors
bal_edits = read_excel("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/201014_missing_accession_numbers_final.xlsx") %>% 
  mutate(bal_date = as.Date(bal_date, origin = "1899-12-30"))
discards = bal_edits$tc_pt_study_id[bal_edits$action == "discard"]
accession_conv = subset(accession_conv, !(tc_pt_study_id %in% discards)) #these will then get removed from EDW data without a bal number
#recode incorrect entries
recodes = bal_edits$tc_pt_study_id[bal_edits$action == "update"]
for(i in 1:nrow(accession_conv))
{
  cur_bal = accession_conv$tc_pt_study_id[i]
  if(cur_bal %in% recodes)
  {
    accession_conv$order_accession_num[i] = bal_edits$correct_number[bal_edits$tc_pt_study_id == cur_bal]
    accession_conv$bal_date[i] = bal_edits$bal_date[bal_edits$tc_pt_study_id == cur_bal]
  }
}
#add missing entries
keepers = bal_edits %>% 
  dplyr::filter(action == "keep") %>% 
  dplyr::select(tc_pt_study_id, order_accession_num = correct_number, 
                study_id, bal_date) %>% 
  mutate(order_accession_num = as.character(order_accession_num),
         study_id = as.character(study_id))
accession_conv = bind_rows(accession_conv, keepers) %>% 
  dplyr::filter(order_accession_num != "see comments")

edw_molecular = left_join(edw_molecular, accession_conv, by = c("order_accession_num"),
                          na_matches = "never")

# some BAL samples do not have EDW entries. Add in from rna metadata, endpoints, and flow
edw_molecular$script_day = as.numeric(substring(edw_molecular$tc_pt_study_id, 
                                     nchar(edw_molecular$tc_pt_study_id) - 1)) #useful below

flow_and_bulk = unique(c(rna_metadata$tc_pt_study_id,
                  flow_data$BAL_sample))
missing_samples = base::setdiff(flow_and_bulk,
                                edw_molecular$tc_pt_study_id)
missing_samples = missing_samples[grepl("\\d{4}-BAL-\\d{2}", missing_samples)]
missing_sample_df = data.frame(study_id = substring(missing_samples, 1, 4),
                               tc_pt_study_id = missing_samples)
missing_sample_df = edw_molecular %>% 
  dplyr::mutate(first_bal_date = as.Date(BAL_order_date) - script_day) %>% 
  dplyr::select(ir_id:hosp_disch_date, study_id, first_bal_date) %>% 
  dplyr::filter(!is.na(ir_id)) %>% #empties
  unique() %>% 
  left_join(missing_sample_df, .,
            by = "study_id",
            na_matches = "never") %>% 
  group_by(study_id) %>% 
  dplyr::mutate(first_bal_date = min(first_bal_date)) %>% #at most +/- 1 day
  ungroup() %>% 
  unique() %>%
  dplyr::mutate(BAL_order_date = first_bal_date + as.numeric(substring(tc_pt_study_id,
                                                                nchar(tc_pt_study_id) - 1)),
         day_of_intubation = as.numeric(difftime(BAL_order_date, first_intubation_date, units = "days"))) %>% 
  select(-first_bal_date)
edw_molecular = bind_rows(edw_molecular, missing_sample_df)

#import pneumonia diagnosis   
diagnosis = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200929 U19SCRIPTVersion20-PneumoniaDiagnosis_DATA_2020-09-29_1335.csv", 
                     na.strings = c("", "NA"),
                     strip.white = T,
                     check.names = T) %>% 
  dplyr::filter(!is.na(pt_category)) %>% 
  dplyr::select(record_id, pt_category:clin_hap) %>% 
  dplyr::mutate(pneu_assess_dt = as.Date(pneu_assess_dt)) %>% 
  unique() %>% 
  arrange(record_id, pneu_assess_dt)

record_id_conv = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 U19SCRIPTVersion20-RecordIDConversion_DATA_2020-09-18_0022.csv", 
                          na.strings = c("", "NA"),
                          strip.white = T,
                          check.names = T) %>% 
  dplyr::filter(!is.na(pt_study_id)) %>% 
  dplyr::select(record_id, pt_study_id) %>% 
  unique()
diagnosis_conv = data.frame(pt_category = c(1:4),
                            diagnosis = factor(c("CAP", "HAP", "VAP", "Non-Pneumonia Control")),
                            simplified_diagnosis = factor(c(rep("Pneumonia", 3), "Non-Pneumonia Control")))
diagnosis = full_join(diagnosis, record_id_conv, by = "record_id", na_matches = "never") %>% 
  left_join(., diagnosis_conv, by = "pt_category", na_matches = "never") %>% 
  dplyr::select(-record_id) %>% 
  group_by(pt_study_id) %>% 
  dplyr::mutate(study_dates = list(as.character(pneu_assess_dt)), 
         diagnoses = list(as.character(diagnosis)),
         simplified_diagnoses = list(as.character(simplified_diagnosis)),
         pt_categories = list(pt_category),
         ever_pna = any(simplified_diagnosis == "Pneumonia", na.rm = T)) %>% 
  ungroup() %>% 
  unique()
#all patients with multiple diagnoses go from control --> some type of pneumonia (makes sense)
#so best practice is probably to code these cases as control for all samples until PNA, then always PNA
#for patients with multiple PNA types, just always code as PNA (current scheme is to bin all PNA for now)
#for patients with 1 diagnosis, just pick all BAL
pts = c()
assoc_bal = c()
simp = c()
pneu_assess_dt = c()
for(patient in unique(diagnosis$pt_study_id))
{
  diagnoses = subset(diagnosis, pt_study_id == patient)$diagnosis
  
  #for consistent pna patients, link all samples
  if(!("Non-Pneumonia Control" %in% diagnoses))
  {
    bal_samples = sort(subset(edw_molecular, study_id == patient)$tc_pt_study_id)
    pts = append(pts, rep(patient, length(bal_samples)))
    simp = append(simp, rep("Pneumonia", length(bal_samples)))
    assoc_bal = append(assoc_bal, bal_samples)
  } else #i.e. if at least one control diagnosis
  {
    if(length(diagnoses) == 1)
    {
      bal_samples = sort(subset(edw_molecular, study_id == patient)$tc_pt_study_id)
      pts = append(pts, rep(patient, length(bal_samples)))
      simp = append(simp, rep("Non-Pneumonia Control", length(bal_samples)))
      assoc_bal = append(assoc_bal, bal_samples)
    } else #split by diagnosis date. May be hazy in the middle, but still better
    {
      first_pna = min(subset(diagnosis, pt_study_id == patient & diagnosis != "Non-Pneumonia Control")$pneu_assess_dt)
      control_bal_samples = sort(subset(edw_molecular, study_id == patient & BAL_order_date < first_pna)$tc_pt_study_id)
      pna_bal_samples = sort(subset(edw_molecular, study_id == patient & BAL_order_date >= first_pna)$tc_pt_study_id)
      pts = append(pts, rep(patient, (length(control_bal_samples) + length(pna_bal_samples))))
      simp = append(simp, c(rep("Non-Pneumonia Control", length(control_bal_samples)),
                            rep("Pneumonia", length(pna_bal_samples))))
      assoc_bal = append(assoc_bal, c(control_bal_samples, pna_bal_samples))
    }
  }
}
bal_diagnoses = edw_molecular %>% 
  dplyr::select(tc_pt_study_id, BAL_order_date) %>% 
  right_join(., data.frame(pt_study_id = pts,
                           simplified_diagnosis = simp,
                           tc_pt_study_id = assoc_bal))
#match BALs with PNA assessments
pneu_assess_dt = vector(mode = "character", length = nrow(bal_diagnoses))
for(i in 1:nrow(bal_diagnoses))
{
  cur_pt = bal_diagnoses$pt_study_id[i]
  cur_bal = bal_diagnoses$tc_pt_study_id[i]
  cur_bal_date = bal_diagnoses$BAL_order_date[i]
  sub = diagnosis %>% 
    dplyr::filter(pt_study_id == cur_pt) %>% 
    mutate(timediff = abs(difftime(pneu_assess_dt, cur_bal_date, units = "days")))
  if(nrow(sub) == 1)
  {
    pneu_assess_dt[i] = as.character(sub$pneu_assess_dt[1])
  } else if(all(is.na(sub$pneu_assess_dt)) | is.na(cur_bal_date))
  {
    pneu_assess_dt[i] = NA
  } else
  {
    pneu_assess_dt[i] = as.character(unique(sub$pneu_assess_dt[which.min(sub$timediff)]))
  }
}
bal_diagnoses$pneu_assess_dt = as.Date(pneu_assess_dt)

#for rendering unique patients for stats later
diagnosis_for_pt_data = diagnosis %>% 
  dplyr::select(-c(diagnosis, pt_category, pneu_assess_dt, simplified_diagnosis)) %>% 
  unique() %>% 
  dplyr::filter(!duplicated(pt_study_id)) #unique doesn't work on list cols

#pull together patient metadata
ir_conversion = read_excel("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/201015_SCRIPT_ID_IRID.xlsx") %>% 
  dplyr::rename(study_id = pt_study_id, ir_id = patient_ir_id) %>% 
  dplyr::mutate(study_id = as.numeric(study_id))
covid_cases$clarity.west.mrn = as.character(covid_cases$clarity.west.mrn)
patient_data = full_join(covid_cases,
                         all_patients,
                         by = c("clarity.west.mrn" = "nmhc_record_number"),
                         na_matches = "never") %>% 
  dplyr::mutate(case.number = as.integer(case.number)) %>% 
  select(-c(first.name.x, first.name.y, last.name.x, last.name.y, address.line.1:email, nmff_record_number, nmh_record_number, ir.id))
colnames(patient_data)[colnames(patient_data) == "case.number"] = "study_id"
patient_data = left_join(patient_data, ir_conversion)

#import later adjudications
chichi_covid = read_excel("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200927_adjudication_data_CP_edit.xlsx", na = c("", "NA")) %>% 
  mutate(covid_confirmed = case_when(`Covid Status` == "covid" ~ TRUE,
                                     is.na(`Covid Status`) ~ FALSE))
for(i in 1:nrow(patient_data))
{
  if(!(patient_data$clarity.west.mrn[i] %in% covid_cases$clarity.west.mrn) &
     patient_data$study_id[i] %in% chichi_covid$study_id)
  {
    patient_data$covid_confirmed[i] = chichi_covid$covid_confirmed[chichi_covid$study_id == patient_data$study_id[i]]
  }
}

diagnosis = diagnosis %>% 
  dplyr::select(-c(diagnoses, study_dates, simplified_diagnoses, pt_categories, ever_pna)) %>% 
  dplyr::select(-c(diagnosis, pt_category)) %>%  #keep only simple (other data preserved later)
  left_join(., bal_diagnoses) %>% 
  unique()


#bind with molecular
diagnosis = diagnosis %>% 
  mutate(pt_study_id = as.character(pt_study_id)) %>% 
  dplyr::select(-c(BAL_order_date, pneu_assess_dt))
edw_molecular = left_join(edw_molecular, 
                          diagnosis,
                          by = c("study_id" = "pt_study_id",
                                 "tc_pt_study_id"),
                          na_matches = "never")

#import smoker status data
smoking_codes = data.frame(pt_smoker = c(0, 1, 2),
                           smoking_status = factor(c("Never Smoker", "Current Smoker", "Past Smoker")))
smoking_data = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 U19SCRIPTVersion20-SmokerStatus_DATA_2020-09-18_0023.csv",
                        na.strings = c("", "NA", "n/a", "N/A", "na"),
                        strip.white = T) %>% 
  dplyr::mutate(pt_study_id = as.character(pt_study_id)) %>% 
  dplyr::filter(!is.na(pt_study_id)) %>% 
  left_join(., smoking_codes, by = "pt_smoker", na_matches = "never")
smoking_data$pack_years = ifelse(smoking_data$smoking_status == "Past Smoker",
                                 yes = smoking_data$former_smoke_pk_yrs,
                                 no = (smoking_data$smoke_pk_day * smoking_data$smoke_years))
edw_endpoints = left_join(edw_endpoints, smoking_data, by = "pt_study_id",
                          na_matches = "never")

#import BMI data   
bmi_data = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 SCRIPT BMI data.csv",
                    na.strings = c("", "NA", "n/a", "N/A", "na"),
                    strip.white = T)

#import SOFA scores
#also includes unique stays!
sofa_data = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 SCRIPT Serial SOFA Score.csv",
                     strip.white = T,
                     check.names = T,
                     na.strings = c("", "NA", "n/a", "N/A", "na"))
date_cols = setdiff(colnames(sofa_data)[grepl("date|day|dt", colnames(sofa_data), ignore.case = T)],
                    c("ICU_Day"))
sofa_data = sofa_data %>% 
  mutate_at(.vars = date_cols, .funs = function(x){ as.Date(x, format = "%m/%d/%Y", tz = "CST") }) %>% 
  dplyr::select(-c(MRN, admission_datetime, discharge_datetime)) #remove duplicates

aps_data = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 SCRIPT Serial APS Score.csv",
                     strip.white = T,
                     check.names = T,
                     na.strings = c("", "NA", "n/a", "N/A", "na"))
date_cols = setdiff(colnames(aps_data)[grepl("date|day|dt", colnames(aps_data), ignore.case = T)],
                    c("icu_day"))
aps_data = aps_data %>% 
  mutate_at(.vars = date_cols, .funs = function(x){ as.Date(x, format = "%m/%d/%Y", tz = "CST") }) %>% 
  dplyr::select(-c(MRN, admission_datetime, discharge_datetime)) #remove duplicates

#actually can allow NA matches here
sofa_data = full_join(sofa_data, 
                      suffix = c("_SOFA", "_APS"),
                      aps_data,
                      by = c("pt_study_id", "icu_rank",
                             "icu_start_dt", "icu_stop_dt", "day_bucket_starts", "day_bucket_ends", 
                             "bilirubin", "creatinine", "visit_id" = "visit_key", "ICU_Day" = "icu_day"), )

#join by date range for each associated BAL
merge_dates = data.frame(pt_study_id = c(),
                         visit_id = c(),
                         ICU_Day = c(),
                         day_bucket_starts = c(),
                         day_bucket_ends = c(),
                         merge_date = c())
for(i in 1:nrow(sofa_data))
{
  pt = sofa_data$pt_study_id[i]
  visit = sofa_data$visit_id[i]
  ICU_Day = sofa_data$ICU_Day[i]
  min_date = sofa_data$day_bucket_starts[i]
  max_date = sofa_data$day_bucket_ends[i]
  cor_BAL = subset(edw_molecular, study_id == pt &
                     BAL_order_date >= min_date &
                     BAL_order_date <= max_date)
  rows_out = data.frame(pt_study_id = rep(pt, nrow(cor_BAL)),
                        visit_id = rep(visit, nrow(cor_BAL)),
                        ICU_Day = rep(ICU_Day, nrow(cor_BAL)),
                        day_bucket_starts = rep(min_date, nrow(cor_BAL)),
                        day_bucket_ends = rep(max_date, nrow(cor_BAL)),
                        merge_date = cor_BAL$BAL_order_date)
  merge_dates = rbind(merge_dates, rows_out)
}
sofa_data_for_edw_molecular = left_join(sofa_data, merge_dates) #for merging into total metadata, any downstream
#join with patient data without duplicating data
#and isolate first score
sofa_data_for_patient_data = sofa_data %>% 
  dplyr::select(-enrollment_dt1) %>% 
  arrange(pt_study_id, visit_id, icu_rank, ICU_Day) %>% 
  group_by(pt_study_id, visit_id) %>%
  dplyr::summarize(earliest_sofa = first(SOFA), 
         mean_sofa = mean(SOFA, na.rm = T),
         median_sofa = median(SOFA, na.rm = T),
         earliest_aps = first(total_APS_score_num), 
         mean_aps = mean(total_APS_score_num, na.rm = T), 
         median_aps = median(total_APS_score_num, na.rm = T))
#merge with patient data
patient_data = left_join(patient_data, 
                         sofa_data_for_patient_data,
                         by = c("study_id" = "pt_study_id"),
                         na_matches = "never")

#import ventilator settings   
vent_settings = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 Ventilator settings.csv",
                     strip.white = T,
                     check.names = T,
                     na.strings = c("", "NA", "n/a", "N/A", "na")) %>% 
  dplyr::select(ir_id, meas_date, disp_name, meas_value) %>% 
  dplyr::mutate(meas_date = as.Date(meas_date, format = "%m/%d/%Y", tz = "CST")) %>% 
  pivot_wider(names_from = disp_name, 
              values_from = meas_value, 
              names_repair = "universal")
numeric_cols = colnames(vent_settings)[c(3:12, 14)]
vent_settings = vent_settings %>%
  mutate_at(.vars = numeric_cols, .funs = as.numeric) %>% #fix non-numeric cols
  dplyr::mutate(driving_pressure = Plateau.Pressure - PEEP)
#make single IBW/kg column
vent_settings$IBW_kg = ifelse(is.na(vent_settings$IBW.kg..Calculated..MALE),
                               yes = vent_settings$IBW.kg..Calculated..FEMALE,
                               no = vent_settings$IBW.kg..Calculated..MALE)

#import blood gases
blood_gas = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200918 NMH COVID pos ventilated or SCRIPT patient blood gases.csv",
                     strip.white = T,
                     check.names = T,
                     na.strings = c("", "NA", "n/a", "N/A", "na")) %>% 
  dplyr::select(ir_id, order_datetime, NAME, value_numeric) %>% 
  dplyr::mutate(order_datetime = as.POSIXct(order_datetime, format = "%m/%d/%Y %X %p", tz = "America/Chicago")) %>% 
  pivot_wider(names_from = NAME, 
              values_from = value_numeric, 
              names_repair = "universal",
              values_fn = mean)
#determine closest realistic BAL order date (if any)   
#setting a max of 24hrs for now
blood_gas$merge_datetime = NA
for(i in 1:nrow(edw_molecular))
{
  cur_bal = edw_molecular[i, ]
  possible_bloodgas = blood_gas %>% 
    dplyr::filter(ir_id == cur_bal$ir_id) %>% 
    dplyr::mutate(timediff = difftime(cur_bal$BAL_order_timestamp, order_datetime, tz = "America/Chicago", units = "hours")) %>% 
    dplyr::filter(timediff < 24 & timediff >= 0)
  if(nrow(possible_bloodgas) == 0)
  {
    blood_gas$merge_datetime[i] = NA
  } else
  {
    closest = possible_bloodgas[which.min(possible_bloodgas$timediff), ]
    blood_gas$merge_datetime[blood_gas$ir_id == closest$ir_id & blood_gas$order_datetime == closest$order_datetime] = as.character(cur_bal$BAL_order_timestamp)
  }
}
blood_gas$merge_datetime = as.POSIXct(blood_gas$merge_datetime, tz = "America/Chicago")

#merge metadata
edw_molecular$ir_id = as.character(edw_molecular$ir_id)
edw_endpoints$ir_id = as.character(edw_endpoints$ir_id)
bmi_data$ir_id = as.character(bmi_data$ir_id)
patient_data$study_id = as.character(patient_data$study_id)
patient_data$ir_id = as.character(patient_data$ir_id)
sofa_data_for_edw_molecular$pt_study_id = as.character(sofa_data_for_edw_molecular$pt_study_id)
edw_molecular$MRN = as.character(edw_molecular$MRN)
vent_settings$ir_id = as.character(vent_settings$ir_id)
blood_gas$ir_id = as.character(blood_gas$ir_id)
complete_bal_data = full_join(edw_molecular, 
                           subset(patient_data, !is.na(study_id)),
                           by = c("study_id"), 
                           na_matches = "never") %>%
  select(-c(MRN)) %>% 
  #fix missing ir ids
  dplyr::select(-ir_id.x) %>% 
  dplyr::rename(ir_id = ir_id.y) %>% 
  full_join(.,
            edw_endpoints,
            by = c("ir_id", "study_id" = "pt_study_id"),
            na_matches = "never") %>% 
  select(-west_mrn) %>% 
  left_join(., bmi_data, by = "ir_id", na_matches = "never") %>% 
  left_join(., vent_settings, by = c("ir_id", "BAL_order_date" = "meas_date"), na_matches = "never")  %>% 
  left_join(., blood_gas, by = c("ir_id", "BAL_order_timestamp" = "merge_datetime"), na_matches = "never")

#fix colnames
colnames(flow_data) = gsub("\\.", "_", colnames(flow_data)) #I like underscores
colnames(flow_data) = gsub("_+$", "", colnames(flow_data)) # remove trailing
colnames(flow_data) = gsub("^_+", "", colnames(flow_data)) # remove leading
colnames(flow_data) = gsub("_+", "_", colnames(flow_data)) #remove dup underscores
colnames(complete_bal_data) = gsub("\\.", "_", colnames(complete_bal_data)) 
colnames(complete_bal_data) = gsub("_+$", "", colnames(complete_bal_data)) 
colnames(complete_bal_data) = gsub("^_+", "", colnames(complete_bal_data))
colnames(complete_bal_data) = gsub("_+", "_", colnames(complete_bal_data))
   
#add in healthy controls, which have no associated metadata
#no need to clean this as I put it together
healthy_metadata = read.csv("~/Box/RGrant/SCRIPT/200717_healthy_control_metadata.csv",
                            colClasses = c("character", "character", "numeric",
                                           "character", "logical", "character",
                                           "character", "character"))
complete_bal_data = bind_rows(complete_bal_data, healthy_metadata)

#derive additional metadata
complete_bal_data$PF_ratio = complete_bal_data$PO2_ART / complete_bal_data$FiO2
complete_bal_data$days_from_death = as.numeric(difftime(complete_bal_data$BAL_order_date, complete_bal_data$death_date, units = "days"))
complete_bal_data$covid_confirmed[is.na(complete_bal_data$covid_confirmed)] = FALSE
complete_bal_data$covid_confirmed = factor(complete_bal_data$covid_confirmed)
complete_bal_data$day_of_intubation = as.numeric(difftime(complete_bal_data$BAL_order_date, complete_bal_data$first_intubation_date, units = "days"))
complete_bal_data$days_intube_prior_enroll_redcap[complete_bal_data$days_intube_prior_enroll_redcap > 100] = Inf
complete_bal_data$days_intube_prior_enroll_redcap[is.na(complete_bal_data$days_intube_prior_enroll_redcap)] = 0
complete_bal_data$day_of_intubation = complete_bal_data$day_of_intubation + complete_bal_data$days_intube_prior_enroll_redcap
complete_bal_data$age = ifelse(!is.na(complete_bal_data$birth_date),
                            yes = time_length(difftime(complete_bal_data$BAL_order_date, 
                                                       complete_bal_data$birth_date), unit = "years"),
                            no = complete_bal_data$age)

#factor by major pathogen
major_pathogen_cols = c("any_viral_non_covid", "any_bacterial", 
                        "covid_confirmed")
pathogen = vector(mode = "character", length = nrow(complete_bal_data))
pathogens_list = vector(mode = "list", length = nrow(complete_bal_data))
for(i in 1:nrow(complete_bal_data))
{
  cur = complete_bal_data[i, major_pathogen_cols]
  pathogens = as.character(na.omit(colnames(cur)[cur == TRUE]))
  pathogens = gsub("^any_", "", pathogens)
  pathogens = gsub("covid_confirmed", "SARS-CoV-2", pathogens)
  if(length(pathogens) == 0)
  {
    pathogen[i] = NA
  } else if(length(pathogens) > 0)
  {
    pathogens_list[[i]] = pathogens
    if(length(pathogens) == 1)
    {
      pathogen[i] = pathogens
    } else
    {
      pathogen[i] = "multiple"
    }
  }
}
pathogen = factor(pathogen)
complete_bal_data$pathogen = pathogen
complete_bal_data$pathogen_list = pathogens_list

#factor by infection type
pathogen_type_cols = c("any_bacterial", "any_viral")
pathogen = vector(mode = "character", length = nrow(complete_bal_data))
pathogens_list = vector(mode = "list", length = nrow(complete_bal_data))
for(i in 1:nrow(complete_bal_data))
{
  cur = complete_bal_data[i, pathogen_type_cols]
  pathogens = as.character(na.omit(colnames(cur)[cur == TRUE]))
  pathogens = gsub("^any_", "", pathogens)
  if(length(pathogens) == 0)
  {
    pathogen[i] = NA
  } else if(length(pathogens) > 0)
  {
    pathogens_list[[i]] = pathogens
    if(length(pathogens) == 1)
    {
      pathogen[i] = pathogens
    } else
    {
      pathogen[i] = "multiple"
    }
  }
}
pathogen = factor(pathogen)
complete_bal_data$infection_type = pathogen
complete_bal_data$infection_type_list = pathogens_list

#add broad diagnosis
final_diagnosis = vector(mode = "character", length = nrow(complete_bal_data))
for(i in 1:nrow(complete_bal_data))
{
  #no matter what, this stays the same (even with COVID)
  if(!is.na(complete_bal_data$simplified_diagnosis[i]) && complete_bal_data$simplified_diagnosis[i] == "Non-Pneumonia Control")
  {
    final_diagnosis[i] = "Non-Pneumonia Control"
  } else if(!is.na(complete_bal_data$simplified_diagnosis[i]) && complete_bal_data$simplified_diagnosis[i] == "Healthy Control")
  {
    final_diagnosis[i] = "Healthy Control"
  } else if(complete_bal_data$covid_confirmed[i] == TRUE) #even if NA
  {
    final_diagnosis[i] = "COVID-19"
  } else if(is.na(complete_bal_data$simplified_diagnosis[i]))
  {
    final_diagnosis[i] = NA
  } else if(complete_bal_data$simplified_diagnosis[i] == "Pneumonia") #determine if viral
  {
    if((!is.na(complete_bal_data$any_viral_non_covid[i]) && complete_bal_data$any_viral_non_covid[i] == TRUE) || 
       (!is.na(complete_bal_data$clin_cap[i]) && complete_bal_data$clin_cap[i] %in% c(2,4)) ||
       (!is.na(complete_bal_data$clin_hap[i]) && complete_bal_data$clin_hap[i] == 4) ||
       (!is.na(complete_bal_data$clin_vap[i]) && complete_bal_data$clin_vap[i] == 4))
    {
      final_diagnosis[i] = "Other Viral Pneumonia"
    } else
    {
      final_diagnosis[i] = "Other Pneumonia"
    }
  } else
  {
    final_diagnosis[i] = NA
  }
}
final_diagnosis = factor(final_diagnosis,
                         levels = c("Healthy Control", "Non-Pneumonia Control", "COVID-19", 
                                    "Other Viral Pneumonia", "Other Pneumonia"))
complete_bal_data$pna_type = final_diagnosis

#remove extra diagnosis cols
complete_bal_data = complete_bal_data %>% 
  dplyr::select(-c(clin_cap, clin_vap, clin_hap)) %>% 
  unique()

#remove any remaining PHI
complete_bal_data = complete_bal_data %>% 
  select(-c(clarity_west_mrn, birth_date))


#adjust vent days to exclude days when extubated (useful for VAPs)
complete_bal_data = complete_bal_data %>% 
  dplyr::rename(BAL_order_accession_num = order_accession_num)
adjusted = correct_vent_days(endpoints = edw_endpoints, bal_data = complete_bal_data)
endpoints_long = adjusted$endpoints
complete_bal_data = adjusted$bal

#add superinfection
complete_bal_data = complete_bal_data %>% 
  mutate(super_type = case_when(pna_type == "Other Pneumonia" ~ NA_character_,
                                (any_nonviral == T & day_of_intubation <= 2 & day_of_intubation >= 0) ~ "Co-Infection",
                                any_nonviral == T ~ "Superinfection",
                                is.na(any_nonviral) ~ NA_character_, #this is annoying, but works
                                TRUE ~ "Primary Only"),
         Superinfection = case_when(super_type %in% c("Co-Infection", "Superinfection") ~ "Superinfection",
                                    super_type == "Primary Only" ~ "Primary Only")) %>% 
group_by(study_id) %>% 
  mutate(infection_category = case_when(pna_type == "Other Pneumonia" ~ NA_character_,
                                        any(super_type == "Co-Infection") ~ "Co-Infection",
                                        any(super_type == "Superinfection") ~ "Superinfection",
                                        all(super_type == "Primary Only") ~ "Primary Only")) %>% 
  ungroup()

#finally, fix ~2 remaining patients missing metadata   
still_unk = complete_bal_data %>% 
  dplyr::filter(is.na(ir_id)) %>% #always tracks with empty metadata
  dplyr::filter(!is.na(tc_pt_study_id) & !grepl("CO000\\d", tc_pt_study_id)) #no fix for these
if(nrow(still_unk) > 0)
{
  stop("Error: missing data still")
}


#merge with flow data
flow_data = flow_data %>% 
  dplyr::mutate(tc_pt_study_id = substring(Sample, 10, 20)) %>% 
  left_join(.,
            complete_bal_data,
            by = c("ID" = "study_id", "tc_pt_study_id"),
            na_matches = "never")

#add extra columns
flow_data = flow_data %>% 
  rowwise() %>% 
  dplyr::mutate(infection_detected = factor(!is.null(organism_quantity))) %>% 
  ungroup() %>% 
  unique() %>% 
  dplyr::filter(!duplicated(Sample)) #some must have duplicate barcodes

#cast into long form
flow_data = unique(flow_data)
mfi_cols = colnames(flow_data)[grepl("MFI", colnames(flow_data), ignore.case = T)]
percentage_cols = colnames(flow_data)[grepl("percent", colnames(flow_data), ignore.case = T)]
flow_data_long = flow_data %>% 
  pivot_longer(cols = c(mfi_cols, percentage_cols), 
               names_to = "flow_parameter",
               values_to = "value") %>% 
  arrange(ID, day_of_intubation) #for easy viewing later
flow_data_long$flow_parameter = factor(flow_data_long$flow_parameter)

#make separate item for only script patients
#remove bad data from EDW, patients past cutoff
script_bal_data = complete_bal_data %>% 
  dplyr::filter(!is.na(tc_pt_study_id) & #only known script samples
                  !(tc_pt_study_id %in% discards) & #manually remove ghost samples
                  as.numeric(study_id) < 1400 & #any others are typos, currently none
                  (hosp_admission_date <= as.Date("2020-07-06") | #date cutoff for cohort
                     study_id %in% c("0001", "0002", "0003", "0004", "0005") | #healthy controls, not hospitalized
                     tc_pt_study_id %in% c(keepers$tc_pt_study_id, recodes)))

                  

#export for other use
outname = paste0("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/",
                 Sys.Date(),
                 "_",
                 "SCRIPT_clinical_metadata_processed.rds")
saveRDS(object = script_bal_data,
        file = outname)
outname = paste0("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/",
                 Sys.Date(),
                 "_",
                 "SCRIPT_flow_plus_clinical_metadata_processed.rds")
saveRDS(object = flow_data,
        file = outname)

#for subsetting later
observations = table(flow_data$patient_id)
serial_patients = names(observations[observations > 1])

#add finite days
flow_data$finite_day_of_intubation = flow_data$day_of_intubation
flow_data$finite_day_of_intubation[is.infinite(flow_data$finite_day_of_intubation)] = NA

flow_data
```

### Adjudicate patients without diagnosis   
```{r eval=FALSE}
diags = script_bal_data %>% 
  dplyr::select(ir_id, study_id, pna_type) %>%
  unique()
missing_diagnosis = diags %>% 
  dplyr::filter(is.na(pna_type)) %>% 
  .$study_id

adjudication_data = script_bal_data %>% 
  dplyr::filter(study_id %in% missing_diagnosis) %>%
  dplyr::select(ir_id, study_id, tc_pt_study_id, 
                BAL_order_accession_num, BAL_order_date) %>% 
  mutate(bal_number = as.numeric(substring(tc_pt_study_id, nchar(tc_pt_study_id) - 1)),
         ir_id = as.numeric(ir_id)) %>% 
  left_join(., mrn_conv) %>% 
  group_by(study_id) %>% 
  slice(which.min(bal_number))
#write.csv(adjudication_data, "~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200927_adjudication_data.csv")
```


### Sanity checks   
```{r}
if(any(duplicated(script_bal_data$tc_pt_study_id)))
{
  stop("Duplicated samples: BAL")
}
if(any(duplicated(flow_data$tc_pt_study_id)))
{
  stop("Duplicated samples: flow")
}
```

```{r}
ggplot(flow_data, aes(x = percent_total_CD206_low, y = percent_neutrophils, color = panel)) +
  geom_point() +
  stat_smooth(method = "lm")
```

# Patient timelines   
## Prepare data and plot   
```{r}
col_classes = script_bal_data %>% 
  summarise_all(class)
important_dates = data.frame(column = colnames(col_classes),
                             class = as.character(col_classes[1,])) %>% 
  #remove uninteresting and/or redundant columns
  dplyr::filter(class == "Date" & !(column %in% c("consent_dt", "admission_datetime", 
                                                  "First_intub_start", "tracheostomy_dt")))
timeline_data_covid = script_bal_data %>% 
  dplyr::filter(covid_confirmed == TRUE) %>% 
  dplyr::select(all_of(setdiff(important_dates$column, c("BAL_first_date", "bal_date"))), binned_outcome, study_id, visit_id) %>% 
  pivot_longer(cols = first_intubation_date:Third_intub_stop, names_to = "event", values_to = "date") %>% 
  dplyr::filter(!is.na(date)) %>% 
  mutate(event = gsub("_date|_datetime|First_|first_|Second_|Third_|_start", "", event)) %>% 
  mutate(event = gsub("^intub$", "intubation", event)) %>% 
  mutate(event = gsub("^intub_", "intubation_", event)) %>% 
  mutate(event = gsub("hosp_disch", "discharge", event)) %>% 
  mutate(event = factor(event)) %>% 
  #remove in-progress patients   
  dplyr::filter(!is.na(binned_outcome))

#relative dates
first_intubations = timeline_data_covid %>% 
  group_by(study_id) %>% 
  dplyr::filter(event == "intubation") %>% 
  slice(which.min(date)) %>% 
  select(study_id, first_intubation_date = date)
timeline_data_covid = timeline_data_covid %>% 
  left_join(., first_intubations, na_matches = "never") %>% 
  dplyr::mutate(relative_day = difftime(date, first_intubation_date, units = "days"))  %>% 
  dplyr::mutate(study_id = fct_reorder(.f = study_id, .x = relative_day, .fun = max, .desc = FALSE)) %>%  #reorder descending
  dplyr::filter(!is.na(binned_outcome)) %>% 
  unique() %>% 
  mutate(event = gsub("Deceased", "death", event)) %>% 
  #some deaths still have a discharge date...not relevant
  dplyr::filter(!(binned_outcome == "Deceased" & event == "discharge"))

#remove discharge vent stops
remove = c()
end_events = c("death", "Discharged",
               "Inpatient Facility", "Other")
for(ev in which(timeline_data_covid$event == "intubation_stop"))
{
  cur_pt = timeline_data_covid$study_id[ev]
  event_date = timeline_data_covid$date[ev]
  cur_sub = subset(timeline_data_covid, study_id == cur_pt & date == event_date)
  if(any(cur_sub$event %in% end_events))
  {
    remove = append(remove, ev)
  }
}

intubation_periods_covid = timeline_data_covid %>% 
  dplyr::filter(event %in% c("intubation", "intubation_stop", 
                             "death", "Discharged",
                             "Inpatient Facility")) 
#for each intubation start, find nearest end event (extubation or discharge)
stop = vector(mode = "character", length = nrow(intubation_periods_covid))
for(pt in unique(intubation_periods_covid$study_id))
{
  starts = which(intubation_periods_covid$study_id == pt & 
                   intubation_periods_covid$event == "intubation")
  for(intub in starts)
  {
    intub_date = intubation_periods_covid$date[intub]
    stop_subset = subset(intubation_periods_covid, 
                             study_id == pt &
                               date >= intub_date &
                               event != "intubation")
    first_stop = as.character(min(stop_subset$date))
    stop[intub] = first_stop
  }
}
intubation_periods_covid = intubation_periods_covid %>% 
  mutate(stop = as.Date(stop)) %>% 
  dplyr::filter(event == "intubation" & binned_outcome != "Other") %>% 
  dplyr::mutate(relative_day_end = difftime(stop, first_intubation_date, units = "days"))

  
#remove unnecessary event types
timeline_data_covid = timeline_data_covid %>% 
  dplyr::filter(!grepl("intubation", event) & binned_outcome != "Other") %>% 
  dplyr::mutate(event = factor(as.character(event))) #refactor

get_max_day = function(patient)
{
  max_intub = intubation_periods_covid %>% 
    dplyr::filter(study_id == patient) %>% 
    .$relative_day_end %>% 
    as.numeric() %>% 
    max(., na.rm = T)
  max_event = timeline_data_covid %>% 
    dplyr::filter(study_id == patient) %>% 
    .$relative_day %>% 
    as.numeric() %>% 
    max(., na.rm = T)
  
  return(max(max_intub, max_event, na.rm = T))
}
get_min_day = function(patient)
{
  min_intub = intubation_periods_covid %>% 
    dplyr::filter(study_id == patient) %>% 
    .$relative_day %>% 
    as.numeric() %>% 
    min(., na.rm = T)
  min_event = timeline_data_covid %>% 
    dplyr::filter(study_id == patient) %>% 
    .$relative_day %>% 
    as.numeric() %>% 
    min(., na.rm = T)
  
  return(min(min_intub, min_event, na.rm = T))
}
stay_lengths = timeline_data_covid %>% 
  dplyr::select(study_id, binned_outcome) %>% 
  unique() %>% 
  rowwise() %>% 
  mutate(stay_length = get_max_day(study_id),
         stay_start = get_min_day(study_id))

final_timeline = ggplot(timeline_data_covid, aes(x = relative_day, y = study_id)) +
  geom_segment(data = stay_lengths, 
               aes(x = stay_start, xend = stay_length, y = study_id, yend = study_id),
               color = "grey60",
               size = 0.5) +
  geom_segment(data = intubation_periods_covid, 
            aes(x = relative_day, xend = relative_day_end, y = study_id, yend = study_id),
            color = pal[1],
            alpha = 0.5,
            lineend = "round", 
            size = 2) +
  geom_point(aes(color = event, shape = event), size = 2.5) +
  facet_grid(binned_outcome ~ ., scales = "free_y", space = "free_y") +
  xlab("Day of Intubation") +
  ylab("") +
  scale_color_manual(name = "",
                     values = c("hosp_admission" = pal[6],
                                "BAL_order" = "grey44",
                                "discharge" = pal[3],
                                "death" = pal[8]),
                     labels = c("hosp_admission" = "Hospital Admission",
                                "BAL_order" = "BAL",
                                "discharge" = "Discharged",
                                "death" = "Death")) +
  scale_shape_manual(name = "",
                     values = c("hosp_admission" = 15,
                                "BAL_order" = 18,
                                "discharge" = 19,
                                "death" = 13),
                     labels = c("hosp_admission" = "Hospital Admission",
                                "BAL_order" = "BAL",
                                "discharge" = "Discharged",
                                "death" = "Death")) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        axis.ticks.y = element_blank(), 
        axis.text.y = element_blank(), 
        panel.grid.major.y = element_blank())

final_timeline
```
   
### Compare distributions of sampling times   
By day of BAL   
```{r}
sampling_dists = script_bal_data %>% 
  dplyr::select(study_id, BAL_order_accession_num, pna_type, day_of_intubation) %>% 
  unique() %>% 
  dplyr::filter(!is.na(day_of_intubation) & 
                  is.finite(day_of_intubation) &
                  !is.na(pna_type))
#sanity check
any(!is.na(sampling_dists$BAL_order_accession_num) & duplicated(sampling_dists$BAL_order_accession_num))

shapiro.test(sampling_dists$day_of_intubation)
hist(sampling_dists$day_of_intubation) #huge leftward skew, as expected

#compare by PNA type
kruskal.test(day_of_intubation ~ pna_type, data = sampling_dists) #so there is some skew
sampling_comps = pairwise.wilcox.test(sampling_dists$day_of_intubation, sampling_dists$pna_type, p.adjust.method = "fdr")

ggplot(sampling_dists, aes(x = pna_type, y = day_of_intubation, fill = pna_type)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.5, width = 0.15) +
  scale_fill_npg() +
  xlab("Diagnosis") +
  ylab("Day of Intubation") +
  theme_bw() +
  theme(legend.position = "none") +
   geom_signif(comparisons = list(c("COVID-19", "Non-Pneumonia Control"),
                                  c("Other Pneumonia", "Non-Pneumonia Control")),
              annotations = c(format(sampling_comps$p.value["COVID-19", "Non-Pneumonia Control"], digits = 2, scientific = T),
                              format(sampling_comps$p.value["Other Pneumonia", "Non-Pneumonia Control"], digits = 2, scientific = T)),
              y_position = c(65, 75, 100, 85), 
              tip_length = 0,
              aes(y = 1))
```   
   
```{r}
sampling_freqs = script_bal_data %>% 
  dplyr::select(study_id, pna_type, day_of_intubation, total_icu_los_days) %>% 
  unique() %>% 
  dplyr::filter(!is.na(day_of_intubation) & 
                  is.finite(day_of_intubation) &
                  !is.na(pna_type) &
                  !is.na(total_icu_los_days)) %>% 
  group_by(study_id) %>% 
  dplyr::summarize(study_id = first(study_id), #these are all the same type
                   pna_type = first(pna_type),
                   total_icu_los_days = first(total_icu_los_days),
                   n_bal = n()) %>% 
  group_by(study_id) %>% 
  dplyr::summarize(study_id = first(study_id),
                   pna_type = first(pna_type),
                   bal_per_vent_day = n_bal / total_icu_los_days)

shapiro.test(sampling_freqs$bal_per_vent_day)
hist(sampling_freqs$bal_per_vent_day) #huge leftward skew again

#compare by PNA type
kruskal.test(bal_per_vent_day ~ pna_type, data = sampling_freqs) #so there is some skew
sampling_comps = pairwise.wilcox.test(sampling_freqs$bal_per_vent_day, sampling_freqs$pna_type, p.adjust.method = "fdr")

ggplot(sampling_freqs, aes(x = pna_type, y = bal_per_vent_day, fill = pna_type)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.5, width = 0.15) +
  scale_fill_npg() +
  xlab("Diagnosis") +
  ylab("BALs per Ventilator Day") +
  theme_bw() +
  theme(legend.position = "none") +
   geom_signif(comparisons = list(c("COVID-19", "Non-Pneumonia Control"),
                                  c("Other Pneumonia", "COVID-19")), 
              annotations = c(format(sampling_comps$p.value["COVID-19", "Non-Pneumonia Control"], digits = 2, scientific = T),
                              format(sampling_comps$p.value["Other Pneumonia", "COVID-19"], digits = 2, scientific = T)),
              y_position = c(0.6, 0.75), 
              tip_length = 0,
              aes(y = 1))
```
   
### Distribution by day   
```{r}
script_bal_data = script_bal_data %>% 
  mutate(finite_day_of_intubation = ifelse(is.infinite(day_of_intubation),
                                           yes = NA,
                                           no = day_of_intubation))
days = c(0:max(script_bal_data$finite_day_of_intubation, na.rm = T))
sampling_timing = data.frame(day_of_intubation = rep(days, 4),
                             Diagnosis = c(rep("Non-Pneumonia Control", length(days)),
                                                  rep("COVID-19", length(days)),
                                                  rep("Other Viral Pneumonia", length(days)),
                                                  rep("Other Pneumonia", length(days)))) %>% 
  group_by(Diagnosis, day_of_intubation) %>% 
  mutate(n_bal = sum(script_bal_data$pna_type == Diagnosis & 
                            script_bal_data$day_of_intubation == day_of_intubation,
                          na.rm = T),
         n_patients = length(unique(script_bal_data$study_id[script_bal_data$pna_type == Diagnosis & 
                            script_bal_data$total_icu_los_days >= day_of_intubation])),
         sampling_rate = n_bal / n_patients)

pts_enrolled_plot = ggplot(sampling_timing, aes(x = day_of_intubation, y = n_patients, color = Diagnosis)) +
        geom_line(size = 2) +
        scale_color_manual(name = "",
                          values = c("Non-Pneumonia Control" = pal[2],
                                     "COVID-19" = pal[1],
                                     "Other Viral Pneumonia" = pal[3],
                                     "Other Pneumonia" = pal[4])) +
        theme_bw(base_family = "Arial") +
        xlab("Day of Intubation") +
        ylab("Patients") +
  ggtitle("Enrolled Patients") +
  theme(plot.title = element_text(hjust = 0.5),
        legend.position = "none") +
  guides(color=guide_legend(nrow=2,byrow=TRUE))

daily_BAL_plot = ggplot(sampling_timing, aes(x = day_of_intubation, y = n_bal, color = Diagnosis)) +
        geom_line(size = 2) +
        scale_color_manual(name = "",
                          values = c("Non-Pneumonia Control" = pal[2],
                                     "COVID-19" = pal[1],
                                     "Other Viral Pneumonia" = pal[3],
                                     "Other Pneumonia" = pal[4])) +
        theme_bw(base_family = "Arial") +
        xlab("Day of Intubation") +
        ylab("BALs") +
  ggtitle("BALs per Day") +
  theme(plot.title = element_text(hjust = 0.5),
        legend.position = "bottom") +
  guides(color=guide_legend(nrow=2,byrow=TRUE))

sampling_rate_plot = ggplot(sampling_timing, aes(x = day_of_intubation, y = sampling_rate, color = Diagnosis)) +
        geom_line(size = 2) +
        scale_color_manual(name = "",
                          values = c("Non-Pneumonia Control" = pal[2],
                                     "COVID-19" = pal[1],
                                     "Other Viral Pneumonia" = pal[3],
                                     "Other Pneumonia" = pal[4])) +
        theme_bw(base_family = "Arial") +
        xlab("Day of Intubation") +
        ylab("BAL Sampling Rate") +
  ggtitle("BALs per Day") +
  theme(plot.title = element_text(hjust = 0.5),
        legend.position = "none") +
  guides(color=guide_legend(nrow=2,byrow=TRUE))
```

### Metadata  
```{r}
complete_patient_data = script_bal_data %>% 
  dplyr::select(study_id, covid_confirmed, pna_type) %>% 
  unique()
missing_diagnosis = subset(complete_patient_data, is.na(pna_type))
missing_diagnosis
```   
   
### Flow   
```{r}
table(flow_data$covid_confirmed, useNA = "always") #samples

#pts
flow_patients = flow_data %>% 
  dplyr::select(ID, covid_confirmed) %>% 
  unique()
table(flow_patients$covid_confirmed, useNA = "always")
```   
   
### BALs   
```{r}
# per patient
n_bals = script_bal_data %>% 
  group_by(study_id) %>% 
  tally()
mean(n_bals$n)
range(n_bals$n)

#by group
script_bal_data %>% 
  dplyr::select(study_id, pna_type) %>% 
  unique() %>% 
  group_by(pna_type) %>% 
  tally()
```
   
### Superinfection   
```{r}
script_bal_data %>% 
  dplyr::filter(pna_type == "COVID-19") %>% 
  dplyr::select(study_id, infection_category) %>% 
  unique() %>% 
  group_by(infection_category) %>% 
  dplyr::summarize(pct = n() / nrow(.) * 100)

script_bal_data %>% 
  dplyr::filter(pna_type == "COVID-19" & infection_category %in% c("Co-Infection", "Superinfection")) %>% 
  dplyr::select(study_id, infection_category) %>% 
  unique() %>% 
  group_by(infection_category) %>% 
  dplyr::summarize(pct = n() / nrow(.) * 100)
```   
   
### Date range   
```{r}
min(script_bal_data$hosp_admission_date, na.rm = T)
max(script_bal_data$hosp_admission_date, na.rm = T)
min(subset(script_bal_data, pna_type == "COVID-19")$hosp_admission_date, na.rm = T)
max(subset(script_bal_data, pna_type == "COVID-19")$hosp_admission_date, na.rm = T)
```



## Neutrophilic patients   
### Flow (all days)   
```{r}
table(flow_data$covid_confirmed, flow_data$neutrophilic)
```   

### Flow (D0)   
```{r}
flow0 = subset(flow_data, day_of_intubation <= 2 & day_of_intubation >= 0)
flow0_patients = flow0 %>% 
  select(pna_type, ID) %>% 
  unique()
covid_neutrophilia_early = table(flow0$covid_confirmed, flow0$neutrophilic)
covid_neutrophilia_early["TRUE", "Neutrophilic"] / sum(covid_neutrophilia_early["TRUE", "Neutrophilic"],
                                                      covid_neutrophilia_early["TRUE", "Non-Neutrophilic"])
```
   
## More aesthetic   
### Aggregate   
```{r}
fig1_data = script_bal_data %>% 
  rowwise() %>% 
  mutate(`Days of Intubation` = sum(difftime(First_intub_stop, First_intub_start, units = "days"),
                                    difftime(Second_intub_stop, Second_intub_start, units = "days"),
                                    difftime(Third_intub_stop, Third_intub_start, units = "days"),
                                    na.rm = T)) %>% 
  ungroup() %>% 
  dplyr::select(study_id, Ethnicity = ethnicity, Race = races, 
                Age = age, Sex = gender, `Discharge Status` = binned_outcome,
                `Smoking Status` = smoking_status, `Pack Years` = pack_years,
                Diagnosis = pna_type, `Days of Intubation`,
                `Length of ICU Stay` = total_icu_los_days, BMI, covid_confirmed,
                SOFA = mean_sofa, APS = mean_aps) %>% 
  mutate_at(all_of(c("study_id", "Ethnicity", "Race", "Sex")), factor) %>% 
  dplyr::mutate(Age = ifelse(Age >= 110, yes = NA, no = Age), #this is standard NMH code
                `Days of Intubation` = ifelse(`Days of Intubation` > 365, #chronic vent patients
                                              yes = NA,
                                              no = `Days of Intubation`)) %>% 
  group_by(study_id) %>% 
  #at least 3 patients had a birthday in the hospital
  dplyr::mutate(Age = floor(mean(Age, na.rm = T)),
                `Days of Intubation` = max(`Days of Intubation`, na.rm = T), #not strictly necessary, but removes NA
                `Length of ICU Stay` = max(`Length of ICU Stay`, na.rm = T)) %>% 
  ungroup() %>% 
  unique() %>%
  #ignore chronic vent for this purpose
  dplyr::mutate(`Days of Intubation` = ifelse(is.infinite(`Days of Intubation`), yes = NA, no = `Days of Intubation`)) %>% 
  dplyr::filter(!duplicated(study_id) & !is.na(study_id) & grepl("\\d{4}", study_id)) %>% 
  dplyr::filter(!is.na(`Discharge Status`)) %>%  #remove in-progress 
  dplyr::mutate(Outcome = factor(ifelse(`Discharge Status` == "Deceased",
                                 yes = "Deceased",
                                 no = "Discharged"))) %>% 
  dplyr::filter(!is.na(Diagnosis)) %>% 
  column_to_rownames(var = "study_id")

#make some manual fixes
fig1_data$Race[is.na(fig1_data$Race)] = "Unknown or Not Reported"
fig1_data$Ethnicity[is.na(fig1_data$Ethnicity)] = "Unknown or Not Reported"
fig1_data$Ethnicity = factor(fig1_data$Ethnicity,
                             levels = c("Hispanic or Latino", "Not Hispanic or Latino", "Unknown or Not Reported"))
fig1_data$Sex = as.character(fig1_data$Sex)
fig1_data$Sex[is.na(fig1_data$Sex)] = "Unknown or Not Reported"
fig1_data$Sex = factor(fig1_data$Sex)
fig1_data$`Smoking Status` = as.character(fig1_data$`Smoking Status`)
fig1_data$`Smoking Status`[is.na(fig1_data$`Smoking Status`)] = "Unknown or Not Reported"
fig1_data$`Smoking Status` = factor(fig1_data$`Smoking Status`)
fig1_data$`Pack Years`[fig1_data$`Pack Years` > 150] = NA
#not sure how to handle chronic vent pts
fig1_data$`Days of Intubation`[is.infinite(fig1_data$`Days of Intubation`)] = NA
fig1_data$`Length of ICU Stay`[is.infinite(fig1_data$`Length of ICU Stay`)] = NA
fig1_data$Diagnosis = str_wrap(fig1_data$Diagnosis, width = 14)
fig1_data$Diagnosis = gsub("- ", "-", fig1_data$Diagnosis)
fig1_data$Diagnosis = factor(fig1_data$Diagnosis,
                             levels = c("Non-Pneumonia\nControl", "COVID-19",
                                        "Other Viral\nPneumonia", "Other\nPneumonia"))

print(dfSummary(fig1_data, plain.ascii = FALSE, style = "multiline",
                varnumbers = F, na.col = F, headings = F,
                graph.magnif = 0.75, valid.col = FALSE, tmp.img.dir = "/tmp"),
      method = "viewer")
```
   
## List potential confounds   
### Experimental therapies   
```{r}
id_conv = script_bal_data %>% 
  dplyr::select(ir_id, study_id) %>% 
  unique() 

n_covid = sum(complete_patient_data$covid_confirmed == TRUE)
ms1_t1 = read_excel("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/200927 TABLE 1 missing data v3 FINAL.xlsx") %>%
  dplyr::filter(ir_id %in% unique(script_bal_data$ir_id)) %>% 
  rowwise() %>% 
  dplyr::mutate(remdesivir = max(`Remdesivir study`, `Remdesivir EUA`, na.rm = T),
                sarilumab = max(`Sari/Toci off label`, `Sarilumab study`, na.rm = T),
                ir_id = as.character(ir_id)) %>% 
  ungroup() %>% 
  dplyr::select(ir_id, remdesivir, sarilumab, Steroids, HCQ) %>% 
  left_join(., id_conv)

ms1_t1_sum = ms1_t1 %>% 
  dplyr::summarize(n_sarilumab = sum(sarilumab, na.rm = T),
                   pct_sarilumab = sum(sarilumab, na.rm = T) / n_covid * 100,
                   n_remdesivir = sum(remdesivir, na.rm = T),
                   pct_remdesivir = sum(remdesivir, na.rm = T) / n_covid * 100,
                   n_steroids = sum(Steroids, na.rm = T),
                   pct_steroids = sum(Steroids, na.rm = T) / n_covid * 100,
                   n_hydroxychloroquine = sum(HCQ, na.rm = T),
                   pct_hydroxychloroquine = sum(HCQ, na.rm = T) / n_covid * 100)

ms1_t1_sum
```
   
### Standard table   
```{r}
options(qwraps2_markup = "markdown")
summary_data = fig1_data %>%
  dplyr::select(-c(covid_confirmed, `Pack Years`)) %>%
  rownames_to_column("study_id") %>% 
  left_join(., ms1_t1) %>% 
  dplyr::select(-c(ir_id)) %>% 
  column_to_rownames("study_id") %>% 
  dplyr::rename(`Sarilumab Study` = sarilumab, `Remdesivir Study` = remdesivir,
                Hydroxychloroquine = HCQ) %>% 
  dplyr::mutate(Diagnosis = factor(as.character(gsub("\n", " ", Diagnosis)))) %>%
  dplyr::mutate(SOFA = round(SOFA, digits = 2), APS = round(APS, digits = 2)) %>% 
  dplyr::mutate(`Sarilumab Study` = case_when(`Sarilumab Study` == 1 ~ "Enrolled",
                                              TRUE ~ "Not Enrolled"),
                `Remdesivir Study` = case_when(`Remdesivir Study` == 1 ~ "Enrolled",
                                              TRUE ~ "Not Enrolled"),
                Hydroxychloroquine = case_when(Hydroxychloroquine == 1 ~ "Treated",
                                              TRUE ~ "Unreated"),
                Steroids = case_when(Steroids == 1 ~ "Treated",
                                              TRUE ~ "Unreated")) %>% 
  group_by(Diagnosis) %>%
  summary_table(.,
                summaries = qsummary(., numeric_summaries = list("Minimum" = "~ min(%s)",
                                         "Median (IQR)" = "~median_iqr(%s)",
                                         "Mean (SD)" = "~mean_sd(%s)",
                                         "Maximum" = "~ max(%s)"),
                                     n_perc_args = list(digits = 1, show_symbol = TRUE, 
                                                        show_denom = "always")))
  
#copied directly into md file below, rendered with knitr
print(summary_data, file = "~/Box/RGrant/SCRIPT/manuscript_1/200804_table_1.md")
```

   
## List pathogens   
### All detected (table 2)   
```{r eval=FALSE}
covid_bal = script_bal_data %>% 
  dplyr::filter(pna_type == "COVID-19")
covid_pathogens = data.frame(Pathogen = sort(unique(unlist(covid_bal$organism_name))))
covid_pathogens$Pathogen = gsub(" Note.+$", "", covid_pathogens$Pathogen)
covid_pathogens$Pathogen = gsub(" \\([^G].+| #.+| Ceftolozane.+| Beta Lactamase.+| Too.+|Methicillin-Resistant | group.+| Group$", 
                              "", 
                              covid_pathogens$Pathogen)
covid_pathogens = unique(covid_pathogens) %>% 
  remove_rownames()

all_pathogens = data.frame(Pathogen = sort(unique(unlist(script_bal_data$organism_name))))
all_pathogens$Pathogen = gsub(" Note.+$", "", all_pathogens$Pathogen)
all_pathogens$Pathogen = gsub(" \\([^G].+| #.+| Ceftolozane.+| Beta Lactamase.+| Too.+|Methicillin-Resistant | group.+| Group$", 
                              "", 
                              all_pathogens$Pathogen)
all_pathogens = unique(all_pathogens) %>% 
  remove_rownames()

tex = print.xtable(xtable(all_pathogens), 
                   print.results = FALSE, 
                   floating = FALSE)
setwd("~/Box/RGrant/SCRIPT/")
writeLines(
  c(
     "\\documentclass[12pt]{standalone}",
    "\\usepackage{caption}",
    "\\begin{document}",
    "\\minipage{\\textwidth}",
    tex,
     "\\captionof{table}{Pneumonia-causing pathogens detected in the SCRIPT cohort}",
    "\\endminipage",
    "\\end{document}"
  ),
  con = "~/Box/RGrant/SCRIPT/200728_table_2.tex"
)
tools::texi2pdf("200728_table_2.tex", clean = TRUE)
write.csv(all_pathogens, "~/Box/RGrant/SCRIPT/manuscript_1/200727_pathogens_list.csv")
```   
   
### Day 0-2   
```{r}
day0 = subset(script_bal_data, day_of_intubation <= 2 & day_of_intubation >= 0)
all_pathogens_0 = data.frame(Pathogen = sort(unique(unlist(day0$organism_name))))
all_pathogens_0$Pathogen = gsub(" Note.+$", "", all_pathogens_0$Pathogen)
all_pathogens_0$Pathogen = gsub(" \\([^G].+| #.+| Ceftolozane.+| Beta Lactamase.+| Too.+|Methicillin-Resistant | group.+", 
                              "", 
                              all_pathogens_0$Pathogen)
all_pathogens_0 = unique(all_pathogens_0) %>% 
  remove_rownames()

tex = print.xtable(xtable(all_pathogens_0), 
                   print.results = FALSE, 
                   floating = FALSE)
setwd("~/Box/RGrant/SCRIPT/")
writeLines(
  c(
     "\\documentclass[12pt]{standalone}",
    "\\usepackage{caption}",
    "\\begin{document}",
    "\\minipage{\\textwidth}",
    tex,
     "\\captionof{table}{Pneumonia-causing pathogens detected in the SCRIPT cohort}",
    "\\endminipage",
    "\\end{document}"
  ),
  con = "~/Box/RGrant/SCRIPT/200728_table_2.tex"
)
tools::texi2pdf("200728_table_2.tex", clean = TRUE)
write.csv(all_pathogens_0, "~/Box/RGrant/SCRIPT/200727_pathogens_list.csv")
```


   
## Comparisons   
Race/ethnicity   
```{r}
race_plots = ggplot(fig1_data, aes(x = Diagnosis, fill = Race)) +
  geom_bar(position = "fill") +
  ylab("Proportion") +
  guides(fill=guide_legend(nrow=3,byrow=TRUE)) +
  scale_fill_manual(name = "",
                    values = c("American Indian/Alaska Native" = "#3C5488FF",
                               "Asian" = "#00A087FF",
                               "Black/African American" = "#E64B35FF",
                               "Unknown or Not Reported" = "#F39B7FFF",
                               "White" = "#4DBBD5FF")) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5)) +
  ggtitle("Race") +
  xlab("")
race_table = table(fig1_data$Diagnosis, fig1_data$Race)
white_table = cbind(race_table[, "White", drop = FALSE], (rowSums(race_table) - race_table[, "White", drop = FALSE]))
colnames(white_table) = c("White", "Other")
white_comps = pairwise.prop.test(white_table, p.adjust.method = "fdr")
ethnicity_table = table(fig1_data$Diagnosis, fig1_data$Ethnicity)
ethnicity_table = ethnicity_table[, 1:2]
ethnicity_comps = pairwise.prop.test(ethnicity_table, p.adjust.method = "fdr")
ethnicity_plots = ggplot(fig1_data, aes(x = Diagnosis, fill = Ethnicity)) +
  geom_bar(position = "fill") +
  ylab("Proportion") +
  guides(fill=guide_legend(nrow=2,byrow=TRUE)) +
  scale_fill_npg(name = "") +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5)) +
  ggtitle("Ethnicity") +
  xlab("") +
  geom_signif(comparisons = list(c("COVID-19", "Non-Pneumonia\nControl"),
                                 c("Other Viral\nPneumonia", "COVID-19"),
                                 c("Other\nPneumonia", "COVID-19"),
                                 c("Other\nPneumonia", "Other Viral\nPneumonia")), 
              annotations = c(format(ethnicity_comps$p.value["COVID-19", "Non-Pneumonia\nControl"], digits = 2, scientific = T),
                              format(ethnicity_comps$p.value["Other Viral\nPneumonia", "COVID-19"], digits = 2, scientific = T),
                              format(ethnicity_comps$p.value["Other\nPneumonia", "COVID-19"], digits = 2, scientific = T),
                              format(ethnicity_comps$p.value["Other\nPneumonia", "Other Viral\nPneumonia"], digits = 2, scientific = T)),
              y_position = c(1.05, 1.15, 1.25, 1.05), 
              tip_length = 0,
              aes(y = 1)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2))
race_ethnicity = race_plots + ethnicity_plots
race_ethnicity
```

Age, Sex, BMI   
```{r}
fig1_data$Age[fig1_data$Age > 99] = NA #two erroneous entries
age_comps = pairwise.t.test(x = fig1_data$Age, g = fig1_data$Diagnosis, p.adjust.method = "fdr") #none significant
age_plot = ggplot(fig1_data, aes(x = Diagnosis, y = Age, fill = Diagnosis)) +
  geom_boxplot(notch = F, width = 0.5, outlier.shape = NA) + 
  ylab("Age at Admission (Years)") +
  geom_jitter(width = 0.25, size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia\nControl" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral\nPneumonia" = pal[3],
                               "Other\nPneumonia" = pal[4])) +
  theme_bw(base_family = "Arial") +
   theme(legend.position = "none",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5)) +
  ggtitle("Age") +
  xlab("")

sex_data = table(fig1_data$Diagnosis, fig1_data$Sex)
sex_comps = pairwise.prop.test(sex_data, p.adjust.method = "fdr") #none significant
sex_test = prop.test(sex_data)
sex_plot = ggplot(fig1_data, aes(x = Diagnosis, fill = Sex)) +
  geom_bar(position = "fill") +
  ylab("Proportion") +
  scale_fill_npg(name = "") +
  theme_bw(base_family = "Arial") +
  guides(fill = guide_legend(ncol=2)) +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5)) +
  ggtitle("Sex") +
  xlab("")

bmi_comps = pairwise.t.test(x = fig1_data$BMI, g = fig1_data$Diagnosis, p.adjust.method = "fdr") 
bmi_plot = ggplot(fig1_data, aes(x = Diagnosis, y = BMI, fill = Diagnosis)) +
  geom_boxplot(notch = F, width = 0.5, outlier.shape = NA) + 
  ylab("Body Mass Index (BMI)") +
  geom_jitter(width = 0.25, size = 0.15) +
  geom_signif(comparisons = list(c("COVID-19", "Other Viral\nPneumonia"),
                                 c("COVID-19", "Other\nPneumonia")), 
              annotations = c(format(bmi_comps$p.value["Other Viral\nPneumonia", "COVID-19"], digits = 2, scientific = T),
                              format(bmi_comps$p.value["Other\nPneumonia", "COVID-19"], digits = 2, scientific = T)),
              y_position = c(60, 70), 
              tip_length = 0,
              aes(y = 1)) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia\nControl" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral\nPneumonia" = pal[3],
                               "Other\nPneumonia" = pal[4])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5)) +
  ggtitle("BMI") +
  xlab("")

age_sex_bmi = sex_plot + age_plot + bmi_plot
age_sex_bmi
```

LOS   
```{r}
los_comps = pairwise.t.test(x = fig1_data$`Length of ICU Stay`, g = fig1_data$Diagnosis, p.adjust.method = "fdr") #all significant
los_plot = ggplot(fig1_data, aes(x = Diagnosis, y = `Length of ICU Stay`, fill = Diagnosis)) +
  geom_boxplot(notch = F, width = 0.5, outlier.shape = NA) + 
  ylab("Length of ICU Stay (days)") +
  geom_jitter(width = 0.25, size = 0.15) +
  geom_signif(comparisons = list(c("COVID-19", "Other Viral\nPneumonia"),
                                 c("COVID-19", "Other\nPneumonia"),
                                   c("COVID-19", "Non-Pneumonia\nControl"),
                                 c("Other\nPneumonia", "Non-Pneumonia\nControl")), 
              annotations = c(format(los_comps$p.value["Other Viral\nPneumonia", "COVID-19"], digits = 2, scientific = T),
                              format(los_comps$p.value["Other\nPneumonia", "COVID-19"], digits = 2, scientific = T),
                              format(los_comps$p.value["COVID-19", "Non-Pneumonia\nControl"], digits = 2, scientific = T),
                              format(los_comps$p.value["Other\nPneumonia", "Non-Pneumonia\nControl"], digits = 2, scientific = T)),
              y_position = c(70, 90, 95, 105), 
              tip_length = 0,
              aes(y = 1)) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia\nControl" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral\nPneumonia" = pal[3],
                               "Other\nPneumonia" = pal[4])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5)) +
  ggtitle("Length of Stay") +
  xlab("") +
  ylim(0, 110)

vd_comps = pairwise.t.test(x = fig1_data$`Days of Intubation`, g = fig1_data$Diagnosis, p.adjust.method = "fdr") #all significant
vent_days_plot = ggplot(fig1_data, aes(x = Diagnosis, y = `Days of Intubation`, fill = Diagnosis)) +
  geom_boxplot(notch = F, width = 0.5, outlier.shape = NA) + 
  ylab("Days on Ventilator") +
  geom_jitter(width = 0.25, size = 0.15) +
  geom_signif(comparisons = list(c("COVID-19", "Non-Pneumonia\nControl"),
                                 c("Other Viral\nPneumonia", "COVID-19"),
                                 c("Other\nPneumonia", "COVID-19")), 
              annotations = c(format(vd_comps$p.value["COVID-19", "Non-Pneumonia\nControl"], digits = 2, scientific = T),
                              format(vd_comps$p.value["Other Viral\nPneumonia", "COVID-19"], digits = 2, scientific = T),
                              format(vd_comps$p.value["Other\nPneumonia", "COVID-19"], digits = 2, scientific = T)), 
              y_position = c(70, 85, 120), 
              tip_length = 0,
              aes(y = 1)) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia\nControl" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral\nPneumonia" = pal[3],
                               "Other\nPneumonia" = pal[4])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5)) +
  ggtitle("Duration of Ventilation") +
  xlab("")

duration_plots = los_plot + vent_days_plot
duration_plots
```

Outcomes   
```{r}
discharge_plot =  ggplot(fig1_data, aes(x = Diagnosis, fill = `Discharge Status`)) +
  geom_bar(position = "fill") +
  ylab("Proportion")  +
  guides(fill=guide_legend(nrow=2,byrow=TRUE)) +
  scale_fill_npg(name = "") +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5)) +
  ggtitle("Outcomes") +
  xlab("")

mortality_sum = table(fig1_data$Diagnosis, fig1_data$Outcome)
mortality_test = pairwise.prop.test(mortality_sum, p.adjust.method = "fdr") #none significant
mortality_sum_covid = table(fig1_data$covid_confirmed, fig1_data$Outcome)
mortality_test_covid = prop.test(mortality_sum_covid)

outcome_plot =  ggplot(fig1_data, aes(x = Diagnosis, fill = Outcome)) +
  geom_bar(position = "fill") +
  ylab("Proportion") +
  scale_y_continuous(breaks = seq(0, 1, by = 0.25)) +
  scale_fill_npg(name = "") +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5)) +
  ggtitle("Mortality") +
  xlab("")

outcome_plots = discharge_plot + outcome_plot
outcome_plots
```


Smoking data   
```{r}
fig1_data$active_smoker = factor(fig1_data$`Smoking Status` == "Current Smoker",
                                 levels = c("TRUE", "FALSE"))
fig1_data$`Smoking Status` = factor(fig1_data$`Smoking Status`,
                                    levels = c("Current Smoker",
                                               "Past Smoker",
                                               "Never Smoker",
                                               "Unknown or Not Reported"))
smoker_sum = table(fig1_data$Diagnosis, fig1_data$active_smoker)
smoker_test = pairwise.prop.test(smoker_sum, p.adjust.method = "fdr")
smoker_type_plot =  ggplot(fig1_data, aes(x = Diagnosis, fill = `Smoking Status`)) +
  geom_bar(position = "fill") +
  geom_signif(comparisons = list(c("COVID-19", "Other Viral\nPneumonia"),
                                 c("COVID-19", "Other\nPneumonia"),
                                   c("COVID-19", "Non-Pneumonia\nControl")), 
              annotations = c(format(smoker_test$p.value["Other Viral\nPneumonia", "COVID-19"], digits = 2, scientific = T),
                              format(smoker_test$p.value["Other\nPneumonia", "COVID-19"], digits = 2, scientific = T),
                              format(smoker_test$p.value["COVID-19","Non-Pneumonia\nControl"], digits = 2, scientific = T)),
              y_position = c(1.05, 1.2, 1.35), 
              tip_length = 0,
              aes(y = 1), 
              textsize = 6) +
  ylab("Proportion") +
  guides(fill=guide_legend(nrow=2,byrow=TRUE)) +
  scale_fill_manual(name = "",
                 values = c("Current Smoker" = pal[8],
                            "Never Smoker" = pal[7],
                            "Past Smoker" = pal[5],
                            "Unknown or Not Reported" = pal[6])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom", 
        text = element_text(family = "Arial"), 
        plot.title = element_text(hjust = 0.5)) +
  xlab("") +
  ggtitle("Smoking Status")


pack_year_plot = ggplot(fig1_data, aes(x = Diagnosis, y = `Pack Years`, fill = Diagnosis)) +
  geom_boxplot(notch = F, width = 0.5, outlier.shape = NA) + 
  ylab("Pack-Years") +
  geom_jitter(width = 0.25, size = 0.15) +
  geom_signif(comparisons = list(c("COVID-19", "Other")), 
              test = "t.test", 
              tip_length = 0) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia\nControl" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral\nPneumonia" = pal[3],
                               "Other\nPneumonia" = pal[4])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none", text = element_text(family = "Arial")) +
  xlab("")
smoker_plot = smoker_type_plot + pack_year_plot
smoker_plot
```   
   
SOFA and APS   
```{r}
shapiro.test(fig1_data$SOFA) #very bad, plus it's not a linear scale
sofa_comps = pairwise.wilcox.test(x = fig1_data$SOFA, g = fig1_data$Diagnosis, p.adjust.method = "fdr") #NSD

sofa_plot = ggplot(fig1_data, aes(x = Diagnosis, y = SOFA, fill = Diagnosis)) +
  geom_boxplot(notch = F, width = 0.5, outlier.shape = NA) + 
  ylab("Mean SOFA Score") +
  geom_jitter(width = 0.25, size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia\nControl" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral\nPneumonia" = pal[3],
                               "Other\nPneumonia" = pal[4])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none", text = element_text(family = "Arial")) +
  xlab("") +
  ggtitle("SOFA Score")

sofa_plot

aps_comps = pairwise.wilcox.test(x = fig1_data$APS, g = fig1_data$Diagnosis, p.adjust.method = "fdr") #NSD

aps_plot = ggplot(fig1_data, aes(x = Diagnosis, y = SOFA, fill = Diagnosis)) +
  geom_boxplot(notch = F, width = 0.5, outlier.shape = NA) + 
  ylab("Mean APS Score") +
  geom_jitter(width = 0.25, size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia\nControl" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral\nPneumonia" = pal[3],
                               "Other\nPneumonia" = pal[4])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none", text = element_text(family = "Arial")) +
  xlab("") +
  ggtitle("APS Score")

aps_plot
```


All together (Figure 2)   
```{r}
#unit conversions   
max_width_nature = 7.20472 #inches
max_height_nature = 9.72441

layout = "AABCD\nAAEFG\nAAHIJ"
demographics_plot = (final_timeline + theme(legend.text = element_text(size = 15))) + 
  age_plot + sex_plot + ethnicity_plots + 
  bmi_plot + sofa_plot + aps_plot +
  los_plot + vent_days_plot + outcome_plot + 
  plot_layout(design = layout) +
  plot_annotation(tag_levels = 'a') & theme(plot.tag = element_text(size = 20, family = "Arial"),
                                            plot.title = element_text(size = 15, hjust = 0.5),
                                            strip.text = element_text(size = 15),
                                            axis.text = element_text(angle = 45, hjust = 1, size = 12),
                                            axis.title.x = element_text(size = 20))

CairoPDF("~/Box/RGrant/SCRIPT/manuscript_1/201003_demographics_v4.pdf",
    width = max_width_nature * 3,
    height = 15, family = "Arial")
demographics_plot
dev.off()

demographics_plot
```


## Sample info   
### Number of serial samples   
```{r}
length(serial_patients)
```   
 
### Distribution of days   
All   
```{r}
ggplot(script_bal_data, aes(x = day_of_intubation)) +
  geom_histogram(fill = "dodgerblue4", binwidth = 1)
```

Only flow:   
```{r}
ggplot(script_bal_data, aes(x = day_of_intubation)) +
  geom_histogram(fill = "dodgerblue4", binwidth = 1) +
  ylab("Count") +
  xlab("Day of Ventilation")
```   

### Samples per patient   
Only flow   
```{r}
patient_counts = as.data.frame(table(flow_data$ID))
patient_counts_non_covid = as.data.frame(table(subset(flow_data, covid_confirmed == FALSE)$ID))
patient_counts_covid = as.data.frame(table(subset(flow_data, covid_confirmed == TRUE)$ID))
non_covid_samples = ggplot(patient_counts_non_covid, aes(x = Freq)) +
  geom_histogram(fill = "dodgerblue4", binwidth = 1) +
  ylab("Count") +
  xlab("") +
  scale_x_continuous(n.breaks = 10) +
  ggtitle("COVID-") +
  xlim(0,6)
covid_samples = ggplot(patient_counts_covid, aes(x = Freq)) +
  geom_histogram(fill = "firebrick", binwidth = 1) +
  ylab("Count") +
  xlab("Samples per Patient") +
  scale_x_continuous(n.breaks = 10) +
  ggtitle("COVID+") +
  xlim(0,6)

non_covid_samples / covid_samples
```

## Distributions   
### Percentages   
```{r}
percentage_data = subset(flow_data_long, grepl("percent", flow_data_long$flow_parameter))
shapiro.test(percentage_data$value)
hist(percentage_data$value, breaks = 50)
```   

Extremely non-normal in both cases. At the very least, we need nonparametric tests. In reality, we probably need non-parametric tests or beta regression. However, this may not be true of individual parameters. First let's see if transformations help.   

## Progressions   
### Granular   
```{r}
serial_data = subset(percentage_data, patient_id %in% serial_patients & 
                       flow_parameter %in% c("percent_CD4_total", "percent_CD8_total",
                                             "percent_macs_CD206_high", "percent_macs_CD206_low",
                                             "percent_monocytes", "percent_neutrophils",
                                             "percent_others") &
                       !is.na(day_of_intubation)) %>% 
  unique() %>% 
  arrange(binned_outcome, ID)
observations = table(serial_data$patient_id) / length(unique(serial_data$flow_parameter))
usable_serial_patients = names(observations[observations > 1])
serial_data = subset(serial_data,  patient_id %in% usable_serial_patients & covid_confirmed == TRUE)

ggplot(serial_data, aes(x = day_of_intubation, y = value, fill = flow_parameter)) +
  geom_bar(position = "stack", stat = "identity", width = 1) +
  facet_grid(binned_outcome + ID  ~ ., 
             drop = T, 
             labeller = labeller(binned_outcome = label_value, ID = label_value)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_fill_npg(name = "Flow Parameter",
                     labels = c("percent_CD4_total" = "% CD4 T",
                                "percent_CD8_total" = "% CD8 T",
                                "percent_macs_CD206_high" = "% CD206-high Macrophages",
                                "percent_macs_CD206_low" = "% CD206-low Macrophages",
                                "percent_monocytes" = "% Monocytes",
                                "percent_neutrophils" = "% Neutrophils",
                                "percent_others" = "% Others")) +
  ylab("Percent of Total Cells") +
  theme_bw(base_family = "Arial") +
  theme(text = element_text(family = "Arial"),
        axis.ticks.y = element_blank(),
        axis.text.y = element_blank()) +
  xlab("Day of Intubation")
  
coinfection_data = serial_data %>% 
  dplyr::filter(any_nonviral == TRUE) %>% 
  dplyr::select(day_of_intubation, ID, binned_outcome)
serial_plot_coinfection = ggplot(serial_data, aes(x = day_of_intubation, y = value, fill = flow_parameter)) +
  geom_tile(data = coinfection_data, 
            aes(x = day_of_intubation, y = 50), 
            width = 1.8, height = 100, 
            fill = "firebrick4",
            alpha = 0.1) +
  geom_bar(position = "stack", stat = "identity", width = 0.9, size = 1) +
  facet_grid(binned_outcome + ID  ~ ., 
             drop = T,
             labeller = labeller(binned_outcome = label_value, ID = label_value)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_fill_npg(name = "Flow Parameter",
                     labels = c("percent_CD4_total" = "% CD4 T",
                                "percent_CD8_total" = "% CD8 T",
                                "percent_macs_CD206_high" = "% CD206-high Macrophages",
                                "percent_macs_CD206_low" = "% CD206-low Macrophages",
                                "percent_monocytes" = "% Monocytes",
                                "percent_neutrophils" = "% Neutrophils",
                                "percent_others" = "% Others")) +
  ylab("Percent of Total Cells") +
  theme_bw(base_family = "Arial") +
  theme(text = element_text(family = "Arial"),
        axis.ticks.y = element_blank(),
        axis.text.y = element_blank()) +
  xlab("Day of Intubation") +
  scale_color_manual(name = "",
                       values = c("FALSE" = alpha("white", 0),
                                  "TRUE" = "black"),
                       labels = c("FALSE" = "Primary Only",
                                  "TRUE" = "Superinfection"))
serial_plot_coinfection
```   
   
### Simplified   
```{r}
serial_data_simplified = serial_data %>% 
  pivot_wider(names_from = flow_parameter, values_from = value) %>% 
  dplyr::mutate(percent_macs = percent_macs_CD206_high + percent_macs_CD206_low,
         percent_T_cells = percent_CD4_total + percent_CD8_total) %>% 
  dplyr::select(-c(percent_CD4_total, percent_CD8_total, percent_macs_CD206_high, percent_macs_CD206_low)) %>% 
  pivot_longer(percent_neutrophils:percent_T_cells, names_to = "flow_parameter", values_to = "value") %>% 
  group_by(ID, day_of_intubation) %>% 
  dplyr::mutate(dominant_cell = factor(gsub("percent_", "", flow_parameter[which.max(value)]))) %>% 
  dplyr::filter(!is.na(binned_outcome) & binned_outcome != "Other") %>% 
  dplyr::mutate(coinfection = factor(ifelse(any_nonviral == TRUE,
                                     yes = "Superinfection",
                                     no = "Primary Only")))

covid_serial_simp = subset(serial_data_simplified, covid_confirmed == T)
table(covid_serial_simp$coinfection, covid_serial_simp$dominant_cell)
table(covid_serial_simp$coinfection, covid_serial_simp$neutrophilic)

simplified_progression_plot = ggplot(serial_data_simplified, aes(x = day_of_intubation, y = ID)) +
  geom_line(color = "gray50") +
  facet_grid(binned_outcome ~ ., scales = "free_y", space = "free_y") +
  geom_point(aes(color = dominant_cell, shape = coinfection)) +
  scale_shape_manual(name = "",
                       guide = guide_legend(ncol = 1, byrow = T),
                       values = c("Primary Only" = 16,
                                  "Superinfection" = 10)) +
  scale_color_npg(name = "",
                       labels = c("T_cells" = "T Lymphocytes",
                                  "neutrophils" = "Neutrophils",
                                  "monocytes" = "Monocytes",
                                  "macs" = "Macrophages",
                                  "others" = "Other"),
                  guide = guide_legend(ncol = 2, byrow = T)) +
  theme_bw(base_family = "Arial") +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(family = "Arial"),
        legend.text = element_text(family = "Arial"),
        legend.title = element_text(family = "Arial"),
        legend.position = "bottom") +
  xlab("Day of Intubation") +
  ylab("")
simplified_progression_plot
```


As expected, there appear to be neutrophilic and non-neutrophilic trajectories captured here.   
   
## What are the kinetics of MoAM recruitment and maturation?   
### TRAM levels over time   
```{r}
covid_flow = subset(flow_data, covid_confirmed == TRUE)
ggplot(flow_data, aes(x = day_of_intubation, y = percent_macs_CD206_high, color = pna_type)) +
  facet_wrap(~pna_type, scale = "free_y") +
  geom_point() +
  geom_smooth(se = T)
```   
Might be an uptick at later timepoints as would be expected, but this is very unclear.   
   
### MoAM levels over time   
```{r}
covid_flow = subset(flow_data, covid_confirmed == TRUE)
ggplot(flow_data, aes(x = day_of_intubation, y = percent_macs_CD206_low, color = pna_type)) +
  facet_wrap(~pna_type, scale = "free_y") +
  geom_point() +
  geom_smooth(se = T)
```
   
### Plot contributions (just day 0 for now)   
```{r}
day0_percentages = subset(percentage_data, day_of_intubation <= 2 & day_of_intubation >= 0 &
                            !(flow_parameter %in% c("percent_CD4_total", "percent_CD206_total", "percent_total_CD206_low", "percent_total_CD206_high", "percent_macrophages", "percent_CD3_total")))
ggplot(day0_percentages, aes(x = Sample, y = value, fill = flow_parameter)) +
  geom_bar(position = "stack", stat = "identity") +
  facet_grid(. ~ covid_confirmed, scales = "free_x") +
  theme(axis.text.x = element_blank())
```

# Analysis   
## Associations with neutrophilia   
### Predictive value for COVID-19   
```{r}
initial_coinfection = flow_data %>% 
  dplyr::filter(covid_confirmed == T & day_of_intubation <= 2 & day_of_intubation >= 0 & any_nonviral == T) %>% 
  dplyr::select(ID) %>% 
  unique()
initial_coinfection = as.character(initial_coinfection$ID)
covid_prediction_neutrophilia = flow_data %>% 
  dplyr::filter(covid_confirmed == T & !(ID %in% initial_coinfection)) %>% 
  dplyr::select(Sample, ID, any_nonviral, neutrophilic) %>% 
  dplyr::mutate(prediction = factor(ifelse(neutrophilic == "Neutrophilic", yes = "TRUE", no = "FALSE"))) %>% 
  dplyr::mutate(any_nonviral = factor(any_nonviral))
ppv(data = covid_prediction_neutrophilia, truth = any_nonviral, estimate = prediction)
npv(data = covid_prediction_neutrophilia, truth = any_nonviral, estimate = prediction)
```

## Visualize   
Area plot   
```{r}
area_data = percentage_data %>% 
  group_by(covid_confirmed, day_of_intubation, flow_parameter) %>% 
  dplyr::summarize(mean_pct = mean(value))
area_data = subset(area_data, !(flow_parameter %in% c("percent_CD4_total", "percent_CD206_total", "percent_total_CD206_low", "percent_total_CD206_high")))
# ggplot(area_data, aes(x = day_of_intubation, y = mean_pct, fill = flow_parameter)) +
#   geom_area(alpha = 0.7) +
#   facet_grid(covid_confirmed ~ .)
```
This give a false impression between samples. A line plot may actually be better.   

### Line plot   
Grouped   
```{r eval=FALSE}
ggplot(percentage_data, aes(x = day_of_intubation, y = value, color = flow_parameter)) +
  stat_summary(fun = mean, geom = "line") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5, alpha = 0.5) +
  facet_grid(binned_outcome ~ covid_confirmed)
```

By patient   
```{r eval=FALSE}
ggplot(percentage_data, aes(x = day_of_intubation, y = value, color = flow_parameter)) +
  geom_line() +
  facet_wrap(~ ID)
```
Not enough data at this point to glean much of anything.   

## Heatmaps of flow parameters   
### D0     
All parameters (for internal use)   
```{r}
#cast into matrix of patient x flow parameter
all_day0 = flow_data %>% 
  dplyr::filter(as.numeric(ID) >= 1211) %>%  #select only new panel
  dplyr::filter(day_of_intubation <= 2 & day_of_intubation >= 0) %>% 
  dplyr::filter(!duplicated(ID)) %>% #keep only earliest sample
  select(ID, percent_CD4_total:percent_monocytes, binned_outcome, covid_confirmed)
annotation_data = all_day0 %>% 
  select(ID, binned_outcome, covid_confirmed) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "ID")

matrix_data = all_day0 %>% 
  select(ID, percent_CD4_total:percent_monocytes) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "ID") %>% 
  as.matrix() %>% 
  t()

pheatmap(mat = matrix_data, 
         cluster_rows = T,
         show_colnames = F,
         cluster_cols = T,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         angle_col = 45,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101))
```   
 
Limited (for manuscript)   
```{r}
#cast into matrix of patient x flow parameter
all_day0 = flow_data %>% 
  dplyr::filter(day_of_intubation <= 2 & day_of_intubation >= 0) %>% 
  dplyr::filter(!duplicated(ID)) %>% #keep only earliest sample
  select(ID, percent_CD4_total, percent_CD8_total, percent_macs_CD206_high, percent_macs_CD206_low, 
         percent_monocytes, percent_neutrophils, Diagnosis = pna_type, super_type)

annotation_data = all_day0 %>% 
  select(ID, Diagnosis, super_type) %>% 
  remove_rownames() %>% 
  dplyr::mutate(Diagnosis = factor(as.character(Diagnosis))) %>% 
  column_to_rownames(var = "ID") 

matrix_data = all_day0 %>% 
  select(ID, percent_CD4_total:percent_neutrophils) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "ID") %>% 
  as.matrix() %>% 
  t()

rownames(matrix_data) = c("CD4 T", "CD8 T", "CD206-high AM", "CD206-low AM",
                          "Monocytes", "Neutrophils")

fig2_heatmap = pheatmap(mat = matrix_data, 
         cluster_rows = T,
         show_colnames = F,
         cluster_cols = T,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         angle_col = 45,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101),
         annotation_names_col = F,
         annotation_colors = list(Diagnosis = c("Healthy Control" = pal[6],
                                                "Non-Pneumonia Control" = pal[2],
                                                "COVID-19" = pal[1],
                                                "Other Viral Pneumonia" = pal[3],
                                                "Other Pneumonia" = pal[4]),
                                  super_type = c("Co-Infection" = pal[9],
                                                 "Primary Only" = pal[7])),
         fontsize = 15,
         border_color = NA)

fig2_heatmap_nolegend = pheatmap(mat = matrix_data, 
         cluster_rows = T,
         show_colnames = F,
         cluster_cols = T,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         angle_col = 45,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101),
         annotation_names_col = F,
         annotation_colors = list(Diagnosis = c("Healthy Control" = pal[6],
                                                "Non-Pneumonia Control" = pal[2],
                                                "COVID-19" = pal[1],
                                                "Other Viral Pneumonia" = pal[3],
                                                "Other Pneumonia" = pal[4]),
                                  super_type = c("Co-Infection" = pal[9],
                                                 "Primary Only" = pal[7])),
         legend = F,
         annotation_legend = T, 
         fontsize = 12,
         border_color = NA)
```   

### All days   
```{r}
all_days = flow_data %>% 
  dplyr::filter(!is.na(day_of_intubation) & !is.na(pna_type)) %>% 
  select(Sample, percent_CD4_total, percent_CD8_total, percent_macs_CD206_high, percent_macs_CD206_low, 
         percent_monocytes, percent_neutrophils, Diagnosis = pna_type, finite_day_of_intubation,
         any_nonviral) %>% 
  arrange(Diagnosis, finite_day_of_intubation) %>% 
  unique()
annotation_data = all_days %>% 
  select(Sample, Diagnosis, finite_day_of_intubation, any_nonviral) %>% 
  dplyr::mutate(Diagnosis = factor(Diagnosis), 
         Superinfection = factor(ifelse(any_nonviral == TRUE,
                                        yes = "Superinfection",
                                        no = "Primary Only"))) %>% 
  dplyr::select(-any_nonviral) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "Sample")

matrix_data = all_days %>% 
  select(Sample, percent_CD4_total, percent_CD8_total, percent_macs_CD206_high, percent_macs_CD206_low, 
         percent_neutrophils, percent_monocytes) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "Sample") %>% 
  as.matrix() %>% 
  t()

rownames(matrix_data) = c("CD4T", "CD8 T", "CD206-high Macrophages", "CD206-low Macrophages",
                          "Neutrophils", "Monocytes")

intubation_pal = colorRampPalette(brewer.pal(n = 9, name = "Purples")[3:9])(max(flow_data$finite_day_of_intubation, na.rm = T) + 1)

all_clustered = pheatmap(mat = matrix_data, 
         cluster_rows = T,
         cluster_cols = T,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101),
         annotation_colors = list(Diagnosis = c("Non-Pneumonia Control" = pal[2],
                                                "COVID-19" = pal[1],
                                                "Other Viral Pneumonia" = pal[3],
                                                "Other Pneumonia" = pal[4]),
                                  finite_day_of_intubation = intubation_pal,
                                  Superinfection = c("Superinfection" = pal[9],
                                                     "Primary Only" = pal[7])))
all_clustered_nolegend = pheatmap(mat = matrix_data, 
         cluster_rows = T,
         cluster_cols = T,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101),
         annotation_colors = list(Diagnosis = c("Non-Pneumonia Control" = pal[2],
                                                "COVID-19" = pal[1],
                                                "Other Viral Pneumonia" = pal[3],
                                                "Other Pneumonia" = pal[4]),
                                  finite_day_of_intubation = intubation_pal,
                                  Superinfection = c("Superinfection" = pal[9],
                                                     "Primary Only" = pal[7])),
         legend = F,
         annotation_legend = T,
         fontsize = 12)

all_unclustered = pheatmap(mat = matrix_data, 
         cluster_rows = T,
         cluster_cols = F,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101),
         annotation_colors = list(Diagnosis = c("Non-Pneumonia Control" = pal[2],
                                                "COVID-19" = pal[1],
                                                "Other Viral Pneumonia" = pal[3],
                                                "Other Pneumonia" = pal[4]),
                                  finite_day_of_intubation = intubation_pal,
                                  Superinfection = c("Superinfection" = pal[9],
                                                     "Primary Only" = pal[7])))
all_unclustered_nolegend = pheatmap(mat = matrix_data, 
         cluster_rows = T,
         cluster_cols = F,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101),
         annotation_colors = list(Diagnosis = c("Non-Pneumonia Control" = pal[2],
                                                "COVID-19" = pal[1],
                                                "Other Viral Pneumonia" = pal[3],
                                                "Other Pneumonia" = pal[4]),
                                  finite_day_of_intubation = intubation_pal,
                                  Superinfection = c("Superinfection" = pal[9],
                                                     "Primary Only" = pal[7])),
         legend = F,
         annotation_legend = T,
         fontsize = 12)
```

### Only serial   
```{r}
serial_patients = unique(flow_data$patient_id[duplicated(flow_data$patient_id)])
patient_pal = sample(colorRampPalette(pal)(length(unique(flow_data$ID))))
names(patient_pal) = unique(flow_data$ID)

all_days_serial = flow_data %>% 
  dplyr::filter(!is.na(day_of_intubation) & ID %in% serial_patients) %>% 
  select(ID, percent_CD4_total, percent_CD8_total, percent_macs_CD206_high, percent_macs_CD206_low, 
         percent_neutrophils, percent_monocytes, binned_outcome, finite_day_of_intubation, 
         Diagnosis = pna_type, Superinfection, Sample) %>% 
  arrange(Diagnosis, ID, finite_day_of_intubation)

annotation_data = all_days_serial %>% 
  select(Sample, finite_day_of_intubation, Superinfection, Diagnosis, ID) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "Sample")

matrix_data = all_days_serial %>% 
  select(Sample, percent_CD4_total, percent_CD8_total, percent_macs_CD206_high, percent_macs_CD206_low, 
         percent_neutrophils, percent_monocytes) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "Sample") %>% 
  as.matrix() %>% 
  t()

rownames(matrix_data) = c("CD4T", "CD8 T", "CD206-high Macrophages", "CD206-low Macrophages",
                          "Neutrophils", "Monocytes")

serial_clustered = pheatmap(mat = matrix_data, 
         cluster_rows = T,
         cluster_cols = T,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101),
         annotation_colors = list(Diagnosis = c("Healthy Control" = pal[6],
                                                "Non-Pneumonia Control" = pal[2],
                                                "COVID-19" = pal[1],
                                                "Other Viral Pneumonia" = pal[3],
                                                "Other Pneumonia" = pal[4]),
                                  finite_day_of_intubation = intubation_pal,
                                  ID = patient_pal,
                                  Superinfection = c("Superinfection" = pal[9],
                                                     "Primary Only" = pal[7])))
serial_unclustered = pheatmap(mat = matrix_data, 
         cluster_rows = T,
         cluster_cols = F,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101),
         annotation_colors = list(Diagnosis = c("Non-Pneumonia Control" = pal[2],
                                                "COVID-19" = pal[1],
                                                "Other Viral Pneumonia" = pal[3],
                                                "Other Pneumonia" = pal[4]),
                                  finite_day_of_intubation = intubation_pal,
                                  ID = patient_pal,
                                  Superinfection = c("Superinfection" = pal[9],
                                                     "Primary Only" = pal[7])))
```   
   
## Render figure 2   
### Cell type percentages   
```{r}

fig2_data = flow_data_long %>%
  dplyr::filter(!is.na(pna_type))  %>% 
  dplyr::filter(flow_parameter %in% c("percent_CD4_total", "percent_CD8_total", "percent_neutrophils", 
                                      "percent_macs_CD206_high", "percent_macs_CD206_low", 
                                      "percent_monocytes"))
celltype_names = data.frame(flow_parameter = unique(fig2_data$flow_parameter),
                            cell_type = factor(c("CD4 T", "CD8 T", "Neutrophils",
                                                 "CD206hi Macrophages", "CD206lo Macrophages",
                                                 "Monocytes"),
                                               levels = c("Neutrophils", sort(c("CD4 T", 
                                                                                      "CD8 T",
                                                                                      "CD206hi Macrophages",
                                                                                      "CD206lo Macrophages",
                                                                                      "Monocytes")))))
fig2_data = left_join(fig2_data, celltype_names) 
                                          
  
fig2_data$Diagnosis = str_wrap(fig2_data$pna_type, width = 14)
fig2_data$Diagnosis = gsub("- ", "-", fig2_data$Diagnosis)
fig2_data$Diagnosis = factor(fig2_data$Diagnosis,
                             levels = c("Non-Pneumonia\nControl","COVID-19", 
                                        "Other Viral\nPneumonia","Other\nPneumonia"))
fig2_data = fig2_data %>% 
  dplyr::mutate(combined_factor = factor(paste(Diagnosis, Superinfection, sep = "_")))

fig2_data_0 = dplyr::filter(fig2_data, day_of_intubation <= 2 & day_of_intubation >= 0)
```   
   
Evertything at once :')   
Early vs late   
```{r}
fig2_data_complete = fig2_data %>% 
  dplyr::filter(cell_type %in% c("CD4 T", "CD8 T", "Neutrophils")) %>% 
  mutate(Sampling = factor(case_when((day_of_intubation <= 2 & day_of_intubation >= 0) ~ "Early",
                              day_of_intubation > 2 ~ "Late",
                              is.na(day_of_intubation) ~ NA_character_))) %>% #not relevant for these conditions, compare the whole
  dplyr::filter(!is.na(Sampling)) %>% 
  mutate(super_factor = factor(paste(Diagnosis, Sampling, sep = "_")))

#only make relevant comparisons (super vs non within group, same Sampling status across groups)
relevant_comps = combn(levels(fig2_data_complete$super_factor), 2) %>% 
  t() %>% 
  as.data.frame() %>% 
  rowwise() %>% 
  mutate(first_group = strsplit(V1, split = "_")[[1]][1],
         first_super = strsplit(V1, split = "_")[[1]][2],
         second_group = strsplit(V2, split = "_")[[1]][1],
         second_super = strsplit(V2, split = "_")[[1]][2]) %>% 
  ungroup() %>% 
  dplyr::filter(first_group == second_group | first_super == second_super)
all_comps = lapply(unique(fig2_data_complete$cell_type), function(ct){
  p_vals = vector(mode = "numeric", length = nrow(relevant_comps))
  for(i in 1:nrow(relevant_comps))
  {
    sub = fig2_data_complete %>% 
      dplyr::filter((super_factor == relevant_comps$V1[i] |
                      super_factor == relevant_comps$V2[i]) &
                    cell_type == ct)
    comp = pairwise.wilcox.test(sub$value, 
                                g = sub$super_factor,
                                p.adjust.method = "none") #adjust later
    p_vals[i] = comp$p.value[1,1]
  }
  relevant_comps = relevant_comps %>%
    mutate(pval = p_vals,
           cell_type = ct) })
all_comps = bind_rows(all_comps) %>% 
   mutate(padj = p.adjust(pval, method = "fdr")) %>%
    dplyr::filter(padj < 0.05) %>%
    mutate(annot = format(padj, digits = 2, scientific = T))
start_table = data.frame(first_group = rep(levels(fig2_data_complete$Diagnosis), 2),
                         first_super = c(rep("Early", 4), rep("Late", 4)),
                         start = c(c(1:4) - 0.2, c(1:4) + 0.2))
end_table = data.frame(second_group = rep(levels(fig2_data_complete$Diagnosis), 2),
                         second_super = c(rep("Early", 4), rep("Late", 4)),
                         end = c(c(1:4) - 0.2, c(1:4) + 0.2)) 
all_comps = all_comps %>% 
  left_join(., start_table) %>% 
  left_join(., end_table) %>% 
  mutate(arbitrary_group = rownames(.),
         y = c(35, 60, 50, 42.5, 40,
               50, 60, 50, 105,
               115))
                         



combined_plot = ggplot(fig2_data_complete, aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  geom_boxplot(width = 0.5, outlier.shape = NA, aes(alpha = Sampling)) +
  ylab("Percent of Total Cells") +
  xlab("Diagnosis") +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia\nControl" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral\nPneumonia" = pal[3],
                               "Other\nPneumonia" = pal[4])) +
  scale_alpha_manual(name = "",
                     values = c("Late" = 1,
                                "Early" = 0.7)) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        plot.title = element_text(hjust = 0.5), 
        strip.background = element_blank(),
        strip.text = element_text(size = 15),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 10)) +
  geom_jitter(width = 0.25, size = 0.15) +
  guides(alpha=guide_legend(override.aes=list(fill=hcl(c(15,195),100,0,alpha=c(1,0.7)),
                                              colour=NA),
                            nrow = 2)) +
  geom_signif(inherit.aes = F, 
              family = "Arial",
              data = all_comps,
              aes(Group = cell_type, xmin = start, xmax = end, 
                  annotations = annot, y_position = y,
                  group = arbitrary_group),
              tip_length = 0,
              manual=TRUE) +
  facet_wrap(~cell_type, scales = "free_y", ncol = 3) +
  scale_y_continuous(expand = expansion(add = 10), breaks = seq(0, 100, by = 20)) +
  xlab("")

combined_plot
```
   
Early vs late, with and without super   
```{r}
fig2_data_covid = fig2_data_complete %>% 
  dplyr::filter(Diagnosis == "COVID-19") %>% 
  dplyr::filter(!is.na(Superinfection)) %>% 
  mutate(super_factor = factor(paste(Sampling, Superinfection, sep = "_")),
         cell_type = as.character(cell_type))

relevant_comps = combn(levels(fig2_data_covid$super_factor), 2) %>% 
  t() %>% 
  as.data.frame() %>% 
  rowwise() %>% 
  mutate(first_group = strsplit(V1, split = "_")[[1]][1],
         first_super = strsplit(V1, split = "_")[[1]][2],
         second_group = strsplit(V2, split = "_")[[1]][1],
         second_super = strsplit(V2, split = "_")[[1]][2]) %>% 
  ungroup() %>% 
  dplyr::filter(first_group == second_group)
all_comps = lapply(unique(fig2_data_covid$cell_type), function(ct){
  p_vals = vector(mode = "numeric", length = nrow(relevant_comps))
  for(i in 1:nrow(relevant_comps))
  {
    sub = fig2_data_covid %>% 
      dplyr::filter((super_factor == relevant_comps$V1[i] |
                      super_factor == relevant_comps$V2[i]) &
                    cell_type == ct)
    comp = pairwise.wilcox.test(sub$value, 
                                g = sub$super_factor,
                                p.adjust.method = "none") #adjust later
    p_vals[i] = comp$p.value[1,1]
  }
  relevant_comps = relevant_comps %>%
    mutate(pval = p_vals,
           cell_type = ct) })
all_comps = bind_rows(all_comps) %>% 
   mutate(padj = p.adjust(pval, method = "fdr")) %>%
    dplyr::filter(padj < 0.05) %>%
    mutate(annot = format(padj, digits = 2, scientific = T))
# relevant_comps = relevant_comps %>%
#     mutate(pval = p_vals,
#            cell_type = ct)
start_table = data.frame(first_group = rep(levels(fig2_data_complete$Sampling), 2),
                         first_super = c(rep("Primary Only", 4), rep("Superinfection", 4)),
                         start = c(c(1:4) - 0.2, c(1:4) + 0.2))
end_table = data.frame(second_group = rep(levels(fig2_data_complete$Sampling), 2),
                         second_super = c(rep("Primary Only", 4), rep("Superinfection", 4)),
                         end = c(c(1:4) - 0.2, c(1:4) + 0.2)) 
all_comps = all_comps %>% 
  left_join(., start_table) %>% 
  left_join(., end_table) #NSD 

covid_super_early_late = ggplot(fig2_data_covid, aes(x = Sampling, y = value, fill = Diagnosis)) +
  geom_boxplot(width = 0.5, outlier.shape = NA, aes(alpha = Superinfection)) +
  ylab("Percent of Total Cells") +
  xlab("Diagnosis") +
  scale_fill_manual(name = "",
                    values = c("COVID-19" = pal[1])) +
  scale_alpha_manual(name = "",
                     values = c("Primary Only" = 1,
                                "Superinfection" = 0.5)) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        plot.title = element_text(hjust = 0.5), 
        strip.background = element_blank(),
        strip.text = element_text(size = 15),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 10)) +
  geom_jitter(width = 0.25, size = 0.15) +
  guides(alpha=guide_legend(override.aes=list(fill=hcl(c(15,195),100,0,alpha=c(1,0.7)),
                                              colour=NA),
                            nrow = 2)) +
  # geom_signif(inherit.aes = F, 
  #             family = "Arial",
  #             data = all_comps,
  #             aes(Group = cell_type, xmin = start, xmax = end, 
  #                 annotations = annot, y_position = y,
  #                 group = arbitrary_group),
  #             tip_length = 0,
  #             manual=TRUE) +
  facet_wrap(~cell_type, scales = "free_y", ncol = 3) +
  scale_y_continuous(expand = expansion(add = 10), breaks = seq(0, 100, by = 20)) +
  xlab("")

covid_super_early_late
```
   
Evertything at once, D0   
Note: can't split by co-infection due to low N   
```{r}
all_comps = lapply(unique(fig2_data_0$cell_type), function(x){
  sub = subset(fig2_data_0, cell_type == x)
  #we will adjust all p-vals together later
  tests = pairwise.wilcox.test(x = sub$value, g = sub$Diagnosis, p.adjust.method = "none")$p.value %>%
    as.data.frame() %>% 
    rownames_to_column("comp1") %>% 
    pivot_longer(cols = 2:ncol(.), names_to = "comp2", values_to = "pval") %>% 
    dplyr::filter(!is.na(pval)) %>% 
    dplyr::mutate(cell_type = x)
  return(tests) })
all_comps = bind_rows(all_comps) %>% 
  dplyr::mutate(padj = p.adjust(pval, method = "fdr")) %>% 
  dplyr::filter(padj < 0.05)

all_comps = all_comps %>% 
  dplyr::mutate(annot = ifelse(padj < 0.01,
                        no = "*",
                        yes = ifelse(padj < 0.001,
                                     no = "**",
                                     yes = "***"))) %>%
  dplyr::mutate(numeric_annot = format(padj, digits = 2, scientific = T)) %>%
  dplyr::mutate(arbitrary_group = rownames(.)) %>%
  dplyr::mutate(y = c(52, 62, 45, 55, 110,
                      105, 120, 65, 75)) #had to just do this manually

combined_plot_D0 = ggplot(fig2_data_0, aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  geom_boxplot(width = 0.5, outlier.shape = NA) +
  ylab("Percent of Total Cells") +
  xlab("Diagnosis") +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia\nControl" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral\nPneumonia" = pal[3],
                               "Other\nPneumonia" = pal[4])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        plot.title = element_text(hjust = 0.5), 
        strip.background = element_blank(),
        strip.text = element_text(size = 15), 
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 10)) +
  geom_jitter(width = 0.25, size = 0.15) +
  guides(alpha=guide_legend(override.aes=list(fill=hcl(c(15,195),100,0,alpha=c(0.2,0.7)),
                                              colour=NA))) +
  geom_signif(inherit.aes = F,
              family = "Arial",
              data = all_comps,
              aes(Group = cell_type, xmin = comp1, xmax = comp2,
                  annotations = numeric_annot, y_position = y,
                  group = arbitrary_group),
              tip_length = 0,
              manual=TRUE) +
  facet_wrap(~cell_type, scales = "free_y", ncol = 2) +
  scale_y_continuous(expand = expansion(c(0,0.2)))

combined_plot_D0
```
   
General t cells   
```{r}
fig2_tcells = fig2_data %>% 
  dplyr::select(-cell_type) %>% 
  dplyr::filter(flow_parameter %in% c("percent_CD8_total", "percent_CD4_total")) %>% 
  pivot_wider(names_from = flow_parameter, values_from = value) %>% 
  dplyr::mutate(percent_t = percent_CD8_total + percent_CD4_total) %>% 
  dplyr::mutate(lymphocytic = percent_t > 20)

tcell_sum = fig2_tcells %>% 
  group_by(pna_type) %>% 
  dplyr::summarize(mean_tcells = CI(percent_t)["mean"], 
                   lower_ci_tcells = CI(percent_t)["lower"],
                   upper_ci_tcells = CI(percent_t)["upper"],
                   .groups = "keep")
tcell_sum

tcell_sum = fig2_tcells %>% 
  dplyr::filter(day_of_intubation <= 2 & day_of_intubation >= 0) %>% 
  group_by(pna_type) %>% 
  dplyr::summarize(mean_tcells = CI(percent_t)["mean"], 
                   lower_ci_tcells = CI(percent_t)["lower"],
                   upper_ci_tcells = CI(percent_t)["upper"],
                   .groups = "keep")
tcell_sum


table(fig2_tcells$Diagnosis, fig2_tcells$lymphocytic)
```
   
T cells and outcome   
```{r}
fig2_tcells = fig2_data %>% 
  dplyr::filter(flow_parameter %in% c("percent_CD8_total", "percent_CD4_total") &
                  Diagnosis == "COVID-19")

all_comps = lapply(unique(fig2_tcells$cell_type), function(x){
  sub = subset(fig2_tcells, cell_type == x)
  #we will adjust all p-vals together later
  tests = pairwise.wilcox.test(x = sub$value, g = sub$binned_outcome, p.adjust.method = "none")$p.value %>%
    as.data.frame() %>% 
    rownames_to_column("comp1") %>% 
    pivot_longer(cols = 2:ncol(.), names_to = "comp2", values_to = "pval") %>% 
    dplyr::filter(!is.na(pval)) %>% 
    dplyr::mutate(cell_type = x)
  return(tests) })
all_comps = bind_rows(all_comps) %>% 
  dplyr::mutate(padj = p.adjust(pval, method = "fdr")) %>% 
  dplyr::filter(padj < 0.05) %>% 
  mutate(y = 20)
all_comps = all_comps %>% 
  dplyr::mutate(annot = ifelse(padj < 0.01,
                        no = "*",
                        yes = ifelse(padj < 0.001,
                                     no = "**",
                                     yes = "***"))) %>%
  dplyr::mutate(numeric_annot = format(padj, digits = 2, scientific = T)) %>%
  dplyr::mutate(arbitrary_group = rownames(.)) %>%
  dplyr::mutate(y = c(60, 70, 40))


ggplot(fig2_tcells, aes(x = binned_outcome, y = value, fill = binned_outcome)) +
  geom_boxplot(width = 0.5, outlier.shape = NA) +
  ylab("Percent of Total Cells") +
  xlab("Diagnosis") +
  scale_fill_npg() +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        plot.title = element_text(hjust = 0.5), 
        strip.background = element_blank(),
        strip.text = element_text(size = 15), 
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 10)) +
  geom_jitter(width = 0.25, size = 0.15) +
  guides(alpha=guide_legend(override.aes=list(fill=hcl(c(15,195),100,0,alpha=c(0.2,0.7)),
                                              colour=NA))) +
   geom_signif(inherit.aes = F,
              family = "Arial",
              data = all_comps,
              aes(Group = cell_type, xmin = comp1, xmax = comp2,
                  annotations = numeric_annot, y_position = y,
                  group = arbitrary_group),
              tip_length = 0,
              manual=TRUE) +
  facet_wrap(~cell_type, scales = "free_y", ncol = 3) +
  scale_y_continuous(expand = expansion(add = 20))

#just early
fig2_tcells = fig2_data %>% 
  dplyr::filter(flow_parameter %in% c("percent_CD8_total", "percent_CD4_total") &
                  Diagnosis == "COVID-19" & true_early == T)

ggplot(fig2_tcells, aes(x = binned_outcome, y = value, fill = binned_outcome)) +
  geom_boxplot(width = 0.5, outlier.shape = NA) +
  ylab("Percent of Total Cells") +
  xlab("Diagnosis") +
  scale_fill_npg() +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        plot.title = element_text(hjust = 0.5), 
        strip.background = element_blank(),
        strip.text = element_text(size = 15), 
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 10)) +
  geom_jitter(width = 0.25, size = 0.15) +
  guides(alpha=guide_legend(override.aes=list(fill=hcl(c(15,195),100,0,alpha=c(0.2,0.7)),
                                              colour=NA))) +
  facet_wrap(~cell_type, scales = "free_y", ncol = 3) +
  scale_y_continuous(expand = expansion(add = 20))
```   
Overall, seems like recruitment of T cells leads to positive outcomes, but only if this is either short-lived or occurs early in infection. Fits well with the current model.   
   
Superinfection and neutrophils (now + T cells)      
```{r}
fig2_neutrophils_super = flow_data_long %>% 
  dplyr::filter(!is.na(pna_type) & 
                  flow_parameter == "percent_neutrophils" &
                  !is.na(super_type))
fig2_neutrophils_super$Diagnosis = str_wrap(fig2_neutrophils_super$pna_type, width = 14)
fig2_neutrophils_super$Diagnosis = gsub("- ", "-", fig2_neutrophils_super$Diagnosis)
fig2_neutrophils_super$Diagnosis = factor(fig2_neutrophils_super$Diagnosis,
                             levels = c("Non-Pneumonia\nControl","COVID-19", 
                                        "Other Viral\nPneumonia", "Other\nPneumonia"))
fig2_neutrophils_super = fig2_neutrophils_super %>% 
  dplyr::mutate(super_factor = factor(paste(Diagnosis, Superinfection, sep = "_")))

#only make relevant comparisons (super vs non within group, same superinfection status across groups)
relevant_comps = combn(levels(fig2_neutrophils_super$super_factor), 2) %>% 
  t() %>% 
  as.data.frame() %>% 
  rowwise() %>% 
  mutate(first_group = strsplit(V1, split = "_")[[1]][1],
         first_super = strsplit(V1, split = "_")[[1]][2],
         second_group = strsplit(V2, split = "_")[[1]][1],
         second_super = strsplit(V2, split = "_")[[1]][2]) %>% 
  ungroup() %>% 
  dplyr::filter(first_group == second_group | first_super == second_super)
p_vals = vector(mode = "numeric", length = nrow(relevant_comps))
for(i in 1:nrow(relevant_comps))
{
  sub = fig2_neutrophils_super %>% 
    dplyr::filter(super_factor == relevant_comps$V1[i] |
                    super_factor == relevant_comps$V2[i])
  comp = pairwise.wilcox.test(sub$value, 
                              g = sub$super_factor,
                              p.adjust.method = "none") #adjust later
  p_vals[i] = comp$p.value[1,1]
}
# relevant_comps = relevant_comps %>% 
#   mutate(pval = p_vals,
#          padj = p.adjust(pval, method = "fdr")) %>% 
#   dplyr::filter(padj < 0.05) %>% 
#   mutate(annot = format(padj, digits = 2, scientific = T),
#          y = c(115, 105),
#          xmin = c(1.9, 3.9),
#          xmax = c(4.1, 4.1))

coinfection_neutrophil_plot = ggplot(fig2_neutrophils_super, aes(x = Diagnosis, y = value, fill = Superinfection)) +
  geom_boxplot(width = 0.5, outlier.shape = NA) +
  ylab("Percent of Total Cells") +
  xlab("Diagnosis") +
  scale_fill_manual(name = "",
                 values = c("Superinfection" = pal[9],
                            "Primary Only" = pal[7])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        plot.title = element_text(hjust = 0.5), 
        strip.background = element_blank(),
        strip.text = element_text(size = 15), 
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 10)) +
  geom_jitter(width = 0.25, size = 0.15) +
  # geom_signif(inherit.aes = F, 
  #             data = relevant_comps,
  #             aes(xmin = xmin, xmax = xmax, annotations = annot, y_position = y),
  #             tip_length = 0,
  #             manual=TRUE) +
  # scale_y_continuous(breaks = seq(from = 0, to = 100, by = 25),
  #                    expand = expansion(c(0,0.1))) +
  ggtitle("Neutrophils and Superinfection")
coinfection_neutrophil_plot

co_infection_table = table(fig2_neutrophils_super$Diagnosis, fig2_neutrophils_super$Superinfection)
co_infection_comps = pairwise.prop.test(co_infection_table[1:3, ], p.adjust.method = "fdr")
infection_data = fig2_neutrophils_super %>% 
  dplyr::filter(!is.na(Superinfection))
infected_proportion_plot = ggplot(infection_data, aes(x = Diagnosis, fill = Superinfection)) +
  geom_bar(position = "fill", width = 0.7) + 
  geom_signif(comparisons = list(c("Other Viral\nPneumonia", "COVID-19"),
                                 c("COVID-19", "Non-Pneumonia\nControl")),
              annotations = c(format(co_infection_comps$p.value["Other Viral\nPneumonia", "COVID-19"],
                                     digits = 2, scientific = T),
                              format(co_infection_comps$p.value["COVID-19", "Non-Pneumonia\nControl"],
                                     digits = 2, scientific = T)),
              y_position = c(1.15, 1.05),
              tip_length = 0,
              aes(y = 1)) +
  ylab("Proportion") +
  xlab("") +
  scale_fill_npg(name = "") +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 10)) +
  scale_y_continuous(breaks = seq(0, 1, 0.25), expand = c(0, 0.2)) +
  ggtitle("Co-Infection Rate")
infected_proportion_plot
```   
   
COVID neutrophils   
```{r}
covid_coinfection_data = flow_data_long %>% 
  dplyr::filter(flow_parameter %in% c("percent_CD4_total", "percent_CD8_total", "percent_neutrophils") &
                  !is.na(super_type)) %>% 
  dplyr::filter(pna_type == "COVID-19") %>% 
  mutate(Superinfection = ifelse(Superinfection == "Primary Only",
                                 yes = "COVID-19 Only",
                                 no = Superinfection))
coinfection_comps = covid_coinfection_data %>% 
  group_by(flow_parameter) %>% 
  dplyr::summarize(comp = first(flow_parameter), 
                   pval = pairwise.wilcox.test(x = value, g = Superinfection, p.adjust.method = "none")$p.value[1,1]) %>% 
  mutate(padj = p.adjust(pval, method = "fdr")) #all just barely NS


covid_neutrophil_plot = ggplot(covid_coinfection_data, aes(x = flow_parameter, y = value, fill = Superinfection)) +
   geom_boxplot(width = 0.5, outlier.shape = NA) +
  ylab("Percent of Total Cells") +
  xlab("Cell Type") +
  scale_fill_manual(name = "",
                 values = c("Superinfection" = pal[9],
                            "COVID-19 Only" = pal[7])) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom",
        plot.title = element_text(hjust = 0.5), 
        strip.background = element_blank(),
        strip.text = element_text(size = 15), 
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 10)) +
  geom_jitter(width = 0.25, size = 0.15) +
  geom_signif(test = "wilcox.test") +
  scale_y_continuous(breaks = seq(from = 0, to = 100, by = 25),
                     expand = expansion(c(0,0.1))) +
  scale_x_discrete(labels = c("percent_CD4_total" = "CD4 T", "percent_CD8_total" = "CD8 T",
                              "percent_neutrophils" = "Neutrophils")) +
  ggtitle("Effect of Superinfection")

covid_neutrophil_plot
```
   
COVID only, after initial   
```{r}
covid_only_on_intake = fig2_neutrophils_super %>% 
  dplyr::select(ID, day_of_intubation, Superinfection, covid_confirmed) %>% 
  dplyr::filter(covid_confirmed == T & day_of_intubation >=0 & day_of_intubation <= 2) %>% 
  dplyr::select(ID)
covid_only_on_intake = as.character(covid_only_on_intake$ID)
fig2_neutrophils_covid = fig2_neutrophils_super %>% 
  dplyr::filter(ID %in% covid_only_on_intake)
super_table_covid = table(fig2_neutrophils_covid$Superinfection, 
                          fig2_neutrophils_covid$neutrophilic, 
                          useNA = "always")
super_table_covid

fig2_neutrophils_covid = fig2_neutrophils_covid %>% 
  group_by(ID) %>% 
  dplyr::mutate(ever_neutrophilic = any(neutrophilic == "Neutrophilic")) %>% 
  dplyr::select(ID, ever_neutrophilic, Superinfection) %>% 
  unique()
table(fig2_neutrophils_covid$Superinfection, fig2_neutrophils_covid$ever_neutrophilic)
```

   
All cell types at D0
```{r}
fig2_data_all = fig2_data_0 %>% 
  dplyr::filter(flow_parameter %in% c("percent_CD4_total", "percent_CD8_total",
                                      "percent_macs_CD206_high", "percent_macs_CD206_low",
                                      "percent_monocytes", "percent_neutrophils",
                                      "percent_others") &
                  !is.na(super_type)) %>% 
  mutate(super_type = factor(super_type,
                             levels = c("Primary Only", "Co-Infection")))
d0_barplots = ggplot(fig2_data_all, aes(x = Diagnosis, fill = flow_parameter)) +
  stat_summary(aes(y = value), fun = "mean", geom = "bar", position = "stack") +
  scale_fill_npg(name = "",
                     labels = c("percent_CD4_total" = "% CD4 T",
                                "percent_CD8_total" = "% CD8 T",
                                "percent_macs_CD206_high" = "% CD206-high Macrophages",
                                "percent_macs_CD206_low" = "% CD206-low Macrophages",
                                "percent_monocytes" = "% Monocytes",
                                "percent_neutrophils" = "% Neutrophils",
                                "percent_others" = "% Others"),
                 guide = guide_legend(ncol = 2, byrow = T)) +
  ylab("Percent of Total Cells") +
  xlab("") +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15),
        legend.text = element_text(size = 10),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15)) +
  ggtitle("Average Composition")

d0_barplots_split = ggplot(fig2_data_all, aes(x = Diagnosis, fill = flow_parameter)) +
  facet_grid(. ~ super_type) +
  stat_summary(aes(y = value), fun = "mean", geom = "bar", position = "stack") +
  scale_fill_npg(name = "",
                     labels = c("percent_CD4_total" = "% CD4 T",
                                "percent_CD8_total" = "% CD8 T",
                                "percent_macs_CD206_high" = "% CD206-high Macrophages",
                                "percent_macs_CD206_low" = "% CD206-low Macrophages",
                                "percent_monocytes" = "% Monocytes",
                                "percent_neutrophils" = "% Neutrophils",
                                "percent_others" = "% Others"),
                 guide = guide_legend(ncol = 3, byrow = T)) +
  ylab("Percent of Total Cells") +
  xlab("") +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15),
        legend.text = element_text(size = 10),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15)) +
  ggtitle("Average Composition")

d0_barplots 
d0_barplots_split
```   
   
All cell types at D0
```{r}
fig2_data_all = fig2_data %>% 
  dplyr::filter(flow_parameter %in% c("percent_CD4_total", "percent_CD8_total",
                                      "percent_macs_CD206_high", "percent_macs_CD206_low",
                                      "percent_monocytes", "percent_neutrophils",
                                      "percent_others"))
all_days_barplots = ggplot(fig2_data_all, aes(x = Diagnosis, fill = flow_parameter)) +
  stat_summary(aes(y = value), fun = "mean", geom = "bar", position = "stack") +
  scale_fill_npg(name = "",
                     labels = c("percent_CD4_total" = "% CD4 T",
                                "percent_CD8_total" = "% CD8 T",
                                "percent_macs_CD206_high" = "% CD206-high Macrophages",
                                "percent_macs_CD206_low" = "% CD206-low Macrophages",
                                "percent_monocytes" = "% Monocytes",
                                "percent_neutrophils" = "% Neutrophils",
                                "percent_others" = "% Others"),
                 guide = guide_legend(ncol = 2, byrow = T)) +
  ylab("Percent of Total Cells") +
  xlab("") +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "bottom", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 10)) +
  ggtitle("Average Composition")

all_days_barplots
```

   
### Join   
Just D0 (currently figure 3)   
```{r}
layout = "AAABB\nAAABB\n###BB"
fig2_heatmap_gg = as.ggplot(fig2_heatmap, scale = 0.98)
flow_plot_fig2 = fig2_heatmap_gg +
  (combined_plot_D0 + xlab("")) +
    plot_layout(design = layout) +
   plot_annotation(tag_levels = 'a') & 
                     theme(plot.tag = element_text(size = 20, family = "Arial"))

CairoPDF("~/Box/RGrant/SCRIPT/manuscript_1/201013_flow_plot_D0.pdf",
    width = max_width_nature * 3,
    height = max_height_nature, 
    family = "Arial")
flow_plot_fig2
dev.off()


fig2_heatmap_nolegend_gg = as.ggplot(fig2_heatmap_nolegend, scale = 0.98)
flow_plot_fig2_nogelend = fig2_heatmap_nolegend_gg +
  (combined_plot_D0 + xlab("")) +
    plot_layout(design = layout) +
   plot_annotation(tag_levels = 'a') & 
                     theme(plot.tag = element_text(size = 20, family = "Arial"))

CairoPDF("~/Box/RGrant/SCRIPT/manuscript_1/201013_flow_plot_nolegend_D0.pdf",
    width = max_width_nature * 3,
    height = max_height_nature, 
    family = "Arial")
flow_plot_fig2_nogelend
dev.off()

flow_plot_fig2_nogelend
```
      
## Compare individual parameters by group   
### Neutrophils, CD206low and CD206high for D0 samples   
```{r}
single_param = flow_data %>% 
  dplyr::select(Sample, ID, day_of_intubation, covid_confirmed, true_early,
                percent_neutrophils, percent_macs_CD206_low, percent_macs_CD206_high, percent_total_CD206_high) %>% 
  dplyr::filter(day_of_intubation <= 2 & 
                  day_of_intubation <= 2 & day_of_intubation >= 0 &
                  as.numeric(ID) >= 1211) %>% 
  pivot_longer(cols = percent_neutrophils:percent_total_CD206_high,
               names_to = "flow_parameter",
               values_to = "percent")
ggplot(single_param, aes(x = covid_confirmed, y = percent, fill = covid_confirmed)) +
  geom_boxplot(width = 0.5, outlier.shape = NA) +
  geom_jitter(width = 0.25, size = 0.15) +
  facet_wrap(~ flow_parameter) +
  geom_signif(comparisons = list(c(1, 2)),
              test = "wilcox.test",
              tip_length = 0)
```




# Co-infection   
## Proportion of microbe-infected individuals   
### Plot   
```{r}
script_bal_data = script_bal_data %>%
  rowwise() %>% 
  dplyr::mutate(infection_detected = !is.null(organism_name)) %>% 
  ungroup()
script_bal_data$infection_detected[is.na(script_bal_data$infection_detected)] = "Not Tested"
script_bal_data$infection_detected = factor(script_bal_data$infection_detected,
                                          levels = c("Not Tested", "TRUE", "FALSE"))
ggplot(subset(script_bal_data, !is.na(covid_confirmed)), aes(x = covid_confirmed, fill = infection_detected)) +
  geom_bar(position = "fill") +
  ylab("Proportion") +
  scale_fill_manual(values = c("Not Tested" = "gray", "FALSE" = "cornflowerblue", "TRUE" = "firebrick"))
```

## Levels of infection   
### Histograms of culture quantities   
```{r}
covid_organism_levels = script_bal_data %>% 
  dplyr::filter(covid_confirmed == T) %>% 
  select(organism_quantity) %>% 
  unlist()
other_organism_levels = script_bal_data %>% 
  dplyr::filter(covid_confirmed == F) %>% 
  select(organism_quantity) %>% 
  unlist()
ggplot(NULL) +
  geom_density(aes(x = covid_organism_levels), fill = "firebrick", alpha = 0.5) +
  geom_density(aes(x = other_organism_levels), fill = "cornflowerblue", alpha = 0.5)
```

## Key biomarkers   
Include: CRP, D-dimer, ferritin, Procalcitonin, LDH, CPK and AST   
   
### Check for normality   
```{r}
shapiro.test(script_bal_data$C_Reactive_Protein) #not normal
hist(script_bal_data$C_Reactive_Protein)
shapiro.test(script_bal_data$D_DIMER) #not normal
hist(script_bal_data$D_DIMER)
shapiro.test(script_bal_data$FERRITIN) #not normal
hist(script_bal_data$FERRITIN)
shapiro.test(script_bal_data$PROCALCITONIN) #not normal
hist(script_bal_data$PROCALCITONIN)
shapiro.test(script_bal_data$LDH) #not normal
hist(script_bal_data$LDH)
shapiro.test(script_bal_data$CREATINE_KINASE_TOTAL) #not normal
hist(script_bal_data$CREATINE_KINASE_TOTAL)
```
   
### D0   
Heatmap   
```{r}
biomarker_d0 = script_bal_data %>% 
  dplyr::filter(day_of_intubation <= 2 & day_of_intubation >= 0) %>% 
  dplyr::select(C_Reactive_Protein, D_DIMER, FERRITIN,
                PROCALCITONIN, LDH, CREATINE_KINASE_TOTAL,
                Diagnosis = pna_type, tc_pt_study_id) 

annotation_data = biomarker_d0 %>% 
  select(tc_pt_study_id, Diagnosis) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "tc_pt_study_id")

matrix_data = biomarker_d0 %>% 
  select(tc_pt_study_id, C_Reactive_Protein:CREATINE_KINASE_TOTAL) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "tc_pt_study_id") %>% 
  as.matrix() %>% 
  t()

rownames(matrix_data) = c("CRP", "D-Dimer", "Ferritin", 
                          "Procalcitonin", "LDH", "CPK")

pheatmap(mat = matrix_data, 
         cluster_rows = F,
         cluster_cols = F,
         clustering_method = "ward.D2",
         annotation_col = annotation_data,
         scale = "row",
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101))
```   
Data are way too sparse. Need to go with boxplots or similar.   
   
Boxplots
```{r}
biomarkers = script_bal_data %>% 
  dplyr::select(CRP = C_Reactive_Protein, `D-Dimer` = D_DIMER, Ferritin = FERRITIN,
                Procalcitonin = PROCALCITONIN, LDH, CPK = CREATINE_KINASE_TOTAL, 
                APS = mean_aps, true_early, Diagnosis = pna_type, tc_pt_study_id, 
                day_of_intubation) %>% 
  pivot_longer(cols = CRP:APS, 
               names_to = "Biomarker",
               values_to = "value") %>% 
  dplyr::filter(!is.na(Diagnosis)) %>% 
  dplyr::mutate(Biomarker = factor(Biomarker, levels = c("APS", "CRP", "D-Dimer",
                                                         "Ferritin", "Procalcitonin",
                                                         "LDH", "CPK")))
biomarker_d0 = biomarkers %>% 
  dplyr::filter(day_of_intubation <= 2 & day_of_intubation >= 0)

comps = lapply(levels(biomarker_d0$Biomarker), function(x){
  sub = subset(biomarker_d0, Biomarker == x)
  comp = pairwise.wilcox.test(x = sub$value, g = sub$Diagnosis, p.adjust.method = "none")$p.value %>%
    as.data.frame() %>% 
    rownames_to_column("comp1") %>% 
    pivot_longer(cols = 2:ncol(.), names_to = "comp2", values_to = "pval") %>% 
    dplyr::filter(!is.na(pval)) %>% 
    dplyr::mutate(Biomarker = x)
  return(comp) })
all_comps = bind_rows(comps) %>% 
  dplyr::mutate(padj = p.adjust(pval, method = "fdr")) %>% 
  dplyr::filter(padj < 0.05) %>% 
  dplyr::mutate(annot = format(padj, digits = 2, scientific = T)) %>% 
  mutate(y = c(500, 1000, 50000, 1800, 900),
         y_log10 = log10(y))
                        

#plot individually for patchwork
aps_plot = biomarker_d0 %>% 
  dplyr::filter(Biomarker == "APS") %>% 
  ggplot(., aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4]),
                    guide = guide_legend(ncol = 2)) +
  theme_bw(base_family = "Arial") +
  xlab("") +
  ylab("") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15)) +
  ggtitle("APS Score") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 14))

crp_plot = biomarker_d0 %>% 
  dplyr::filter(Biomarker == "CRP") %>% 
  ggplot(., aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  ggplot2::annotate(geom = "rect",
                    xmin = -Inf, 
            xmax = Inf, 
            ymin = 0,
            ymax = 10,
            fill = pal[7],
            alpha = 0.4) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4]),
                    guide = guide_legend(ncol = 2)) +
  theme_bw(base_family = "Arial") +
  xlab("") +
  ylab("ng/mL") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15)) +
  ggtitle("C-Reactive Protein") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 14)) +
  geom_signif(inherit.aes = F, 
              data = subset(all_comps, Biomarker == "CRP"),
              aes(xmin = comp1, xmax = comp2, annotations = annot, y_position = y_log10),
              tip_length = 0,
              manual=TRUE, 
              textsize = 6) +
  scale_y_log10()

ddimer_plot = biomarker_d0 %>% 
  dplyr::filter(Biomarker == "Ferritin") %>% 
  ggplot(., aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  ggplot2::annotate(geom = "rect",
                    xmin = -Inf, 
            xmax = Inf, 
            ymin = 0,
            ymax = 229,
            fill = pal[7],
            alpha = 0.4) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4]),
                    guide = guide_legend(ncol = 2)) +
  theme_bw(base_family = "Arial") +
  xlab("") +
  ylab("ng/mL") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15)) +
  ggtitle("D-Dimer") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 14)) +
  geom_signif(inherit.aes = F, 
              data = subset(all_comps, Biomarker == "D-Dimer"),
              aes(xmin = comp1, xmax = comp2, annotations = annot, y_position = y_log10),
              tip_length = 0,
              manual=TRUE, 
              textsize = 6) +
  scale_y_log10()

ferritin_plot = biomarker_d0 %>% 
  dplyr::filter(Biomarker == "D-Dimer") %>% 
  ggplot(., aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  ggplot2::annotate(geom = "rect",
                    xmin = -Inf, 
            xmax = Inf, 
            ymin = 24,
            ymax = 336,
            fill = pal[7],
            alpha = 0.4) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4]),
                    guide = guide_legend(ncol = 2)) +
  theme_bw(base_family = "Arial") +
  xlab("") +
  ylab("ng/mL") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15)) +
  ggtitle("Ferritin") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 14)) +
  geom_signif(inherit.aes = F, 
              data = subset(all_comps, Biomarker == "Ferritin"),
              aes(xmin = comp1, xmax = comp2, annotations = annot, y_position = y_log10),
              tip_length = 0,
              manual=TRUE, 
              textsize = 6) +
  scale_y_log10()

pct_plot = biomarker_d0 %>% 
  dplyr::filter(Biomarker == "Procalcitonin") %>% 
  ggplot(., aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  ggplot2::annotate(geom = "rect",
                    xmin = -Inf, 
            xmax = Inf, 
            ymin = 0,
            ymax = 6.5e-2,
            fill = pal[7],
            alpha = 0.4) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4]),
                    guide = guide_legend(ncol = 2)) +
  theme_bw(base_family = "Arial") +
  xlab("") +
  ylab("ng/mL") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15)) +
  ggtitle("Procalcitonin") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 14)) +
  geom_signif(inherit.aes = F, 
              data = subset(all_comps, Biomarker == "Procalcitonin"),
              aes(xmin = comp1, xmax = comp2, annotations = annot, y_position = y_log10),
              tip_length = 0,
              manual=TRUE, 
              textsize = 6) +
  scale_y_log10()

cpk_plot = biomarker_d0 %>% 
  dplyr::filter(Biomarker == "CPK") %>% 
  ggplot(., aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  ggplot2::annotate(geom = "rect",
           xmin = -Inf, 
            xmax = Inf, 
            ymin = 0,
            ymax = 223,
            fill = pal[7],
            alpha = 0.4) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4]),
                    guide = guide_legend(ncol = 2)) +
  theme_bw(base_family = "Arial") +
  xlab("") +
  ylab("ng/mL") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15)) +
  ggtitle("CPK") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 14)) +
  geom_signif(inherit.aes = F, 
              data = subset(all_comps, Biomarker == "CPK"),
              aes(xmin = comp1, xmax = comp2, annotations = annot, y_position = y_log10),
              tip_length = 0,
              manual=TRUE, 
              textsize = 6) +
  scale_y_log10()

ldh_plot = biomarker_d0 %>% 
  dplyr::filter(Biomarker == "LDH") %>% 
  ggplot(., aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  ggplot2::annotate(geom = "rect",
           xmin = -Inf, 
            xmax = Inf, 
            ymin = 0,
            ymax = 271,
            fill = pal[7],
            alpha = 0.4) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.15) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4]),
                    guide = guide_legend(ncol = 2)) +
  theme_bw(base_family = "Arial") +
  xlab("") +
  ylab("U/L") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 15)) +
  ggtitle("LDH") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 14)) +
  geom_signif(inherit.aes = F, 
              data = subset(all_comps, Biomarker == "LDH"),
              aes(xmin = comp1, xmax = comp2, annotations = annot, y_position = y_log10),
              tip_length = 0,
              manual=TRUE, 
              textsize = 6) +
  scale_y_log10()

biomarker_plot_d0 = ggplot(biomarker_d0, aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  #add "healthy" zones
  geom_rect(data = subset(biomarker_d0, Biomarker == "CRP")[1,], #alpha doesn't work with multitple rows :')
            xmin = -Inf, 
            xmax = Inf, 
            ymin = -Inf,
            ymax = log10(10),
            fill = pal[7],
            alpha = 0.4, 
            inherit.aes = F) +
  geom_rect(data = subset(biomarker_d0, Biomarker == "D-Dimer")[1,],
            xmin = -Inf, 
            xmax = Inf, 
            ymin = -Inf,
            ymax = log10(229),
            fill = pal[7],
            alpha = 0.4, 
            inherit.aes = F) +
    geom_rect(data = subset(biomarker_d0, Biomarker == "Procalcitonin")[1,], 
            xmin = -Inf, 
            xmax = Inf, 
            ymin = -Inf,
            ymax = log10(6.5e-2),
            fill = pal[7],
            alpha = 0.4, 
            inherit.aes = F) +
  geom_rect(data = subset(biomarker_d0, Biomarker == "Ferritin")[1,], 
            xmin = -Inf, 
            xmax = Inf, 
            ymin = log10(24),
            ymax = log10(336),
            fill = pal[7],
            alpha = 0.4, 
            inherit.aes = F) +
  geom_rect(data = subset(biomarker_d0, Biomarker == "LDH")[1,], 
            xmin = -Inf, 
            xmax = Inf, 
            ymin = -Inf,
            ymax = log10(271),
            fill = pal[7],
            alpha = 0.4, 
            inherit.aes = F) +
  geom_rect(data = subset(biomarker_d0, Biomarker == "CPK")[1,], 
            xmin = -Inf, 
            xmax = Inf, 
            ymin = -Inf,
            ymax = log10(223),
            fill = pal[7],
            alpha = 0.4, 
            inherit.aes = F) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.15) +
  facet_wrap(~ Biomarker, scales = "free_y", ) +
  scale_fill_manual(name = "",
                    values = c("Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4]),
                    guide = guide_legend(ncol = 2)) +
  scale_y_log10(expand = expansion(c(0, 0.2))) +
  theme_bw(base_family = "Arial") +
  xlab("") +
  ylab("") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        plot.title = element_text(hjust = 0.5),
        strip.background = element_blank()) +
  geom_signif(inherit.aes = F, 
              data = all_comps,
              aes(xmin = comp1, xmax = comp2, annotations = annot, y_position = y),
              tip_length = 0,
              manual=TRUE)


CairoPDF("~/Box/RGrant/SCRIPT/manuscript_1/200705_biomarkers_d0_v2.pdf",
    width = max_width_nature * 4,
    height = 25, 
    family = "Arial")
biomarker_plot_d0
dev.off()


biomarker_plot_d0
```   
  
### Compare CRP against "normal" range   
I think the most conservative approach is to compare against the upper limit of "normal": 10mg/L   
   
```{r}
crp_data = subset(biomarker_d0, Biomarker == "CRP")
wilcox.test(x = crp_data$value, mu = 10)
```

  
### All days   
```{r eval=FALSE}
comps = lapply(levels(biomarkers$Biomarker), function(x){
  sub = subset(biomarkers, Biomarker == x)
  comp = pairwise.wilcox.test(x = sub$value, g = sub$Diagnosis, p.adjust.method = "fdr")
  return(comp) })
names(comps) = levels(biomarkers$Biomarker)
comps                 
annotation_df = data.frame(Biomarker=c(rep("CRP", 2), rep("Ferritin", 2)), 
                            start=c("Non-Pneumonia Control", "COVID-19",
                                    "Non-Pneumonia Control", "COVID-19"), 
                            end=c("COVID-19", "Other Pneumonia",
                                  "COVID-19", "Other Pneumonia"),
                            y = log10(c(90, 180, 15e3, 40e3)),
                            label=c(format(comps$CRP$p.value["COVID-19", "Non-Pneumonia Control"], digits = 2, scientific = T),
                                    format(comps$CRP$p.value["Other Pneumonia", "COVID-19"], digits = 2, scientific = T),
                                    format(comps$Ferritin$p.value["COVID-19", "Non-Pneumonia Control"], digits = 2, scientific = T),
                                    format(comps$Ferritin$p.value["Other Pneumonia", "COVID-19"], digits = 2, scientific = T)))


biomarker_plot = ggplot(biomarkers, aes(x = Diagnosis, y = value, fill = Diagnosis)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(size = 0.5) +
  facet_wrap(~ Biomarker, scales = "free_y") +
  scale_fill_npg(name = "", guide = guide_legend(nrow = 2)) +
  theme_bw(base_family = "Arial") +
  xlab("") +
  ylab("") +
  theme(legend.position = "bottom",
        text = element_text(family = "Arial"),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        plot.title = element_text(hjust = 0.5),
        strip.background = element_blank()) +
  geom_signif(inherit.aes = F, 
              data = annotation_df,
              aes(xmin = start, xmax = end, annotations = label, y_position = y),
              tip_length = 0,
              manual=TRUE)
biomarker_plot
```  

# Figure S1   
## Alluvial plots   
### Organize data   
   
By sample   
```{r}
bulk_md = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/201023_bulk_md_for_counting.csv")
sc_samples = c("1235-BAL-00", "1236-BAL-00", "1239-BAL-00", 
               "1240-BAL-00", "1241-BAL-00", "1246-BAL-00",
               "1252-BAL-00", "1254-BAL-00", "1266-BAL-00",
               "1300-BAL-00", "1313-BAL-00", "1317-BAL-00")

#track each sample across analysis process
alluvial_data_samples =  script_bal_data %>% 
  dplyr::select(study_id, tc_pt_study_id, Diagnosis = pna_type, day_of_intubation) %>% 
  rownames_to_column("sample_index") %>% 
  dplyr::mutate(BAL = TRUE,
         `Flow Cytometry` = tc_pt_study_id %in% flow_data$tc_pt_study_id,
         `Bulk RNA-seq` = tc_pt_study_id %in% bulk_md$tc_pt_study_id,
         `Single-Cell RNA-seq` = tc_pt_study_id %in% sc_samples) %>% 
  pivot_longer(cols = BAL:`Single-Cell RNA-seq`,
               names_to = "Analysis",
               values_to = "performed") %>% 
  dplyr::filter(performed == T & !is.na(Diagnosis)) %>% 
  dplyr::select(-performed) %>% 
  mutate(Analysis = factor(Analysis,
                           levels = c("BAL", "Flow Cytometry",
                                      "Bulk RNA-seq", "Single-Cell RNA-seq")))

#fix issue with the healthy controls and two SC samples not having flow
missing_samples_na = subset(alluvial_data_samples,
                            Diagnosis == "Healthy Control" & Analysis == "BAL")
missing_samples_na$Diagnosis = "NA"
missing_samples_na$Analysis = "Flow Cytometry"
missing_indices = c(unique(alluvial_data_samples$sample_index[alluvial_data_samples$tc_pt_study_id == "1252-BAL-00"]),
                    unique(alluvial_data_samples$sample_index[alluvial_data_samples$tc_pt_study_id == "1240-BAL-00"]),
                    unique(alluvial_data_samples$sample_index[alluvial_data_samples$tc_pt_study_id == "1240-BAL-00"]),
                    unique(alluvial_data_samples$sample_index[alluvial_data_samples$tc_pt_study_id == "1266-BAL-00"]),
                    unique(alluvial_data_samples$sample_index[alluvial_data_samples$tc_pt_study_id == "1239-BAL-00"]),
                    unique(alluvial_data_samples$sample_index[alluvial_data_samples$tc_pt_study_id == "1241-BAL-00"]),
                    unique(alluvial_data_samples$sample_index[alluvial_data_samples$tc_pt_study_id == "1254-BAL-00"]))
                    
missing_samples_na = rbind(missing_samples_na,
                             data.frame(sample_index = missing_indices,
                                        study_id = c(1252, 1240, 1240, 1266, 1239, 1241, 1254),
                                        tc_pt_study_id = c("1252-BAL-00", "1240-BAL-00", "1240-BAL-00", "1266-BAL-00",
                                                           "1239-BAL-00", "1241-BAL-00", "1254-BAL-00"),
                                        Diagnosis = rep("NA", 7),
                                        day_of_intubation = rep(NA, 7),
                                        Analysis = c("Flow Cytometry", "Flow Cytometry", "Bulk RNA-seq", "Bulk RNA-seq", 
                                                     "Flow Cytometry", "Flow Cytometry", "Flow Cytometry")))
alluvial_data_samples = rbind(alluvial_data_samples,
                              missing_samples_na)
alluvial_data_samples$Diagnosis = factor(alluvial_data_samples$Diagnosis,
                                         levels = c("NA", "Healthy Control", "Non-Pneumonia Control", "COVID-19", 
                                    "Other Viral Pneumonia", "Other Pneumonia"))
alluvial_data_samples$hide = factor(alluvial_data_samples$Diagnosis == "NA")
```   
   
By patient   
```{r}
alluvial_data_patients = alluvial_data_samples %>% 
  dplyr::select(-c(tc_pt_study_id)) %>% 
  group_by(study_id, Analysis) %>% 
  slice_head()
```

   
### Plot   
   
By sample   
```{r}
#add N to each plot
annotations = alluvial_data_samples %>% 
  group_by(Analysis) %>% 
  dplyr::filter(Diagnosis != "NA") %>% 
  tally() %>% 
  mutate(x = c(1:4),
         y = n + 25)

alluvial_sample = ggplot(alluvial_data_samples, aes(x = Analysis, stratum = Diagnosis, alluvium = sample_index,
                                  fill = Diagnosis, label = Diagnosis)) +
  scale_fill_manual(name = "",
                    values = c("NA" = pal[6],
                               "Healthy Control" = pal[6],
                               "Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4])) +
  geom_alluvium() +
  geom_stratum(aes(alpha = hide, color = hide)) +
  scale_color_manual(name = "",
                      values = c("FALSE" = "black",
                               "TRUE" = alpha("white", 0))) +
  scale_alpha_manual(name = "",
                      values = c("FALSE" = 1,
                               "TRUE" = 0)) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none",
        axis.line = element_blank(),
        axis.ticks = element_blank(),
        panel.background=element_blank(),
        panel.border=element_blank(),panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),plot.background=element_blank(),
        plot.title = element_text(hjust = 0.5, size = 15)) +
  ggtitle("BAL Samples") +
  geom_text(data = annotations, 
            aes(x = x, y = y, label = n), 
            inherit.aes = F,
            size = 8) +
  ylab("")+
  xlab("")

alluvial_sample
```
   
By patient   
```{r}
#add N to each plot
annotations = alluvial_data_patients %>% 
  group_by(Analysis) %>% 
  dplyr::filter(Diagnosis != "NA") %>% 
  tally() %>% 
  mutate(x = c(1:4),
         y = n + 15)

alluvial_patient = ggplot(alluvial_data_patients, aes(x = Analysis, stratum = Diagnosis, alluvium = sample_index,
                                  fill = Diagnosis, label = Diagnosis)) +
  scale_fill_manual(name = "",
                    values = c("NA" = pal[6],
                               "Healthy Control" = pal[6],
                               "Non-Pneumonia Control" = pal[2],
                               "COVID-19" = pal[1],
                               "Other Viral Pneumonia" = pal[3],
                               "Other Pneumonia" = pal[4])) +
  geom_alluvium() +
  geom_stratum(aes(alpha = hide, color = hide)) +
  scale_color_manual(name = "",
                      values = c("FALSE" = "black",
                               "TRUE" = alpha("white", 0))) +
  scale_alpha_manual(name = "",
                      values = c("FALSE" = 1,
                               "TRUE" = 0)) +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none",
        axis.line = element_blank(),
        axis.ticks = element_blank(),
        panel.background=element_blank(),
        panel.border=element_blank(), 
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(), 
        plot.background=element_blank(),
        plot.title = element_text(hjust = 0.5, size = 15)) +
  ggtitle("SCRIPT Patients") +
  geom_text(data = annotations, 
            aes(x = x, y = y, label = n), 
            inherit.aes = F,
            size = 8) +
  ylab("") +
  xlab("")

alluvial_patient
```   
   
## Join   
```{r}

figure_s1 = ((alluvial_patient | (alluvial_sample + theme(legend.position = "bottom"))) /
  (((((smoker_type_plot)) |
  (crp_plot  + theme(legend.position = "none"))) |
  (ddimer_plot + theme(legend.position = "none"))) /
  ((ferritin_plot + theme(legend.position = "none")) |
  (pct_plot + theme(legend.position = "none")) |
  (cpk_plot + theme(legend.position = "none"))) / 
    ((ldh_plot | guide_area() | plot_spacer()) + plot_layout(guides = "collect")) +
    theme(scale_y_log10(expand = expansion(0, 0.3)))) / 
    (pts_enrolled_plot | daily_BAL_plot | sampling_rate_plot)) +
  plot_layout(heights = c(1.5,3,0.8), guides = "auto") +
    plot_annotation(tag_levels = 'a') & 
  theme(text = element_text(size = 20, family = "Arial"),
        plot.tag = element_text(size = 35, family = "Arial"),
        plot.title = element_text(size = 25, hjust = 0.5),
        axis.text = element_text(size = 20),
        axis.title.y = element_text(size = 22.5),
        axis.title.x = element_text(size = 22.5),
        legend.text = element_text(size = 22.5))

CairoPDF("~/Box/RGrant/SCRIPT/manuscript_1/201023_figure_S1.pdf",
    width = max_width_nature * 5,
    height = max_height_nature * 5, 
    family = "Arial")
figure_s1
dev.off()  
```

# Figure 4 (flow portion)      
Transcriptomics are in Quest portion   
## Export just complete flow heatmap   
```{r}
layout = "AAA\nBB#"
fig4_serial_heatmap_gg_clustered = as.ggplot(serial_clustered)

flow_plot_fig5 = fig4_serial_heatmap_gg_clustered +
  combined_plot +
    plot_layout(design = layout) +
    plot_annotation(tag_levels = 'a') & theme(plot.tag = element_text(size = 20, family = "Arial"))

CairoPDF("~/Box/RGrant/SCRIPT/manuscript_1/201018_fig4_rough_local.pdf",
    width = max_width_nature * 3,
    height = max_height_nature, 
    family = "Arial")
flow_plot_fig5
dev.off()
```

# Final summaries   
## Patients   
### All pts   
```{r}
table(alluvial_data_patients$Diagnosis, alluvial_data_patients$Analysis)
```
   
### Only with early BAL   
```{r}
alluvial_data_patients_early = alluvial_data_patients %>% 
  dplyr::filter(day_of_intubation >= 0 & day_of_intubation <=2)
table(alluvial_data_patients_early$Diagnosis, alluvial_data_patients_early$Analysis)
```
   
## Samples   
### All samples   
```{r}
table(alluvial_data_samples$Diagnosis, alluvial_data_samples$Analysis)
```
   
### Early   
```{r}
alluvial_data_samples_early = alluvial_data_samples %>% 
  dplyr::filter(day_of_intubation >= 0 & day_of_intubation <=2)
table(alluvial_data_samples_early$Diagnosis, alluvial_data_samples_early$Analysis)
```
   
## Sanity checks   
### Identify any missing
```{r}
missing_flow = alluvial_data_samples %>% 
  group_by(tc_pt_study_id) %>% 
  filter((any(Analysis == "Bulk RNA-seq") | any(Analysis == "Single-Cell RNA-seq")) &
            (!any(Analysis == "Flow Cytometry") | !any(Analysis == "BAL"))) %>% 
  ungroup() %>% 
  arrange(tc_pt_study_id)

#should just be the 3 cryopreserved samples that went for scRNA-seq
missing_flow
```

# Figure S3   
```{r}
all_unclustered_nolegend_gg = as.ggplot(all_unclustered_nolegend)
layout = "AAAA\nBBBC"
bulk_plot_S3 = all_unclustered_nolegend_gg +
  (covid_super_early_late + theme(plot.title = element_text(size = 15, hjust = 0.5),
                                            strip.text = element_text(size = 15),
                                            axis.text = element_text(size = 12),
                                            axis.title.x = element_text(size = 20)) + 
     plot_layout(guides = "collect")) +
  guide_area() +
  plot_layout(design = layout) +
  plot_annotation(tag_levels = 'a') & theme(plot.tag = element_text(size = 20, family = "Arial"))

CairoPDF("~/Box/RGrant/SCRIPT/manuscript_1/201021_figS3_top_local.pdf",
    width = max_width_nature * 4,
    height = max_height_nature * 1.5, family = "Arial")
bulk_plot_S3
dev.off()
```
   
# Data export   
## Create anonymized patient IDs for publication   
```{r eval=FALSE}
#1-1000 will be space for controls, etc
# random_ints = sprintf("%04d", sample(1:10000))
# script_id_conv = data.frame(script_id = sprintf("%04d", 1:10000),
#                             anonymized_id = random_ints)
# write.csv(script_id_conv,
#           "~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/201022_script_anonymized_ids_master.csv",
#           row.names = F)
```
   
## Export flow data (anonymized)   
```{r}
anon_ids = read.csv("~/OneDrive - Northwestern University/Misharin_Lab/SCRIPT/201022_script_anonymized_ids_master.csv",
                    colClasses = c("character", "character"))
sc_ids = read.csv("~/Box/RGrant/SCRIPT/manuscript_1/201022_script-to-sc-patient.csv",
                  col.names = c("tc_pt_study_id", "scRNAseq_id")) %>% 
  mutate(study_id = substring(tc_pt_study_id, 1, 4))
bulk_samples = bulk_md %>% 
  mutate(study_id = as.character(study_id)) %>% 
  mutate(study_id = ifelse(nchar(study_id) == 1,
                           yes = paste0("000", study_id),
                           no = study_id)) %>% 
  dplyr::select(tc_pt_study_id, sample, study_id)
flow_data_export = script_bal_data %>% 
  full_join(., sc_ids) %>% 
  full_join(., bulk_samples) %>% 
  left_join(., anon_ids, by = c("study_id" = "script_id")) %>% 
  left_join(., (flow_data %>% dplyr::rename(study_id = ID))) %>% 
  dplyr::select(patient = anonymized_id, days_from_first_intubation = day_of_intubation,
                age, sex = gender, diagnosis = pna_type, scRNAseq_id, bulk_RNAseq_sample = sample,
                superinfection = Superinfection, percent_macs_CD206_high = percent_total_CD206_high,
                percent_macs_CD206_low = percent_total_CD206_low, percent_CD3_total:percent_CD8_total,
                percent_neutrophils_total = percent_neutrophils, percent_macrophages_total = percent_macrophages,
                percent_CD206high_macs_total = percent_macs_CD206_high, percent_CD206low_macs_total = percent_macs_CD206_low,
                percent_monocytes_total = percent_monocytes) %>% 
  #keep only samples used for flow, bulk, or single-cell
  dplyr::filter(!is.na(percent_macs_CD206_high) | !is.na(scRNAseq_id) | !is.na(bulk_RNAseq_sample)) 
write.csv(flow_data_export, "~/Box/RGrant/SCRIPT/manuscript_1/2010245_flow_bulk_and_sc_data.csv")
```

