---
title: "SCRIPT Bulk High-Level Analysis"
output: html_notebook
---

# Setup   
## Load packages   
```{r setup}
library(tidyverse)
library(ggplot2)
library(ggplotify)
library(ggsignif)
library(Cairo)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(viridis)
library(patchwork)
library(factoextra)
library(parallel)
library(uwot)
library(WGCNA)
library(ggsci)
library(topGO)
library(GO.db)
library(org.Hs.eg.db)
library(parallel)
library(doParallel)
library(biomaRt)
library(openxlsx)
library(dendsort)
library(cowplot)

registerDoParallel(makeCluster(12))
register(MulticoreParam(12))
options(gsubfn.engine = "R")
set.seed(12345)

source("~/utils/R/pretty_MA_plot.R")
source("~/utils/R/k_means_figure.R")
source("~/utils/R/plotPCA_manual.R")

fig2_pal = pal_npg("nrc")(9)

sessionInfo()
```

## Import data   
```{r}
mp = readRDS("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_script_bulk_des_no_hURNA.rds")
mp$covid_confirmed = factor(mp$covid_confirmed)
mp$sex = factor(mp$gender)

#set up for comparison by diagnosis
mp$Diagnosis = gsub("-| ", "_", as.character(mp$pna_type))
mp$Diagnosis = factor(mp$Diagnosis, 
                      levels = c("Healthy_Control", "Non_Pneumonia_Control", "COVID_19", 
                                 "Other_Viral_Pneumonia", "Other_Pneumonia"))
mp$neutrophilic = mp$percent_neutrophils > 50
#for modeling, etc
mp$finite_day_of_intubation = mp$day_of_intubation
mp$finite_day_of_intubation[is.infinite(mp$finite_day_of_intubation)] = NA
mp$day0_sample = factor(mp$day_of_intubation >= 0 & mp$day_of_intubation <=2)
mp$day0_sample[is.na(mp$day0_sample)] = FALSE
design(mp) = as.formula("~ Diagnosis")
metadata = as.data.frame(colData(mp))
```   
   
## Summary stats   
### Number of samples overall   
```{r}
table(mp$covid_confirmed, useNA = "always")
table(mp$pna_type, useNA = "always")
```   
   
### Number of patients   
```{r}
bulk_patients = colData(mp) %>% 
  as.data.frame() %>% 
  dplyr::select(study_id, covid_confirmed, pna_type) %>% 
  unique()
table(bulk_patients$covid_confirmed)
table(bulk_patients$pna_type)
```


### Number of serial samples   
```{r}
n_samples = table(metadata$patient_id)
serial_patients = names(n_samples[n_samples > 1])
length(serial_patients)
```   
   
### Distribution of days   
```{r}
ggplot(metadata, aes(x = day_of_intubation)) +
  geom_histogram(fill = "dodgerblue4")
```
   
## PCA   
### Calculate and organize   
```{r}
mp_vst = vst(mp, nsub = 3000, fitType = "local") #know from later that local is best
mp_counts = counts(estimateSizeFactors(mp), normalized = T)
pca_data = plotPCA_manual(mp_vst,
                          intgroup = c("covid_confirmed"),
                          pcs = 10)
pca_data$data = left_join(pca_data$data,
                     as.data.frame(colData(mp)),
                     by = c("name" = "sample", "covid_confirmed"))

#merge counts and pca data
gene_conv = as.data.frame(rowData(mp)) %>% 
  rownames_to_column(var = "ensembl_gene_id")
merge_counts = mp_counts %>% 
  t() %>%
  as.data.frame() %>% 
  rownames_to_column(var = "name")

#merge together for featureplot dataset
fp_data = left_join(pca_data$data, merge_counts, by = "name")
```

### How many PCs?   
```{r}
#scree plot   
fviz_eig(pca_data$pca, 
         barfill = "dodgerblue4", 
         xlab = "Principal Component",
         ylab = "Percent Variance Explained", 
         main = "",
         ncp = 50,
         ggtheme = theme_bw(),
         addlabels = T) #4-5 PCs will do it
```
   
### Examine loadings   
```{r}
loadings = pca_data$pca$rotation %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "ensembl_gene_id") %>% 
  right_join(gene_conv, .)
```

   
### PC1: MoAM Character   
```{r}
ggplot(fp_data, aes(x = PC1, y = PC2, shape = covid_confirmed, color = percent_total_CD206_high)) +
  geom_point()
fabp4_plot = ggplot(fp_data, aes(x = PC1, y = PC2, color = log2(ENSG00000170323))) +
  geom_point() +
  scale_color_viridis(name = "Log2(FABP4 Counts)")
spp1_plot = ggplot(fp_data, aes(x = PC1, y = PC2, color = log2(ENSG00000118785))) +
  geom_point() +
  scale_color_viridis(name = "Log2(SPP1 Counts)")
fabp4_plot + spp1_plot
```   
    
### PC2: Patterns of activation (also by milieu)   
```{r}
ggplot(fp_data, aes(x = PC1, y = PC2, shape = covid_confirmed, color = C_Reactive_Protein)) +
  geom_point(size = 3)
ggplot(fp_data, aes(x = PC1, y = PC2, shape = covid_confirmed, color = LDH)) +
  geom_point(size = 3)
ggplot(fp_data, aes(x = PC1, y = PC2, shape = covid_confirmed, color = FERRITIN)) +
  geom_point(size = 3)
inhba_plot = ggplot(fp_data, aes(x = PC1, y = PC2, color = log2(ENSG00000122641))) +
  geom_point() +
  scale_color_viridis(name = "Log2(INHBA)")
il1b_plot = ggplot(fp_data, aes(x = PC1, y = PC2, color = log2(ENSG00000125538))) +
  geom_point() +
  scale_color_viridis(name = "Log2(IL1b)")
```
After adding healthy controls this is longer true.   

   
### PC3: Male/female?   
```{r}
ggplot(fp_data, aes(x = PC2, y = PC3, color = gender)) +
  geom_point(size = 3)
xist_plot = ggplot(fp_data, aes(x = PC2, y = PC3, color = log2(ENSG00000229807))) +
  geom_point() +
  scale_color_viridis(name = "Log2(XIST)")
```   
Not really   
   
### Condition
```{r}
ggplot(fp_data, aes(x = PC1, y = PC2, color = Diagnosis)) +
  geom_point(size = 2) +
  stat_ellipse()
```   
   
### Day of intubation
```{r}
ggplot(fp_data, aes(x = PC1, y = PC2, color = log2(day_of_intubation))) +
  geom_point(size = 3)
```   
No clear separation.   
   
## Determine best dispersion estimates   
### Parametric   
```{r}
#note: DESeq is refitting far too many genes. Need to turn this off.
mp_des_parametric = DESeq(mp, 
                          fitType = "parametric",
                          parallel = T,
                          minReplicatesForReplace = Inf)
plotDispEsts(mp_des_parametric, cex = 0.1)
```   
   
This fit really isn't bad, but overestimates at lower expression (probably not a huge deal).   
    
### Local
```{r}
mp_des_local = DESeq(mp,
                     fitType = "local",
                     parallel = T,
                     minReplicatesForReplace=Inf)
plotDispEsts(mp_des_local, cex = 0.1)

mp_des = mp_des_local
rm(mp_des_local, mp_des_parametric)
```  
Still not perfect, but a lot better.   
   
```{r}
#used a few times later; safest to put here
#identify infected samples
non_human_genes = subset(gene_conv, !(grepl("^ENSG", ensembl_gene_id))) #human genes all have this prefix
non_human_genes$gene_name[non_human_genes$ensembl_gene_id == "gene-UJ99_s6gp1"] = "NA" #neuraminidase was misread
non_human_genes$virus = factor(ifelse(grepl("UJ99", non_human_genes$ensembl_gene_id),
                               yes = "Influenza A/California/07/2009",
                               no = "SARS-CoV-2"))
cov2_genes = subset(non_human_genes, grepl("SARS|GU280", ensembl_gene_id))
iav_genes = subset(non_human_genes, grepl("UJ99", ensembl_gene_id))
flu_counts = counts(mp_des, normalized = T)[iav_genes$ensembl_gene_id, ]
flu_detected = colnames(flu_counts[, colSums(flu_counts) > 0])
mp_des$h1n1_counts = colSums(flu_counts)
cov2_counts = counts(mp_des, normalized = T)[cov2_genes$ensembl_gene_id, ]
mp_des$cov2_counts = colSums(cov2_counts)
cov2_detected = colnames(cov2_counts[, colSums(cov2_counts) > 0])
mp_des$sars_cov2_detected = factor(mp_des$sample %in% cov2_detected)
mp_des$iav_h1n1_detected = factor(mp_des$sample %in% flu_detected)

## Replicating CoV2 samples
### overall
antisense_counts = cov2_counts["SARS_CoV_2_antisense_genome", ]
sense_counts = cov2_counts[2:nrow(cov2_counts), ]
replicating_cov2 = intersect(names(antisense_counts)[antisense_counts > 0],
                             colnames(sense_counts[, colSums(sense_counts) > 0]))
mp_des$cov2_replication = mp_des$sample %in% replicating_cov2

#output for Taylor
taylor_data_full = colData(mp_des) %>% 
  as.data.frame() %>% 
  dplyr::select(study_id, tc_pt_study_id, sample, sars_cov2_detected, cov2_counts, 
                cov2_replication, iav_h1n1_detected, h1n1_counts, day_of_intubation,
                Diagnosis)
taylor_data = taylor_data_full %>% 
  as.data.frame() %>% 
  dplyr::filter(sars_cov2_detected == TRUE)
write.csv(taylor_data_full, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200906_complete_data_for_TP.csv")
write.csv(taylor_data, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200906_data_for_TP.csv")

### just covid
percent_detected_covid = length(unique(subset(metadata, sample %in% cov2_detected & covid_confirmed == TRUE)$study_id)) / length(unique(subset(metadata, covid_confirmed == TRUE)$study_id)) * 100
percent_detected_covid
percent_replicating_covid = length(unique(subset(metadata, sample %in% replicating_cov2 & covid_confirmed == TRUE)$study_id)) / length(unique(subset(metadata, covid_confirmed == TRUE)$study_id)) * 100
percent_replicating_covid

percent_replicating_covid / percent_detected_covid * 100

#all viruses   
virus_counts_d0 = counts(mp_des[, !is.na(mp_des$day_of_intubation) & mp_des$day_of_intubation >= 0 & mp_des$day_of_intubation <= 2], normalized = T)[non_human_genes$ensembl_gene_id, ]
n_virus_detected = sum(colSums(virus_counts_d0) > 0)
n_day_0 = sum(!is.na(mp_des$day_of_intubation) & mp_des$day_of_intubation >= 0 & mp_des$day_of_intubation <= 2)
n_virus_detected / n_day_0 * 100
```
      
## Sanity checks  
### Potential sample swaps   
```{r}
sex_counts = counts(mp_des, normalized = T)[c("ENSG00000229807", "ENSG00000129824"), ]
putative_females = colnames(sex_counts)[sex_counts["ENSG00000229807", ] > 100] #XIST
putative_males = colnames(sex_counts)[sex_counts["ENSG00000129824", ] > 100] #RPS4Y1
potential_swaps = colData(mp_des) %>% 
  as_tibble() %>% 
  dplyr::filter((sex == "Male" & sample %in% putative_females) |
                  (sex == "Female" & sample %in% putative_males)) %>% 
  dplyr::select(sample, sex, batch, Batch_sample, study_id, tc_pt_study_id, pna_type) %>% 
  mutate(transcriptional_sex = case_when(sample %in% putative_females ~ "Female",
                                         sample %in% putative_males ~ "Male"))
write.csv(potential_swaps, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201022_potential_sample_swaps.csv")
```

### SARS-CoV-2   
Does expression of viral genes correlate with covid diagnosis?   
```{r}
cov2_counts = lapply(cov2_genes$ensembl_gene_id, function(x){
  counts = plotCounts(mp,
                      gene = x,
                      intgroup = "Diagnosis",
                      returnData = T, 
                      pc = T)
  counts$gene = x
  return(counts)})
cov2_counts = bind_rows(cov2_counts)

ggplot(cov2_counts, aes(x = Diagnosis, y = count, fill = gene)) +
  geom_boxplot() +
  scale_y_log10()
```   
Now that we have all of the necessary metadata, this looks fantastic.  
   

      
### Do counts cluster by virus and diagnosis?   
```{r}
viral_counts = counts(mp_des, normalized = T) 
viral_counts = viral_counts[non_human_genes$ensembl_gene_id, ]
viral_counts = log10(viral_counts + 0.5)

virus_detection = vector(mode = "character", length = ncol(mp_des))
for(i in 1:ncol(mp_des))
{
  has_covid = mp_des$covid_confirmed[i] == TRUE
  has_iav_h1n1 = mp_des$INFLUENZA_A_H1_2009[i] == "Positive"
  has_other_coronavirus = mp_des$any_coronavirus_non[i] == TRUE
  #recode NA as FALSE for safety
  if(is.na(has_iav_h1n1)) 
  {
    has_iav_h1n1 = FALSE
  }
  if(is.na(has_other_coronavirus)) 
  {
    has_other_coronavirus = FALSE
  }
  
  if(has_covid == TRUE && has_iav_h1n1 == TRUE && has_other_coronavirus == TRUE)
  {
    virus_detection[i] = "SARS-CoV-2 & Other Coronavirus & Influenza A/California/07/2009"
  } else if(has_covid == TRUE && has_iav_h1n1 == FALSE && has_other_coronavirus == FALSE)
  {
    virus_detection[i] = "SARS-CoV-2"
  } else if(has_covid == FALSE && has_iav_h1n1 == TRUE && has_other_coronavirus == FALSE)
  {
    virus_detection[i] = "Influenza A/California/07/2009"
  } else if(has_covid == FALSE && has_iav_h1n1 == FALSE && has_other_coronavirus == TRUE)
  {
     virus_detection[i] = "Other Coronavirus"
  } else if(has_covid == TRUE && has_iav_h1n1 == TRUE && has_other_coronavirus == FALSE)
  {
    virus_detection[i] = "SARS-CoV-2 & Influenza A/California/07/2009"
  } else if(has_covid == TRUE && has_iav_h1n1 == FALSE && has_other_coronavirus == TRUE)
  {
    virus_detection[i] = "SARS-CoV-2 & Other Coronavirus"
  } else if(has_covid == FALSE && has_iav_h1n1 == TRUE && has_other_coronavirus == TRUE)
  {
    virus_detection[i] = "Other Coronavirus & Influenza A/California/07/2009"
  } else
  {
    virus_detection[i] = NA
  }
}
mp_des$detected_viruses = factor(virus_detection)
  
diagnosis_annotation = as.data.frame(colData(mp_des)) %>% 
  dplyr::select(`PCR-Detected Viruses` = detected_viruses)
gene_annotation = non_human_genes %>% 
  remove_rownames() %>% 
  column_to_rownames("ensembl_gene_id") %>% 
  dplyr::select(`Viral Origin` = virus)
gene_names = non_human_genes$gene_name
gene_names[gene_names == "SARS_CoV_2_antisense_genome"] = "Antisense"

callback = function(hc, ...){dendsort(hc)}
viral_expression_heatmap = pheatmap(viral_counts,
        clustering_method = "ward.D2",
        show_colnames = F,
        show_rownames = T,
        color = inferno(100),
        annotation_col = diagnosis_annotation,
        annotation_row = gene_annotation,
        labels_row = gene_names,
        annotation_colors = list(Diagnosis = c("SARS-CoV-2" = fig2_pal[1], 
                                               "Other Coronavirus" = fig2_pal[2],
                                               "Influenza A/California/07/2009" = fig2_pal[3], 
                                               "Other Coronavirus & Influenza A/California/07/2009" = fig2_pal[4]),
                                  `Viral Origin` = c("SARS-CoV-2" = fig2_pal[1], 
                                                     "Influenza A/California/07/2009" = fig2_pal[3])),
        angle_col = 45, 
        annotation_names_row = F, 
        clustering_callback = callback, 
        fontsize = 15, 
        annotation_legend = T)


early_samples = colData(mp_des) %>% 
  as_tibble() %>% 
  dplyr::filter(day_of_intubation >= 0 & day_of_intubation <= 2) %>% 
  .$sample
viral_counts_early = viral_counts[, early_samples]
viral_expression_heatmap_early = pheatmap(viral_counts_early,
        clustering_method = "ward.D2",
        show_colnames = F,
        show_rownames = T,
        color = inferno(100),
        annotation_col = diagnosis_annotation,
        annotation_row = gene_annotation,
        labels_row = gene_names,
        annotation_colors = list(`PCR-Detected Viruses` = c("SARS-CoV-2" = fig2_pal[1], 
                                               "Other Coronavirus" = fig2_pal[2],
                                               "Influenza A/California/07/2009" = fig2_pal[3], 
                                               "Other Coronavirus & Influenza A/California/07/2009" = fig2_pal[4]),
                                  `Viral Origin` = c("SARS-CoV-2" = fig2_pal[1], 
                                                     "Influenza A/California/07/2009" = fig2_pal[3])),
        angle_col = 45, 
        annotation_names_row = F, 
        clustering_callback = callback, 
        annotation_legend = T,
        legend = T)
```
   
### Do CoV2 reads decrease over time?   
Checking for viral clearance   
```{r}
covid_cases = mp_des[, mp_des$covid_confirmed == TRUE]
covid_counts = counts(covid_cases, normalized = T)
covid_counts = covid_counts[cov2_genes$ensembl_gene_id, ]

covid_counts = covid_counts %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  pivot_longer(cols = "340195046":"340239867",
               names_to = "sample",
               values_to = "Counts") %>% 
  left_join(., non_human_genes, by = c("gene" = "ensembl_gene_id")) %>% 
  dplyr::select(-c(gene, virus)) %>% 
  left_join(., as.data.frame(colData(mp_des)), by = "sample")

ggplot(covid_counts, aes(x = day_of_intubation, y = Counts)) +
  facet_wrap(~ gene_name, scales = "free_y") +
  geom_point() +
  geom_smooth(se = F)
```   
Unfortunately this is somewhat binarized. Better to treat it as such.   
   
```{r}
covid_counts_binarized = covid_counts %>% 
  group_by(sample) %>% 
  mutate(all_viral_counts = sum(Counts)) %>% 
  dplyr::select(sample, all_viral_counts, day_of_intubation) %>% 
  unique() %>% 
  mutate(virus_detected = all_viral_counts > 0)

time_cor = cor.test(covid_counts_binarized$all_viral_counts, covid_counts_binarized$day_of_intubation, method = "spearman")
time_cor_rho = round(time_cor$estimate, digits = 2)

sars_time_cor = ggplot(covid_counts_binarized, aes(x = day_of_intubation, y = (all_viral_counts + 0.5))) +
  geom_point() +
  scale_y_continuous(trans = "log10") +
  xlab("Day of Mechanical Ventilation") +
  ylab("SARS-CoV-2 Counts") +
  geom_smooth(se = T, color = fig2_pal[2], method = "loess") +
  theme_bw(base_family = "Arial") +
  theme(text = element_text(family = "Arial")) +
  annotate(x = 30, y = 5e4, label = paste("Ï = ", time_cor_rho), size = 6, geom = "text")
sars_time_cor
```
      
## Which samples are worth sequencing further?   
### First batch   
```{r eval=FALSE}
good_samples = mp_des[, ((!is.na(mp_des$RIN) & mp_des$RIN >= 7) & 
                    mp_des$STAR_mqc_generalstats_star_uniquely_mapped_percent >= 30 &
                    mp_des$featureCounts_mqc_generalstats_featurecounts_percent_assigned >= 30) &
                    mp_des$RNA_concentration_pg_ul >= 125 |
                    (mp_des$iav_h1n1_detected == TRUE | mp_des$sars_cov2_detected == TRUE)]

selection_data = as.data.frame(colData(good_samples)) %>% 
  dplyr::select(sample, tc_pt_study_id, RIN, STAR_mqc_generalstats_star_uniquely_mapped_percent,
         featureCounts_mqc_generalstats_featurecounts_percent_assigned, iav_h1n1_detected, 
         sars_cov2_detected, covid_confirmed, batch, RNA_concentration_pg_ul, pna_type)
selection_data
```   
Keepers:   
- 340224808 (IAV)
- 304017553 (IAV + likely another coronavirus)
- 340239805 (Cov2)
- 340233896 (Cov2)
- 340239790 (Cov2)
- 340239789 (Cov2)
- 304012699 (Other -- likely another coronavirus)
- 300312389 (other)
- 304020362 (other)
   
```{r eval=FALSE}
keepers = selection_data %>% 
  dplyr::filter(sample %in% c(304020298, 340224808, 304017553, 340239805, 340233896,
                              340239790, 340239789, 304012699, 300312389, 304020362)) %>% 
  mutate(vol_for_250_pg = round(250 / RNA_concentration_pg_ul, digits = 3))
keepers$dilution = ifelse(keepers$vol_for_250_pg > 0.5,
                          yes = 1,
                          no = ifelse(keepers$vol_for_250_pg > 0.1,
                                      yes = 5,
                                      no = 20))
keepers$dilution_vol_for_250_pg = round(250 / (keepers$RNA_concentration_pg_ul / keepers$dilution), digits = 3)
write.csv(keepers, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200612_resequencing_samples.csv")
keepers
```   
   
# Signatures associated with SARS-CoV-2   
## Single genes   
### IL6   
```{r}
il6_counts = plotCounts(mp_des, 
           gene = "ENSG00000136244",
           normalized = T,
           intgroup = "covid_confirmed",
           returnData = T) %>% 
  rownames_to_column(var = "sample") %>% 
  left_join(., 
            metadata,
            by = c("sample", "covid_confirmed"))

cor.test(il6_counts$count, il6_counts$percent_total_CD206_high)
il6_plot = ggplot(il6_counts, aes(x = percent_total_CD206_high, y = count, color = covid_confirmed)) +
  geom_point(size = 3) +
  scale_y_log10() +
  ylab("IL6 Counts")
```   
Later confirmed NSD by group. Clearly correlates with MoAM character.        
   
### IL1b   
```{r}
il1b_counts = plotCounts(mp_des, 
           gene = "ENSG00000125538",
           normalized = T,
           intgroup = "covid_confirmed",
           returnData = T) %>% 
  rownames_to_column(var = "sample") %>% 
  left_join(., 
            metadata,
            by = c("sample", "covid_confirmed"))

cor.test(il1b_counts$count, il1b_counts$percent_total_CD206_high)
il1b_plot = ggplot(il1b_counts, aes(x = percent_total_CD206_high, y = count, color = covid_confirmed)) +
  geom_point(size = 3) +
  scale_y_log10() + 
  ylab("IL1b Counts")

il6_plot / il1b_plot
```
Later confirmed significantly downregulated, actually.   

## Basic DEA   
### Gene hits   
   
Versus healthy controls   
```{r}
cov2_vs_hc_results = as.data.frame(results(object = mp_des,
                                  contrast = c("Diagnosis", "COVID_19", "Healthy_Control"),
                                  alpha = 0.05,
                                  parallel = T))

cov2_vs_hc_results_ids = cov2_vs_hc_results %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  left_join(., gene_conv) %>% 
  dplyr::filter(padj < 0.05 | !grepl("ENSG", ensembl_gene_id)) %>% 
  arrange(log2FoldChange)

write.csv(cov2_vs_hc_results_ids, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200717_standard_DEA_covid_over_healthycontrols.csv")

cov2_vs_hc_results_ids
```   
   
Upregulation of CoV-2 genes is gorgeous
   
Versus non-pna (sick) controls
```{r}
cov2_vs_npc_results = as.data.frame(results(object = mp_des,
                                  contrast = c("Diagnosis", "COVID_19", "Non_Pneumonia_Control"),
                                  alpha = 0.05,
                                  parallel = T))

cov2_vs_npc_results_ids = cov2_vs_npc_results %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  left_join(., gene_conv) %>% 
  dplyr::filter(padj < 0.05 | !grepl("ENSG", ensembl_gene_id)) %>% 
  arrange(log2FoldChange)

write.csv(cov2_vs_npc_results_ids, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200717_standard_DEA_covid_over_nonpneumoniacontrols.csv")

cov2_vs_npc_results_ids
```
   
Versus other viral PNA
```{r}
cov2_vs_ovp_results = as.data.frame(results(object = mp_des,
                                  contrast = c("Diagnosis", "COVID_19", "Other_Viral_Pneumonia"),
                                  alpha = 0.05,
                                  parallel = T))

cov2_vs_ovp_results_ids = cov2_vs_ovp_results %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  left_join(., gene_conv) %>% 
  dplyr::filter(padj < 0.05 | !grepl("ENSG", ensembl_gene_id)) %>% 
  arrange(log2FoldChange)

cov2_vs_ovp_results_ids

write.csv(cov2_vs_ovp_results_ids, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200717_standard_DEA_covid_over_otherviralpneumonia.csv")
```   
   
Versus other PNA
```{r}
cov2_vs_other_pna_results = as.data.frame(results(object = mp_des,
                                  contrast = c("Diagnosis", "COVID_19", "Other_Pneumonia"),
                                  alpha = 0.05,
                                  parallel = T))

cov2_vs_other_pna_results_ids = cov2_vs_other_pna_results %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  left_join(., gene_conv) %>% 
  dplyr::filter(padj < 0.05 | !grepl("ENSG", ensembl_gene_id)) %>% 
  arrange(log2FoldChange)

write.csv(cov2_vs_other_pna_results_ids, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200717_standard_DEA_covid_over_otherpneumonia.csv")

cov2_vs_other_pna_results_ids
```  
   
### Heatmap   
   
How many clusters does it take?   
```{r eval=FALSE}
elbow = k_elbow(dge = mp_des, design = "~Diagnosis", 
        cores = 12, 
        minReps = Inf,
        random_seed = 12345,
        max_k = 25,
        qval_cutoff = 0.01)
elbow
```   
5 seems fair.   
    
```{r eval=FALSE}
intubation_pal = colorRampPalette(brewer.pal(n = 9, name = "Purples")[3:9])(max(mp_des$finite_day_of_intubation, na.rm = T) + 1)
#intubation_pal[1:3] = rep("#000000", 3) #highlight "D0"

cov2_kmeans_noclustering = k_means_figure(dge = mp_des,
                                 design = "~Diagnosis",
                                 k = 5,
                                 go_annotations = "org.Hs.eg.db",
                                 ensembl_db = "hsapiens_gene_ensembl",
                                 legend_factors = c("Diagnosis", "finite_day_of_intubation", "Superinfection"),
                                 breaks = seq(-2, 2, length.out=101),
                                 cores = 1,
                             minReps = Inf,
                             return_genes = T,
                             display_go_terms = F,
                             return_go_terms = F,
                             cluster_columns = F,
                             show_rownames = F,
                             baseMeanCutoff = 20,
                             random_seed = 12345,
                             qval_cutoff = 0.05,
                             annotation_colors = list(Diagnosis = c("Healthy_Control" = fig2_pal[6],
                                                                    "Non_Pneumonia_Control" = fig2_pal[2],
                                                                    "COVID_19" = fig2_pal[1], 
                                                                    "Other_Viral_Pneumonia" = fig2_pal[3],
                                                                    "Other_Pneumonia" = fig2_pal[4]),
                                                      finite_day_of_intubation =  intubation_pal,
                                                      Superinfection = c("Superinfection" = fig2_pal[9],
                                                                         "Primary Only" = fig2_pal[7])), 
                             custom_order = c(3, 1, 4, 2, 5), 
                             sortColumns = T,
                             column_sort_factors = c("Diagnosis", "finite_day_of_intubation"))

cov2_kmeans_clustered = k_means_figure(dge = mp_des,
                                 design = "~Diagnosis",
                                 k = 5,
                                 go_annotations = "org.Hs.eg.db",
                                 ensembl_db = "hsapiens_gene_ensembl",
                                 legend_factors = c("Diagnosis", "finite_day_of_intubation", "Superinfection"),
                                 breaks = seq(-2, 2, length.out=101),
                                 cores = 1,
                             minReps = Inf,
                             return_genes = T,
                             display_go_terms = F,
                             return_go_terms = T,
                             cluster_columns = T,
                             show_rownames = F,
                             baseMeanCutoff = 20,
                             random_seed = 12345,
                             qval_cutoff = 0.05,
                             annotation_colors = list(Diagnosis = c("Healthy_Control" = fig2_pal[6],
                                                                    "Non_Pneumonia_Control" = fig2_pal[2],
                                                                    "COVID_19" = fig2_pal[1], 
                                                                    "Other_Viral_Pneumonia" = fig2_pal[3],
                                                                    "Other_Pneumonia" = fig2_pal[4]),
                                                      finite_day_of_intubation =  intubation_pal,
                                                      Superinfection = c("Superinfection" = fig2_pal[9],
                                                                         "Primary Only" = fig2_pal[7])), 
                             custom_order = c(3, 1, 4, 2, 5))
for(i in 1:length(cov2_kmeans_clustered$GO))
{
  cov2_kmeans_clustered$GO[[i]]$cluster = i
}
cov2_kmeans_clustered$go_df = bind_rows(cov2_kmeans_clustered$GO) %>% 
  arrange(cluster, padj)
cov2_kmeans_clustered$expression = counts(mp_des, normalized = T) %>% 
  as.data.frame() %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  dplyr::filter(ensembl_gene_id %in% cov2_kmeans_clustered$genes$gene) %>% 
  right_join(gene_conv, .)
saveRDS(cov2_kmeans_clustered, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_clustered.rds")
saveRDS(cov2_kmeans_noclustering, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_unclustered.rds")
write.csv(cov2_kmeans_clustered$genes, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_genes.csv")
write.csv(cov2_kmeans_clustered$go_df, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_GO.csv")
```   
   
This is actually very cool (and nicely foreshadows the later WGCNA results). C1: mostly COVID-related(!); highly enriched for type I interferon response (not production so much), as well as MHC-I. C2: mostly nonviral pneumonias. Pretty classic STAT3 inflammatory response. 
   
Just day 0   
```{r eval=FALSE}
intubation_pal = colorRampPalette(brewer.pal(n = 9, name = "Purples")[3:9])(max(mp_des$finite_day_of_intubation, na.rm = T) + 1)
#intubation_pal[1:3] = rep("#000000", 3) #highlight "D0"

mp_des_0 = mp_des[, (!is.na(mp_des$day_of_intubation) & mp_des$day_of_intubation >= 0 & mp_des$day_of_intubation <= 2) |
                    mp_des$Diagnosis == "Healthy_Control"]

kmeans_d0 = k_means_figure(dge = mp_des_0,
                                 design = "~Diagnosis",
                                 k = 5,
                                 go_annotations = "org.Hs.eg.db",
                                 ensembl_db = "hsapiens_gene_ensembl",
                                 legend_factors = c("Diagnosis", "super_type"),
                                 breaks = seq(-2, 2, length.out=101),
                                 cores = 1,
                             minReps = Inf,
                             return_genes = T,
                             display_go_terms = F,
                             return_go_terms = T,
                             cluster_columns = T,
                             show_rownames = F,
                             baseMeanCutoff = 20,
                             random_seed = 12345,
                             qval_cutoff = 0.05,
                             annotation_colors = list(Diagnosis = c("Healthy_Control" = fig2_pal[6],
                                                                    "Non_Pneumonia_Control" = fig2_pal[2],
                                                                    "COVID_19" = fig2_pal[1], 
                                                                    "Other_Viral_Pneumonia" = fig2_pal[3],
                                                                    "Other_Pneumonia" = fig2_pal[4]),
                                                      super_type = c("Co-Infection" = fig2_pal[9],
                                                                         "Primary Only" = fig2_pal[7])), 
                           custom_order = c(4, 5, 2, 1, 3))
for(i in 1:length(kmeans_d0$GO))
{
  kmeans_d0$GO[[i]]$cluster = i
}
kmeans_d0$go_df = bind_rows(kmeans_d0$GO) %>% 
  arrange(cluster, padj)
kmeans_d0$expression = counts(mp_des, normalized = T) %>% 
  as.data.frame() %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  dplyr::filter(ensembl_gene_id %in% kmeans_d0$genes$gene) %>% 
  right_join(gene_conv, .)
saveRDS(kmeans_d0, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_d0.rds")
write.csv(kmeans_d0$genes, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_d0_genes.csv")
write.csv(kmeans_d0$go_df, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_d0_GO.csv")
```   
   
## Individual genes 
### IL-6   
```{r}
il6_counts = plotCounts(mp_des, 
                          gene = "ENSG00000136244",
                          intgroup = "Diagnosis",
                       normalized = T,
                       pc = T,
                       returnData = T)
ggplot(il6_counts, aes(x = Diagnosis, y = count, fill  = Diagnosis)) +
  geom_boxplot(width = 0.5, outlier.shape = NA) +
  geom_jitter(size = 0.15, width = 0.25) +
  scale_y_continuous(trans = "log10") +
  scale_fill_manual(name = "",
                    values = c("Healthy_Control" = fig2_pal[6],
                               "Non_Pneumonia_Control" = fig2_pal[2],
                               "COVID_19" = fig2_pal[1],
                               "Other_Viral_Pneumonia" = fig2_pal[3],
                               "Other_Pneumonia" = fig2_pal[4])) +
  scale_x_discrete(labels = c("Healthy_Control" = "Healthy\nControl",
                               "Non_Pneumonia_Control" = "Non-Pneumonia\nControl",
                               "COVID_19" = "COVID-19",
                               "Other_Viral_Pneumonia" = "Other Viral\nPneumonia",
                               "Other_Pneumonia" = "Other\nPneumonia")) +
  xlab("") +
  ylab("IL-6 Counts") +
  theme_bw() +
  theme(legend.position = "none", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5))
```

## Sasha's picks   
### All to start   
```{r}
sasha_genes = c("CCL20", "CCL24", "CCL3", "CCL8", "CCR2", "CCL2", "IL1A", "IL1B", "AREG", 
                "CXCR4", "TNFSF14", "CXCL3", "CXCL5", "CXCL1", "CXCL8", "ATF4", "CD164", 
                "MFGE8", "IL13RA1", "IL1RL1", "IL18R1", "VSIG4", "VEGFA", "DDIT4", "BAG3", 
                "DNAJB1", "HSP90AA6P", "HSPA1B", "HSPA1A")
sasha_genes = subset(gene_conv, gene_name %in% sasha_genes)

sasha_counts = mclapply(sasha_genes$ensembl_gene_id, function(x){
  counts = plotCounts(mp_des,
                      gene = x,
                      intgroup = "Diagnosis",
                      returnData = T,
                      normalized = T,
                      pc = T)
  counts$gene = x
  return(counts)})
sasha_counts = bind_rows(sasha_counts) %>% 
  left_join(., sasha_genes, by = c("gene" = "ensembl_gene_id"))

sasha_gene_plot = ggplot(sasha_counts, aes(x = Diagnosis, y = count, fill = Diagnosis)) +
  geom_boxplot(width = 0.5, outlier.shape = NA) +
  geom_jitter(size = 0.15, width = 0.25) +
  scale_y_continuous(trans = "log10") +
  scale_fill_manual(name = "",
                    values = c("Healthy_Control" = fig2_pal[6],
                               "Non_Pneumonia_Control" = fig2_pal[2],
                               "COVID_19" = fig2_pal[1],
                               "Other_Viral_Pneumonia" = fig2_pal[3],
                               "Other_Pneumonia" = fig2_pal[4])) +
  scale_x_discrete(labels = c("Healthy_Control" = "Healthy\nControl",
                               "Non_Pneumonia_Control" = "Non-Pneumonia\nControl",
                               "COVID_19" = "COVID-19",
                               "Other_Viral_Pneumonia" = "Other Viral\nPneumonia",
                               "Other_Pneumonia" = "Other\nPneumonia")) +
  xlab("") +
  ylab("Gene Counts") +
  theme_bw() +
  theme(legend.position = "none") +
  facet_wrap(~gene_name)
sasha_gene_plot
```   

   
## Clustering   
### All samples   
```{r}
detection_rate = function(x)
{
  return(sum(x > 0) / length(x)) 
}

all_counts = counts(mp_des, normalized = T)
#get 500 most variable genes with > 10% detection
rv = rowVars(all_counts)
prop_detected = apply(X = all_counts, MARGIN = 1, FUN = detection_rate)
selection_criteria = data.frame(sample = rownames(all_counts), var = rv, prop_detected = prop_detected) %>% 
  filter(prop_detected >= 0.1 & var > 0) %>% 
  arrange(desc(var))
top_genes = as.character(selection_criteria$sample[1:1000])
top_all_counts = all_counts[top_genes, ]

annotation = colData(mp_des) %>% 
  as.data.frame() %>% 
  dplyr::select(covid_confirmed, 
                sars_cov2_detected,
                sex,
                #day_of_intubation,
                infection_type,
                percent_neutrophils,
                percent_total_CD206_high)
gene_annotation = gene_conv %>% 
  dplyr::filter(ensembl_gene_id %in% rownames(top_all_counts)) %>% 
  column_to_rownames(var = "ensembl_gene_id")

pheatmap(top_all_counts,
         cluster_rows = T,
         cluster_cols = T,
         clustering_method = "ward.D2",
         annotation_col = annotation,
         labels_row = gene_annotation$gene_name,
         scale = "row",
         show_rownames = T,
         fontsize_row = 2,
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(100),
         angle_col = 45)
```

### Day 0   
```{r}
day0_samples = which(mp$day_of_intubation <= 2)
day0 = mp_des[, day0_samples]
day0_counts = counts(day0, normalized = T)

#get 1000 most variable genes
rv = rowVars(day0_counts)
prop_detected = apply(X = day0_counts, MARGIN = 1, FUN = detection_rate)
selection_criteria = data.frame(sample = rownames(day0_counts), var = rv, prop_detected = prop_detected) %>% 
  filter(prop_detected >= 0.1 & var > 0) %>% 
  arrange(desc(var))
top_genes = as.character(selection_criteria$sample[1:1000])
top_day0_counts = day0_counts[top_genes, ]

annotation = colData(day0) %>% 
  as.data.frame() %>% 
  dplyr::select(covid_confirmed, 
                sars_cov2_detected,
                gender,
                infection_type,
                percent_neutrophils,
                percent_total_CD206_high)
gene_annotation = gene_conv %>% 
  dplyr::filter(ensembl_gene_id %in% rownames(top_day0_counts)) %>% 
  column_to_rownames(var = "ensembl_gene_id")

pheatmap(top_day0_counts,
         cluster_rows = T,
         cluster_cols = T,
         clustering_method = "ward.D2",
         annotation_col = annotation,
         labels_row = gene_annotation$gene_name,
         scale = "row",
         show_rownames = T,
         show_colnames = F,
         fontsize_row = 2,
         breaks = seq(-3, 3, length.out=101),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(100),
         angle_col = 45)
```
   
## Comparing different pathogens   
### Individual pathogens   
```{r eval=FALSE}
pathogen_set = mp_des[, !is.na(mp$pathogen)]
pathogen_set$pathogen = factor(as.character(pathogen_set$pathogen)) #refactor
pathogen_kmeans = k_means_figure(dge = pathogen_set, 
                                 design = "~pathogen",
                                 k = 6,
                                 go_annotations = "org.Hs.eg.db",
                                 ensembl_db = "hsapiens_gene_ensembl",
                                 legend_factors = c("pathogen", "covid_confirmed"),
                                 breaks = seq(-3, 3, length.out=101),
                                 cores = 12)
```
Probably too granular   
   
### Broad groupings  
```{r eval=FALSE}
infection_type_kmeans = k_means_figure(dge = pathogen_set, 
                                 design = "~infection_type",
                                 k = 4,
                                 go_annotations = "org.Hs.eg.db",
                                 ensembl_db = "hsapiens_gene_ensembl",
                                 legend_factors = c("pathogen", "infection_type"),
                                 breaks = seq(-3, 3, length.out=101),
                                 cores = 12)
```

# Random   
## ACE2 counts   
```{r}
ace2_counts = plotCounts(mp_des, 
                         gene = "ENSG00000130234", 
                         intgroup = "covid_confirmed",
                         returnData = T,
                         normalized = T, 
                         pc = F)
ggplot(ace2_counts, aes(x = count)) +
  geom_histogram(fill = "dodgerblue4", ) +
  ylab("Number of Samples") +
  xlab("ACE2 Count")
```   
   
# Trajectories?   
## All samples   
We will use PCA as is recommended to make computation feasible.      
```{r}
#select genes with > 10% detection, mean counts > 5
all_counts = t(counts(mp_des, normalized = T)) #need size normalization
prop_detected = apply(X = all_counts, MARGIN = 2, FUN = detection_rate)
all_counts = all_counts[, prop_detected > 0.1]
mean_detection = colMeans(all_counts)
all_counts = all_counts[, mean_detection >= 5]

all_pca = prcomp(all_counts)
fviz_eig(all_pca, 
         barfill = "dodgerblue4", 
         xlab = "Principal Component",
         ylab = "Percent Variance Explained", 
         main = "",
         ncp = 50,
         ggtheme = theme_bw(),
         addlabels = T) #20 should capture almost everything

total_umap = umap(X = all_counts, 
                  pca = 20, 
                  pca_center = T, 
                  n_threads = 1, 
                  scale = "Z", 
                  min_dist = 0.2)
rownames(total_umap) = rownames(all_counts)
colnames(total_umap) = c("UMAP_1", "UMAP_2")
total_umap = as.data.frame(total_umap)
total_umap$sample = rownames(total_umap)
total_umap = left_join(total_umap, metadata, by = "sample") %>% 
  arrange(study_id, day_of_intubation) # so we can draw paths

ggplot(total_umap, aes(x = UMAP_1, y = UMAP_2, color = Diagnosis)) +
  geom_point() +
  stat_ellipse()

ggplot(total_umap, aes(x = UMAP_1, y = UMAP_2, shape = study_id, color = day_of_intubation)) +
  geom_path(arrow = grid::arrow()) +
  scale_color_viridis() +
  facet_wrap(~ Diagnosis)
```   
Patients actually separate out really well here. This might be worth including as a supplement. With that said, probably better to just use COVID patients for "trajectories". It looks like there may be some structure to the COVID data.      
   

## Only COVID   
```{r}
#remove non-detected genes
#note that PCA should only use genes detected in all samples (3081)
COVID_counts = t(counts(mp_des[, mp_des$Diagnosis == "COVID_19"], normalized = T)) #need size normalization
prop_detected = apply(X = COVID_counts, MARGIN = 2, FUN = detection_rate)
COVID_counts = COVID_counts[, prop_detected > 0.1]
mean_detection = colMeans(COVID_counts)
COVID_counts = COVID_counts[, mean_detection >= 5]

COVID_pca = prcomp(COVID_counts)
fviz_eig(COVID_pca, 
         barfill = "dodgerblue4", 
         xlab = "Principal Component",
         ylab = "Percent Variance Explained", 
         main = "",
         ncp = 50,
         ggtheme = theme_bw(),
         addlabels = T) #20 should capture almost everything

COVID_umap = umap(X = COVID_counts, 
                  pca = 20, 
                  pca_center = T, 
                  n_threads = 1, 
                  scale = "Z", 
                  min_dist = 0.4)
rownames(COVID_umap) = rownames(COVID_counts)
colnames(COVID_umap) = c("UMAP_1", "UMAP_2")
COVID_umap = as.data.frame(COVID_umap)
COVID_umap$sample = rownames(COVID_umap)
COVID_umap = left_join(COVID_umap, metadata, by = "sample") %>% 
  arrange(study_id, day_of_intubation) # so we can draw paths

ggplot(COVID_umap, aes(x = UMAP_1, y = UMAP_2, color = day_of_intubation)) +
  scale_color_viridis() +
  geom_point()
ggplot(COVID_umap, aes(x = UMAP_1, y = UMAP_2, shape = study_id, color = day_of_intubation)) +
  scale_color_viridis() +
  geom_path(arrow = grid::arrow())
```   
Not so much structure it seems.   

# WGCNA   
What gene modules are associated with covid? Which change substantially over time on ventilator?   
## Thesholding
```{r}
#setup
enableWGCNAThreads(nThreads = 12)
WGCNAnThreads()

# Choose a set of soft-thresholding powers
powers = c(1:20)

# Call the network topology analysis functions
sft = pickSoftThreshold(all_counts, powerVector = powers, verbose = 5, networkType = "signed")

# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2))
cex1 = 0.9

# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], 
     -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",
     ylab="Scale Free Topology Model Fit,signed R^2",
     type="n",
     main = paste("Scale independence"))
text(sft$fitIndices[,1], 
     -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,
     cex=cex1,
     col="red")

# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")

# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], 
     sft$fitIndices[,5],
     xlab="Soft Threshold (power)",
     ylab="Mean Connectivity", 
     type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
```   
7 looks like the intersection after adding healthy controls, removing bad samples   
   
## Co-expression by similarity and adjacency   
```{r}
softPower = 7
adjacency = adjacency(all_counts, power = softPower, type = "signed")
```

## Topological overlap matrix   
```{r}
# Turn adjacency into topological overlap
TOM = TOMsimilarity(adjacency, TOMType = "signed")
dissTOM = 1-TOM
```

## Cluster by TOM distance   
```{r}
geneTree = hclust(as.dist(dissTOM), method = "average")

sizeGrWindow(12,9)
plot(geneTree, 
     xlab="", 
     sub="", 
     main = "Gene clustering on TOM-based dissimilarity",
     labels = FALSE, 
     hang = 0.04)
```   
   
## Create modules   
```{r}
minModuleSize = 30 # may need to adjust

# Make modules based on clusters
dynamicMods = cutreeDynamic(dendro = geneTree, 
                            distM = dissTOM,
                            deepSplit = 2,
                            pamRespectsDendro = FALSE,
                            minClusterSize = minModuleSize)
table(dynamicMods)
```   
   
### Label modules    
```{r}
# Convert numeric labels into colors
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)

# Plot the dendrogram and colors underneath
sizeGrWindow(8,6)
plotDendroAndColors(geneTree, 
                    dynamicColors, 
                    "Dynamic Tree Cut",
                    dendroLabels = FALSE,
                    hang = 0.03,
                    addGuide = TRUE,
                    guideHang = 0.05,
                    main = "Gene dendrogram and module colors")
```   
Looks like we captured the structure pretty well.      
   
### Merge similar modules   
```{r}
# Calculate eigengenes
MEList = moduleEigengenes(all_counts, 
                          colors = dynamicColors)
MEs = MEList$eigengenes

# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs)

# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), 
                method = "average")

# Plot the result
sizeGrWindow(7, 6)
plot(METree, 
     main = "Clustering of module eigengenes",
     xlab = "",
     sub = "")

#Now cut at height of 0.5 to get correlation of 0.5  
#standard is 0.25, but we have some really similar modules that are not relevant
MEDissThres = 0.25

# Plot the cut line into the dendrogram
abline(h=MEDissThres, 
       col = "red")

# Call an automatic merging function
merge = mergeCloseModules(all_counts, 
                          dynamicColors, 
                          cutHeight = MEDissThres, 
                          verbose = 3)

# The merged module colors
mergedColors = merge$colors

# Eigengenes of the new merged modules:
mergedMEs = merge$newMEs

# Rename to moduleColors
moduleColors = mergedColors

# Construct numerical labels corresponding to the colors
colorOrder = c("grey", standardColors(50))
moduleLabels = match(moduleColors, colorOrder)-1
MEs = mergedMEs

#replot
sizeGrWindow(12, 9)
plotDendroAndColors(geneTree, 
                    cbind(dynamicColors, mergedColors),
                    c("Dynamic Tree Cut", "Merged dynamic"),
                    dendroLabels = FALSE, 
                    hang = 0.03,
                    addGuide = TRUE, 
                    guideHang = 0.05)
```   
No change   
   
## Relate back to traits   
### Without vent settings   
```{r}
# Define numbers of genes and samples
nGenes = ncol(all_counts)
nSamples = nrow(all_counts)
metadata = as.data.frame(colData(mp_des))
md_of_interest = metadata %>% 
  mutate(deceased = ifelse(binned_outcome == "Deceased",
                           yes = 1,
                           no = 0)) %>% 
  mutate(has_covid = ifelse(covid_confirmed == "TRUE",
                           yes = 1,
                           no = 0)) %>% 
  mutate(cov2_detected = ifelse(sars_cov2_detected == TRUE,
                                yes = 1,
                                no = 0)) %>% 
  mutate(has_pneumonia = ifelse(pna_type == "Non-Pneumonia Control",
                           yes = 0,
                           no = 1)) %>% 
  mutate(coinfection_detected = ifelse(any_nonviral == "TRUE",
                           yes = 1,
                           no = 0)) %>% 
  mutate(other_viral_pna = ifelse(pna_type == "Other Viral Pneumonia",
                           yes = 1,
                           no = 0)) %>% 
  mutate(nonviral_pna = ifelse(pna_type == "Other Pneumonia",
                           yes = 1,
                           no = 0)) %>% 
  mutate(is_healthy_control = ifelse(pna_type == "Healthy Control",
                           yes = 1,
                           no = 0)) %>% 
  dplyr::select(deceased, has_covid, cov2_detected, has_pneumonia,
                other_viral_pna, nonviral_pna, is_healthy_control,
                percent_neutrophils, percent_total_CD206_high,
                percent_CD4_total, percent_CD8_total, finite_day_of_intubation,
                C_Reactive_Protein, D_DIMER, PROCALCITONIN, mean_aps)
# Recalculate MEs with color labels
MEs0 = moduleEigengenes(all_counts, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = bicor(MEs, 
                     md_of_interest, 
                     use = "pairwise.complete.obs",
                     maxPOutliers = 0.05)
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, 
                                     nSamples) %>% 
  as.numeric() %>% 
  p.adjust(., method = "fdr") %>% 
  matrix(ncol = ncol(moduleTraitCor))
colnames(moduleTraitPvalue) = colnames(moduleTraitCor)
rownames(moduleTraitPvalue) = rownames(moduleTraitCor)
moduleTraitCor_hm = t(moduleTraitCor)
#just make significant or not
pvals_hm = t(moduleTraitPvalue) %>% 
  as.matrix()
pvals_hm[pvals_hm < 0.05] = ""
pvals_hm[pvals_hm != ""] = "NS"

labs = c("Deceased", "COVID-19", "SARS-CoV-2 Detected", "Any Pneumonia", "Other Viral Pneumonia", "Other Pneumonia",
         "Healthy Control", "% Neutrophils", "% Macrophages CD206-high", "% CD4 T", "% CD8 T",
         "Days from First Intubation", "CRP", "D-Dimer", "Procalcitonin", "Mean APS")
module_cor_heatmap = pheatmap(mat = moduleTraitCor_hm,
                              labels_row = labs,
                              labels_col = paste("Module", c(1:ncol(MEs))),
                              display_numbers = pvals_hm,
                              cluster_rows = T,
                              cluster_cols = T,
                              clustering_method = "ward.D2",
                              color = colorRampPalette(rev(brewer.pal(n = 7, 
                                                          name = "RdBu")))(100),
                              angle_col = 45,
                              fontsize = 15)
```
Module 12 (turquoise) looks like the key MoAM cluster, 4 (brown) is key TRAM, and module 15 (tan) is our IFN cluster.

### With vent settings   
```{r}
# Define numbers of genes and samples
nGenes = ncol(all_counts)
nSamples = nrow(all_counts)

md_of_interest = metadata %>% 
  mutate(deceased = ifelse(binned_outcome == "Deceased",
                           yes = 1,
                           no = 0)) %>% 
  mutate(has_covid = ifelse(covid_confirmed == "TRUE",
                           yes = 1,
                           no = 0)) %>% 
  mutate(cov2_detected = ifelse(sars_cov2_detected == TRUE,
                                yes = 1,
                                no = 0)) %>% 
  mutate(has_pneumonia = ifelse(pna_type == "Non-Pneumonia Control",
                           yes = 0,
                           no = 1)) %>% 
  mutate(coinfection_detected = ifelse(any_nonviral == "TRUE",
                           yes = 1,
                           no = 0)) %>% 
  mutate(other_viral_pna = ifelse(pna_type == "Other Viral Pneumonia",
                           yes = 1,
                           no = 0)) %>% 
  mutate(nonviral_pna = ifelse(pna_type == "Other Pneumonia",
                           yes = 1,
                           no = 0)) %>% 
  mutate(is_healthy_control = ifelse(pna_type == "Healthy Control",
                           yes = 1,
                           no = 0)) %>% 
  dplyr::select(deceased, has_covid, cov2_detected, has_pneumonia,
                other_viral_pna, nonviral_pna, is_healthy_control,
                percent_neutrophils, percent_total_CD206_high,
                percent_CD4_total, percent_CD8_total, finite_day_of_intubation,
                C_Reactive_Protein, D_DIMER, PROCALCITONIN, mean_aps,
                Static_Compliance, PF_ratio)

# Recalculate MEs with color labels
MEs0 = moduleEigengenes(all_counts, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = bicor(MEs, 
                     md_of_interest, 
                     use = "pairwise.complete.obs",
                     maxPOutliers = 0.05)

moduleTraitPvalue = corPvalueStudent(moduleTraitCor, 
                                     nSamples) %>% 
  as.numeric() %>% 
  p.adjust(., method = "fdr") %>% 
  matrix(ncol = ncol(moduleTraitCor))
colnames(moduleTraitPvalue) = colnames(moduleTraitCor)
rownames(moduleTraitPvalue) = rownames(moduleTraitCor)

moduleTraitCor_hm = t(moduleTraitCor)
#just make significant or not
pvals_hm = t(moduleTraitPvalue) %>% 
  as.matrix()
pvals_hm[pvals_hm < 0.05] = ""
pvals_hm[pvals_hm != ""] = "NS"


labs = c("Deceased", "COVID-19", "SARS-CoV-2 Detected", "Any Pneumonia", "Other Viral Pneumonia", "Other Pneumonia",
         "Healthy Control", "% Neutrophils", "% Macrophages CD206-high", "% CD4 T", "% CD8 T",
         "Days from First Intubation", "CRP", "D-Dimer", "Procalcitonin", "Mean APS",
         "Static Compliance", "P/F Ratio")
module_cor_heatmap_vent_mini = pheatmap(mat = moduleTraitCor_hm,
                              labels_row = labs,
                              labels_col = paste("Module", c(1:ncol(MEs))),
                              display_numbers = pvals_hm,
                              cluster_rows = T,
                              cluster_cols = T,
                              clustering_method = "ward.D2",
                              color = colorRampPalette(rev(brewer.pal(n = 7, 
                                                          name = "RdBu")))(100),
                              angle_col = 315)
module_cor_heatmap_vent = pheatmap(mat = moduleTraitCor_hm,
                              labels_row = labs,
                              labels_col = paste("Module", c(1:ncol(MEs))),
                              display_numbers = pvals_hm,
                              cluster_rows = T,
                              cluster_cols = T,
                              clustering_method = "ward.D2",
                              color = colorRampPalette(rev(brewer.pal(n = 7, 
                                                          name = "RdBu")))(100),
                              angle_col = 45,
                              fontsize = 15)
saveRDS(module_cor_heatmap_vent, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200803_WGCNA_cor_heatmap.rds")
```   
   
### Sanity check: are the CD206 modules TRAM/MoAM markers?   
```{r}
module_assignments = data.frame(ensembl_gene_id = colnames(all_counts),
                                module = factor(mergedColors)) %>% 
  left_join(., gene_conv)
write.csv(module_assignments,
          "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201022_WGCNA_module_assignments.csv")

cd206_correlated = subset(module_assignments, module == "brown")
cd206_anticorrelated = subset(module_assignments, module == "turquoise")
cd206_correlated
cd206_anticorrelated
```
Pretty ideal, yeah   
   
   
### COVID-associated   
Tan contains all of the CoV-2 genes!   
```{r}
covid_correlated_cd206_uncorrelated = subset(module_assignments, module == "tan")
covid_correlated_cd206_uncorrelated
write.csv(covid_correlated_cd206_uncorrelated,
          "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201022_tan_module_15_genes.csv")
```
Pretty classic interferon response. Look at the GO.   
   
```{r}
#from k_means_figure
complete_counts = counts(mp_des, normalized = T)
universe = rownames(complete_counts[rowSums(complete_counts) > 0, ])
fisherTest = new("classicCount", testStatistic = GOFisherTest, name = "Fisher test")

cluster_genes = covid_correlated_cd206_uncorrelated$ensembl_gene_id
selection = as.numeric(universe %in% cluster_genes)
names(selection) = universe
go_data = new("topGOdata", 
              ontology = "BP", 
              allGenes = selection,
              geneSel = function(x){
                return(x == 1)},
              annot = annFUN.org, 
              mapping = "org.Hs.eg.db", 
              ID = "ensembl")
  
#run Fisher test
test_results = getSigGroups(go_data, fisherTest)
module15_score = as.data.frame(score(test_results))
colnames(module15_score) = "pval"
module15_score = rownames_to_column(module15_score, var = "go_id")
  
#adjust p-values and take significant
module15_score$padj = p.adjust(module15_score$pval, method = "fdr")
module15_score = subset(module15_score, padj < 0.05)
module15_score$description = NA
for(i in 1:nrow(module15_score))
{
  module15_score$description[i] = GOTERM[[module15_score$go_id[i]]]@Term
}

#add descriptions
module15_score$full_go = paste(module15_score$go_id, module15_score$description)
write.csv(module15_score,
          "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201022_tan_module_15_GO.csv")
module15_score
```
Pretty much all Type I interferon related. Most of the paper is here.   
   
## Plot interesting modules   
### Tan module (15) and diagnosis, outcome   
```{r}
total_umap = MEList$averageExpr %>% 
  rownames_to_column("sample") %>% 
  left_join(total_umap, .)
covid_umap = total_umap %>% 
  dplyr::filter(covid_confirmed == TRUE)
saveRDS(covid_umap, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_covid_umap.rds")

ggplot(total_umap, aes(x = UMAP_1, y = UMAP_2, size = AEtan, color = Diagnosis)) +
  geom_point()
module15_umap = ggplot(total_umap, aes(x = UMAP_1, y = UMAP_2, size = AEtan, color = pna_type, shape = binned_outcome)) +
  geom_point() +
   theme_bw() +
  theme(text = element_text(family = "Arial"),
        legend.position = "bottom")  +
  scale_color_manual(name = "Diagnosis",
                    values = c("Healthy_Control" = fig2_pal[6],
                               "Non-Pneumonia Control" = fig2_pal[2],
                               "COVID-19" = fig2_pal[1],
                               "Other Viral Pneumonia" = fig2_pal[3],
                               "Other Pneumonia" = fig2_pal[4]),
                    na.value = "black") +
  scale_shape_discrete(name = "Outcome",
                       na.value = 1) +
  scale_size(name = "Module 15 Expression",
                  range = c(0, 8),
               limits = c(-1, 3)) +
  xlab("UMAP 1") +
  ylab("UMAP 2") +
  guides(shape = guide_legend(ncol = 2, title.position = "top", title.hjust = 0.5, override.aes = list(size = 3)),
         color = guide_legend(ncol = 2, title.position = "top", title.hjust = 0.5, override.aes = list(size = 3)),
         size = guide_legend(ncol = 2, title.position = "top", title.hjust = 0.5))

module15_umap 
```   
   
```{r}
tcell_umap = ggplot(total_umap, aes(x = UMAP_1, y = UMAP_2, size = percent_CD3_total, color = pna_type, shape = binned_outcome)) +
  geom_point() +
  theme_bw() +
  theme(text = element_text(family = "Arial"),
        legend.text = element_text(family = "Arial"),
        legend.title = element_text(family = "Arial")) +
  scale_color_manual(name = "",
                    values = c("Healthy_Control" = fig2_pal[6],
                               "Non-Pneumonia Control" = fig2_pal[2],
                               "COVID-19" = fig2_pal[1],
                               "Other Viral Pneumonia" = fig2_pal[3],
                               "Other Pneumonia" = fig2_pal[4]),
                    na.value = "black") +
  scale_shape_discrete(name = "",
                     na.value = 1) +
scale_size_continuous(name = "Percent Lymphocytes", range = c(1, 10)) +
  xlab("UMAP 1") +
  ylab("UMAP 2")

tcell_umap 
```


   
### Brown/turquoise modules and day of intubation in COVID patients   
TRAM content   
```{r}
ggplot(total_umap, aes(x = day_of_intubation, y = AEbrown)) +
  geom_point() +
  geom_smooth()

ggplot(total_umap, aes(x = day_of_intubation, y = AEturquoise)) +
  geom_point() +
  geom_smooth()
```
Not much of a correlation...maybe a flat line, as I guess would be expected with near-constant recruitment. 
   
## Effect of ORF8 expression   
### Isolate counts   
```{r}
orf8 = cov2_genes$ensembl_gene_id[which(cov2_genes$gene_name == "ORF8")]
orf8_counts = plotCounts(mp_des, 
                         gene = orf8, 
                         intgroup = "pna_type",
                         returnData = T,
                         normalized = T, 
                         pc = T)
orf8_samples = rownames(subset(orf8_counts, count >0))

ggplot(orf8_counts, aes(x = pna_type, y = count)) +
  geom_boxplot() +
  geom_jitter(aes(color = count > 0)) +
  scale_y_log10() +
  ylab("ORF8 Counts") +
  xlab("") 
```   
   
Not a ton of detection but very clean.   
   
### Association with interferon responses   
```{r}
mp_des$orf8_detected = mp_des$sample %in% orf8_samples
total_umap$orf8_detected = total_umap$sample %in% orf8_samples
ggplot(total_umap, aes(x = UMAP_1, y = UMAP_2, size = AEtan, color = Diagnosis, shape = orf8_detected)) +
  geom_point()
```  
Pretty clear spatial correlation...but in the opposite direction?   
   
# Interferon response over time   
## WGCNA Module tan   
### Average expression   
```{r}
m15_cor = cor.test(covid_umap$AEtan, covid_umap$finite_day_of_intubation)
module_15_corplot = ggplot(covid_umap, aes(x = finite_day_of_intubation, y = AEtan)) +
  geom_point() +
  annotate(geom = "text", x = 30, y = 2, label = paste0("R = ", round(m15_cor$estimate, digits = 2)), size = 4) +
  geom_smooth(method = "lm", color = fig2_pal[2]) +
  theme_bw() +
  xlab("Day of Mechanical Ventilation") +
  ylab("Average Expression") +
  ggtitle("WGCNA Module 15") +
  theme(legend.position = "none", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 12),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 12))

module_15_corplot
```
   
## GO Categories   
### Pull from biomart   
```{r}
human_mart = useEnsembl("ensembl", dataset = "hsapiens_gene_ensembl", GRCh = 37)

#now pull down genes based on gene ontology (GO) definitions
typeI_IFN_response = getBM(attributes = c("ensembl_gene_id", "external_gene_name"),
                           filters = "go",
                           values = "GO:0060337",
                           mart = human_mart)

IFNg_response = getBM(attributes = c("ensembl_gene_id", "external_gene_name"),
                           filters = "go",
                           values = "GO:0060333",
                           mart = human_mart)
```   
   
### Assign mean expression   
```{r}
unfiltered_counts = counts(mp_des, normalized = T)
typeI_IFN_response = subset(typeI_IFN_response, ensembl_gene_id %in% rownames(unfiltered_counts))
IFNg_response = subset(IFNg_response, ensembl_gene_id %in% rownames(unfiltered_counts))
type1_counts = unfiltered_counts[typeI_IFN_response$ensembl_gene_id, mp_des$covid_confirmed == TRUE]
type1_counts = as.data.frame(colMeans(type1_counts)) %>% 
  rownames_to_column("sample")
colnames(type1_counts)[2] = "typeI_IFN_response"
type2_counts = unfiltered_counts[IFNg_response$ensembl_gene_id, mp_des$covid_confirmed == TRUE]
type2_counts = as.data.frame(colMeans(type2_counts)) %>% 
  rownames_to_column("sample")
colnames(type2_counts)[2] = "IFNg_response"
ifn_counts = unfiltered_counts[intersect(IFNg_response$ensembl_gene_id, typeI_IFN_response$ensembl_gene_id),
                               mp_des$covid_confirmed == TRUE]
ifn_counts = as.data.frame(colMeans(ifn_counts)) %>% 
  rownames_to_column("sample")
colnames(ifn_counts)[2] = "general_IFN_response"

covid_umap = covid_umap %>% 
  left_join(., type1_counts) %>% 
  left_join(., type2_counts) %>% 
  left_join(., ifn_counts)
```
   
### Correlations   
```{r}
typei_cor = cor.test(covid_umap$typeI_IFN_response, covid_umap$finite_day_of_intubation)
typeii_cor = cor.test(covid_umap$IFNg_response, covid_umap$finite_day_of_intubation)
general_cor = cor.test(covid_umap$general_IFN_response, covid_umap$finite_day_of_intubation)
```
   
### Plot   
```{r}
#GO:0060337
typei_corplot = ggplot(covid_umap, aes(x = finite_day_of_intubation, y = typeI_IFN_response)) +
  geom_point() + 
  geom_smooth(method = "lm", color = fig2_pal[2]) +
  annotate(geom = "text", x = 30, y = 1750, label = paste0("R = ", round(typei_cor$estimate, digits = 2)), size = 4) +
  theme_bw() +
  theme(legend.position = "none", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 12),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 12)) +
  ggtitle("Type I Interferon Response (GO:0060337)") +
  ylab("Average Expression") +
  xlab("")

typei_corplot 

#GO:0060333
typeii_corplot = ggplot(covid_umap, aes(x = finite_day_of_intubation, y = IFNg_response)) +
  geom_point() + 
  geom_smooth(method = "lm", color = fig2_pal[2]) +
  annotate(geom = "text", x = 30, y = 3000, label = paste0("R = ", round(typeii_cor$estimate, digits = 2)), size = 4) +
  theme_bw() +
  theme(legend.position = "none", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 12),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 12))  +
  ggtitle("Interferon Î³ Response (GO:0060333)") +
  ylab("") +
  xlab("")

typeii_corplot

ggplot(covid_umap, aes(x = finite_day_of_intubation, y = general_IFN_response)) +
  geom_point() + 
  geom_smooth(method = "lm", color = fig2_pal[2]) +
  theme_bw() +
  theme(legend.position = "none", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 12),
        axis.text = element_text(angle = 45, hjust = 1, size = 10),
        axis.title.x = element_text(size = 12)) +
  ggtitle("Total Interferon Response") +
  ylab("Average IFN response Expression") +
  xlab("Day of Mechanical Ventilation")
```   


   
# Second batch of pico optimization   
Let's focus on interferon-hi vs interferon-low samples   
### Identify interesting   
```{r eval=FALSE}
last_batch = c(300312389, 304012699, 304017553, 304020298, 304020362,
               340224808, 340233896, 340239789, 340239790, 340239805)
good_samples = mp_des[, ((!is.na(mp_des$RIN) & mp_des$RIN >= 7) & 
                    mp_des$STAR_mqc_generalstats_star_uniquely_mapped_percent >= 30 &
                    mp_des$featureCounts_mqc_generalstats_featurecounts_percent_assigned >= 30) &
                    mp_des$RNA_concentration_pg_ul >= 125]
remaining_interesting = setdiff(good_samples$sample, last_batch)
interesting_metadata = subset(total_umap, sample %in% remaining_interesting)
table(interesting_metadata$pna_type)
ggplot(interesting_metadata, aes(x = pna_type, y = AEtan)) +
  geom_boxplot()
selection = interesting_metadata %>% 
  filter(pna_type != "Other Pneumonia" | sample == 340235110) %>% 
  dplyr::select(sample, tube_loc_macrophRNA_boxID, RNA_concentration_pg_ul) %>% 
  mutate(dilution = ifelse((250 / RNA_concentration_pg_ul) < 0.4,
                           yes = round(1 / (250 / RNA_concentration_pg_ul) / 5) * 5,
                           no = 1)) %>% 
  mutate(vol_for_250_pg = round(250 / RNA_concentration_pg_ul * dilution, digits = 2))
write.csv(selection, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200713_pico_opt_batch2.csv")
selection
```   

# Conclusions   
## Output data for posterity   
```{r eval=FALSE}
final_counts_raw = counts(mp_des, normalized = F)

#clean metadata for safe sharing
sc_conv = read.csv("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201022_script-to-sc-patient.csv",
                   col.names = c("tc_pt_study_id", "scRNAseq_id")) %>% 
  unique()
safe_id_conv = read.csv("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201022_script_anonymized_ids_master.csv",
                        colClasses = c("character", "character"))
final_md_safe = colData(mp_des) %>% 
  as_tibble() %>% 
  left_join(., sc_conv) %>% 
  left_join(., safe_id_conv, by = c("study_id" = "script_id")) %>% 
  dplyr::select(sample, patient_id = anonymized_id, age, sex, diagnosis = pna_type, 
                days_from_first_intubation = day_of_intubation, scRNAseq_id,
                superinfection = Superinfection)

write.csv(final_counts_raw, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/GEO_output/201023_raw_counts_geo.csv")
write.csv(final_md_safe, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/GEO_output/201023_deidentified_metadata_geo.csv")
```
   
## Export supplemental files   
### DEA
```{r eval=FALSE}
all_comparisons = combn(levels(mp_des$Diagnosis), 2, simplify = F)
names(all_comparisons) = vapply(all_comparisons, function(x){
  return(paste0(x[1], " vs ", x[2]))}, "char")
dea_hits = lapply(all_comparisons, function(x){
  hits = as.data.frame(results(object = mp_des,
                                  contrast = c("Diagnosis", x[1], x[2]),
                                  alpha = 0.05,
                                  parallel = T)) %>% 
    rownames_to_column("ensembl_gene_id") %>% 
    left_join(., gene_conv) %>% 
    arrange(padj)
  return(hits)})
names(dea_hits) = names(all_comparisons)
saveRDS(dea_hits, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/GEO_output/201023_bulk_dea_complete.rds")
```
   
### Export as single excel file   
```{r eval=FALSE}
output_genes = cov2_kmeans_clustered$genes %>% 
  dplyr::select(ensembl_gene_id = gene, cluster, external_gene_name) %>% 
  arrange(cluster)
names(dea_hits) = gsub("Healthy_Control", "HC", names(dea_hits))
names(dea_hits) = gsub("Other_Viral_Pneumonia", "OVP", names(dea_hits))
names(dea_hits) = gsub("Other_Pneumonia", "OP", names(dea_hits))
names(dea_hits) = gsub("Non_Pneumonia_Control", "NPC", names(dea_hits))
names(dea_hits) = gsub("COVID_19", "COVID-19", names(dea_hits))
all_sheets = c(list("k-means genes" = output_genes,
                    "k-means GO enrichment" = cov2_kmeans_clustered$go_df),
               dea_hits)
write.xlsx(x = all_sheets,
           file = "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_bulk_supplmental_data.xlsx")
```
   
## Assemble figures   
### Figure 3 (early samples)       
```{r}
max_width_nature = 7.20472 #inches
max_height_nature = 9.72441

dea_hits = readRDS("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/GEO_output/201023_bulk_dea_complete.rds")
d0_hm_gg = readRDS("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_d0.rds")$plot %>% 
  as.ggplot()
viral_expression_heatmap_early_gg = as.ggplot(viral_expression_heatmap_early)

layout = "A\nB"
bulk_plot_fig3 = d0_hm_gg +
  viral_expression_heatmap_early_gg +
  plot_layout(design = layout, widths = c(0.5, 1), guides = "collect") +
    plot_annotation(tag_levels = 'a') & theme(text = element_text(family = "Arial"),
                                              plot.tag = element_text(size = 20, family = "Arial"))

CairoPDF("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_fig3_rough.pdf",
    width = max_width_nature * 2,
    height = max_height_nature * 2, 
    family = "Arial")
bulk_plot_fig3
dev.off()
```
   
### Figure S3  
Marker genes   
```{r}
#convert back
names(dea_hits) = gsub("HC|Healthy_Control", "Healthy Control", names(dea_hits))
names(dea_hits) = gsub("OVP|Other_Viral_Pneumonia", "Other Viral Pneumonia", names(dea_hits))
names(dea_hits) = gsub("OP|Other_Pneumonia", "Other Pneumonia", names(dea_hits))
names(dea_hits) = gsub("NPC|Non_Pneumonia_Control", "Non-Pneumonia Control", names(dea_hits))
names(dea_hits) = gsub("COVID_19", "COVID-19", names(dea_hits))

fig3_genes = c("IL6", "CCL24", "CCL8", "CCL2", "CCL7", 
               "RNASE1", "CCR1", "MAFB", "CSF2RA",
               "HIF1", "CCL20", "IL1A", "IL1B",
               "MFGE8", "FCGR2A", "FABP4", "CXCL10", "CXCL11",
               "MSR1", "AXL", "C1QA", "ALOX5", "MARCO",
               "MFGE8", "FCGR2A", 
               "IL13RA1")
gene_comps = lapply(names(dea_hits), function(x)
{
  sub = subset(dea_hits[[x]], gene_name %in% fig3_genes) %>% 
    dplyr::filter(padj < 0.05)
  if(nrow(sub) == 0)
  {
    return(NULL)
  }
  comp = unlist(strsplit(x, split = " vs "))
  sub$numerator = comp[1]
  sub$denominator = comp[2]
  return(sub)
})
gene_comps = bind_rows(gene_comps) 

fig3_genes = fig3_genes[fig3_genes %in% c("IL6", unique(gene_comps$gene_name))] #keep only genes with sig diffs and IL6
fig3_genes = subset(gene_conv, gene_name %in% fig3_genes)
s3_counts = mclapply(fig3_genes$ensembl_gene_id, function(x){
  counts = plotCounts(mp_des,
                      gene = x,
                      intgroup = "Diagnosis",
                      returnData = T,
                      normalized = T,
                      pc = T)
  counts$gene = x
  return(counts)})
s3_counts = bind_rows(s3_counts) %>% 
  left_join(., fig3_genes, by = c("gene" = "ensembl_gene_id"))

#add y positions automatically
gene_comps = mutate(gene_comps, label = format(padj, digits = 2, scientific = T)) %>% 
  arrange(gene_name) %>% 
  group_by(gene_name, .drop = T) %>% 
  mutate(y = max(dplyr::filter(s3_counts, gene == unique(ensembl_gene_id))$count) * 10 ^ (1 + 0.5 * c(1:n()))) %>% 
  mutate(y_log10 = (log10(y) - 1.3))

#clean up names
s3_counts = s3_counts %>% 
  mutate(Diagnosis = factor(case_when(Diagnosis == "Healthy_Control" ~ "Healthy Control",
                               Diagnosis == "Other_Viral_Pneumonia" ~ "Other Viral Pneumonia",
                               Diagnosis == "Non_Pneumonia_Control" ~ "Non-Pneumonia Control",
                               Diagnosis == "Other_Pneumonia" ~ "Other Pneumonia",
                               Diagnosis == "COVID_19" ~ "COVID-19"),
                            levels = c("Healthy Control", "Non-Pneumonia Control",
                                       "COVID-19", "Other Viral Pneumonia", 
                                       "Other Pneumonia")))

marker_gene_plot = ggplot(s3_counts, aes(x = Diagnosis, y = count, fill = Diagnosis)) +
  geom_boxplot(width = 0.5, outlier.shape = NA) +
  geom_jitter(size = 0.15, width = 0.25) +
  scale_y_continuous(trans = "log10", expand = expansion(0, 0.5)) +
  scale_fill_manual(name = "",
                    values = c("Healthy Control" = fig2_pal[6],
                               "Non-Pneumonia Control" = fig2_pal[2],
                               "COVID-19" = fig2_pal[1],
                               "Other Viral Pneumonia" = fig2_pal[3],
                               "Other Pneumonia" = fig2_pal[4])) +
  scale_x_discrete(labels = c("Healthy Control" = "Healthy\nControl",
                               "Non-Pneumonia Control" = "Non-Pneumonia\nControl",
                               "COVID-19" = "COVID-19",
                               "Other Viral Pneumonia" = "Other Viral\nPneumonia",
                               "Other Pneumonia" = "Other\nPneumonia")) +
  xlab("") +
  ylab("Gene Counts") +
  facet_wrap(~gene_name, scale = "free_y", ncol = 3) +
  geom_signif(inherit.aes = F, 
              data = gene_comps,
              aes(xmin = numerator, xmax = denominator, annotations = label, y_position = y_log10),
              tip_length = 0,
              manual=TRUE,
              family = "Arial",
              textsize = 5) +
theme_bw(base_family = "Arial") +
theme(legend.position = "bottom",
         text = element_text(family = "Arial"),
         plot.title = element_text(hjust = 0.5),
         strip.background = element_blank()) +
  guides(fill=guide_legend(nrow=2, byrow=TRUE))

marker_gene_plot
```
   
### Figure S3   
```{r}
cov2_kmeans_noclustering = readRDS("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_unclustered.rds")
#coverage_plot = readRDS("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/200803_cov2_coverage_plot.rds")
figS3_heatmap = cov2_kmeans_noclustering$plot
figS3_heatmap_gg = as.ggplot(figS3_heatmap)

layout = "AABB\nAABB\nAABB\nAABB\nCDBB\nEEBB"
bulk_plot_S3 = figS3_heatmap_gg +
  (marker_gene_plot + theme(plot.title = element_text(size = 20, hjust = 0.5),
                            strip.text = element_text(size = 20),
                            axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
                            axis.title = element_text(size = 20),
                            legend.text = element_text(size = 20))) +
  (module15_umap + theme(axis.text = element_text(size = 12),
                                          axis.title = element_text(size = 20))) +
  (tcell_umap + theme(axis.text = element_text(size = 12),
                                          axis.title = element_text(size = 20))) +
  (sars_time_cor + theme(plot.title = element_text(size = 15, hjust = 0.5),
                                          strip.text = element_text(size = 15),
                                          axis.text = element_text(size = 12),
                                          axis.title = element_text(size = 20))) +
    plot_layout(design = layout) +
    plot_annotation(tag_levels = 'a') & theme(plot.tag = element_text(size = 30, family = "Arial"))

CairoPDF("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_figS3_bottom_quest.pdf",
    width = max_width_nature * 4,
    height = max_height_nature * 2.5, family = "Arial")
bulk_plot_S3
dev.off()

bulk_plot_S3
```
   
   
```{r}
md = as.data.frame(colData(mp_des))
col_types = unlist(lapply(md, class))
list_cols = names(col_types[col_types == "AsIs"])
safe_md = md %>% 
  dplyr::select(-c(all_of(list_cols))) %>% 
  as.data.frame()
write.csv(safe_md, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201002_bulk_mp_metadata.csv")
```
   
# Individual patient trajectories   
## Setup   
### Construct gene list   
```{r}
trajectory_genes = c(union(typeI_IFN_response$ensembl_gene_id,
                         IFNg_response$ensembl_gene_id),
                         c("ENSG00000108700", #CCL8
                           "ENSG00000106178", #CCL24
                           "ENSG00000115009"))
```

### Identify serial patients and subset   
```{r}
serial_patients = unique(mp_des$patient_id[duplicated(mp_des$patient_id)])
serial_data = mp_des[, mp_des$patient_id %in% serial_patients]
```

## Heatmaps   
### Organize data   
```{r}
serial_counts_list = mclapply(trajectory_genes, function(x){
   counts = plotCounts(dds = serial_data,
                       gene = x,
                       normalized = T, 
                       returnData = T,
                       pc = F, #do not add pseudocount
                       intgroup = c("day_of_intubation", "study_id", 
                                    "pna_type", "binned_outcome")) %>% 
      rownames_to_column("sample") %>% 
      mutate(gene = x) })
serial_counts = bind_rows(serial_counts_list) %>% 
   arrange(pna_type, binned_outcome, study_id, day_of_intubation)

#remove undetected or nonvariable genes
nonvariable_genes = serial_counts %>%
   group_by(gene) %>% 
   dplyr::summarize(sd = sd(count, na.rm = T)) %>% 
   dplyr::filter(sd == 0) %>% 
   .$gene
length(nonvariable_genes) #9
serial_counts = serial_counts %>% 
   dplyr::filter(!(gene %in% nonvariable_genes))

#make matrix and annotation df
serial_counts_mat = serial_counts %>% 
   dplyr::select(count, sample, gene) %>% 
   pivot_wider(names_from = sample, values_from = count) %>% 
   column_to_rownames("gene")
genes_anno = tibble(ensembl_gene_id = rownames(serial_counts_mat)) %>% 
   left_join(., gene_conv) %>% 
   column_to_rownames("ensembl_gene_id")

safe_md = read.csv("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/GEO_output/201023_deidentified_metadata_geo.csv") %>% 
   dplyr::select(sample, anonymized_patient_id)
serial_counts_anno = serial_counts %>% 
   left_join(., safe_md) %>% 
   dplyr::select(sample, anonymized_patient_id, day_of_intubation, 
                 Diagnosis = pna_type, Outcome = binned_outcome) %>% 
   mutate(anonymized_patient_id = factor(anonymized_patient_id)) %>% 
   unique() %>% 
   remove_rownames() %>% 
   column_to_rownames("sample")
```   
   
### Plot   
```{r}
intubation_pal = colorRampPalette(brewer.pal(n = 9, name = "Purples")[3:9])(max(mp_des$finite_day_of_intubation, na.rm = T) + 1)
progression_hm = pheatmap(mat = serial_counts_mat, 
         cluster_rows = T,
         cluster_cols = F,
         clustering_method = "ward.D2",
         annotation_col = serial_counts_anno,
         labels_row = genes_anno$gene_name,
         scale = "row",
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         annotation_colors = list("day_of_intubation" = intubation_pal,
                                  "Diagnosis" = c("Non-Pneumonia Control" = fig2_pal[2],
                                                "COVID-19" = fig2_pal[1],
                                                "Other Viral Pneumonia" = fig2_pal[3],
                                                "Other Pneumonia" = fig2_pal[4]),
                                  "Outcome" = c("Deceased" = fig2_pal[8],
                                                "Home" = fig2_pal[7],
                                                "Inpatient Facility" = fig2_pal[6],
                                                "LTAC" = fig2_pal[5])),
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101))
```
   
### Select just "core" interferon genes in the module and repeat   
```{r}
covid_ifn_genes = module_assignments %>% 
  dplyr::filter(module == "tan" & 
                  grepl("IRF\\d+|IFI\\d+|IFIT\\d+|IFITM\\d+|ISG\\d+|OAS\\d+|MX\\d+", gene_name))

#as before
ifn_hm_md = colData(mp_des) %>% 
  as_tibble() %>% 
  dplyr::filter(pna_type == "COVID-19" & 
                  !is.na(day_of_intubation) &
                  study_id %in% serial_patients) %>% 
  left_join(., safe_md) %>% 
  dplyr::select(day_of_intubation, Patient = anonymized_patient_id, Outcome = binned_outcome, sample) %>% 
  mutate(Outcome = ifelse(Outcome == "Other",
                          yes = NA,
                          no = Outcome),
         Patient = factor(Patient),
         Outcome = factor(Outcome, levels = c("Home", "Inpatient Facility", "LTAC", "Deceased"))) %>% 
  dplyr::filter(!is.na(Outcome)) %>% 
  arrange(Outcome, Patient, day_of_intubation) %>% 
  column_to_rownames("sample")

ifn_hm_mat = counts(mp_des, normalized = T)[covid_ifn_genes$ensembl_gene_id, rownames(ifn_hm_md)]

intubation_pal = colorRampPalette(brewer.pal(n = 9, name = "Purples"))(max(mp_des$finite_day_of_intubation, na.rm = T) + 1)
patient_pal = sample(colorRampPalette(fig2_pal)(length(levels(ifn_hm_md$Patient))))
names(patient_pal) = levels(ifn_hm_md$Patient)

progression_hm = pheatmap(mat = ifn_hm_mat, 
         cluster_rows = T,
         cluster_cols = F,
         clustering_method = "ward.D2",
         annotation_col = ifn_hm_md,
         labels_row = covid_ifn_genes$gene_name,
         scale = "row",
         show_colnames = F,
         breaks = seq(-3, 3, length.out=101),
         annotation_colors = list("day_of_intubation" = intubation_pal,
                                  "Outcome" = c("Home" = fig2_pal[7],
                                                "Inpatient Facility" = fig2_pal[6],
                                                "LTAC" = fig2_pal[5],
                                                "Deceased" = fig2_pal[8]),
                                  "Patient" = patient_pal), 
         border_color = NA,
         color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(101))

saveRDS(progression_hm, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_patient_progression_heatmap.rds")
```

   
## Line plots   
### Organize data   
```{r}
line_data = serial_counts %>% 
  dplyr::filter(pna_type == "COVID-19") %>% 
   dplyr::filter(gene %in% union(typeI_IFN_response$ensembl_gene_id, IFNg_response$ensembl_gene_id)) %>% 
   left_join(., safe_md) %>% 
   group_by(sample) %>% 
   dplyr::summarize(mean_ifn_response = mean(count, na.rm = T),
                    day_of_intubation = day_of_intubation,
                    anonymized_patient_id = factor(anonymized_patient_id),
                    study_id = study_id,
                    Outcome = binned_outcome) %>% 
   unique()
```
   
### Plot   
```{r}
line_plot = ggplot(line_data, aes(x = day_of_intubation, y = mean_ifn_response, color = anonymized_patient_id)) +
   facet_grid(Outcome ~ .) +
   geom_line()

line_plot

line_plot_averaged = ggplot(line_data, aes(x = day_of_intubation, y = mean_ifn_response, shape = anonymized_patient_id)) +
  geom_line(alpha = 0.4, linetype = 2, aes(color = Outcome)) +
   geom_smooth(method = "lm", 
               inherit.aes = F, 
               aes(x = day_of_intubation, y = mean_ifn_response, color = Outcome),
               se = F) +
  xlab("Days from First Intubation") +
  ylab("Mean Interferon-Responsive Gene Expression") +
  scale_color_npg() +
  theme_bw(base_family = "Arial")

line_plot_averaged
```

### All COVID samples   
```{r}
long_ifn_counts = ifn_hm_mat %>% 
  as_tibble(rownames = NA) %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  pivot_longer(2:ncol(.),
               names_to = "sample",
               values_to = "expression", 
               names_repair = "minimal") %>% 
  group_by(sample) %>% 
  dplyr::summarize(mean_module_expression = mean(expression, na.rm = T)) %>% 
  left_join(., rownames_to_column(ifn_hm_md, "sample"))

complete_line_plot = ggplot(long_ifn_counts, aes(x = day_of_intubation, y = mean_module_expression, color = Outcome)) +
  geom_smooth(se = F, method = "lm") +
  geom_point() +
  scale_color_manual(values = c("Deceased" = fig2_pal[8],
                                                "Home" = fig2_pal[7],
                                                "Inpatient Facility" = fig2_pal[6],
                                                "LTAC" = fig2_pal[5],
                                                "Other" = fig2_pal[4])) +
  theme_bw(base_family = "Arial") +
  xlab("Day of Mechanical Ventilation") +
  ylab("") +
  ggtitle("") +
  theme(legend.position = "none", 
        text = element_text(family = "Arial"),
        plot.title = element_text(hjust = 0.5, size = 12),
        axis.text = element_text(size = 10),
        axis.title = element_text(size = 12))
```
   
## Join into final figure   
```{r}
layout = "AAAAAAAAAA\nBBBBBBBBB#"
progression_hm_ggplot = as.ggplot(progression_hm)
covid_progressions = plot_grid(progression_hm_ggplot,
                               complete_line_plot, 
                               ncol = 1, 
                               labels = "auto",
                               label_fontfamily = "Arial",
                               label_fontface = "plain")

covid_progressions
```

# Infected vs uninfected macs in COVID-19   
## Only for COVID   
```{r}
mp_des$replicating = factor(ifelse(mp_des$sample %in% replicating_cov2,
                                   yes = "replication_detected",
                                   no = "no replication"))
infection_data = data.frame(sample = mp_des$sample,
                            cov2_counts = mp_des$cov2_counts,
                            replicating = mp_des$replicating,
                            covid_confirmed = mp_des$covid_confirmed) %>% 
  dplyr::filter(covid_confirmed == T)

ggplot(infection_data, aes(x = cov2_counts, fill = replicating)) +
  geom_density(alpha = 0.3) +
  scale_x_log10() +
  geom_vline(xintercept = 100) #very nice separation at 100 counts with replication

cov2_genes_detected = vector(mode = "numeric", length = ncol(cov2_counts))
for(sample in colnames(cov2_counts))
{
  cur_counts = cov2_counts[, sample]
  n_genes_detected = sum(cur_counts > 0)
  cov2_genes_detected[sample] = n_genes_detected
}

mp_des$infection_detected = factor(case_when(mp_des$sars_cov2_detected == TRUE &
                                               mp_des$covid_confirmed == TRUE ~ "sars_cov2_detected",
                                             mp_des$sars_cov2_detected == FALSE &
                                               mp_des$covid_confirmed == TRUE ~ "sars_cov2_no_detection",
                                             mp_des$iav_h1n1_detected == TRUE & 
                                               mp_des$any_iav == TRUE ~ "h1n1_detected",
                                             mp_des$iav_h1n1_detected == FALSE & 
                                               mp_des$any_iav == TRUE ~ "h1n1_no_detection",
                                             TRUE ~ "other"))
#and just for cov2
mp_des$cov2_infection = factor(case_when(mp_des$sars_cov2_detected == TRUE &
                                               mp_des$covid_confirmed == TRUE ~ "sars_cov2_detected",
                                             mp_des$sars_cov2_detected == FALSE &
                                               mp_des$covid_confirmed == TRUE ~ "sars_cov2_no_detection",
                                             TRUE ~ "other"))
                                             

infection_des = mp_des
design(infection_des) = as.formula("~cov2_infection")

infection_des = DESeq(infection_des, fitType = "local", minReplicatesForReplace = Inf, parallel = T)
infection_results = results(infection_des, c("cov2_infection", "sars_cov2_detected", "sars_cov2_no_detection")) %>% 
  as.data.frame()

hits = rownames(infection_results)[!is.na(infection_results$padj) & infection_results$padj < 0.05]

goi = c(subset(module_assignments, module == "tan" & ensembl_gene_id %in% hits)$gene_name, cov2_genes$gene_name) %>% 
  unique()
gene_conv_formatted = gene_conv %>% 
  dplyr::rename(external_gene_name = gene_name)
infected_ma = pretty_MA_plot(infection_results, custom_annotation = gene_conv_formatted, genes = goi)
infected_ma = infected_ma +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none")

infected_ma
```   
   
## Compared with IAV infected   
```{r}
design(infection_des) = as.formula("~infection_detected")

infection_des = DESeq(infection_des, fitType = "local", minReplicatesForReplace = Inf, parallel = T)
infection_type_results = results(infection_des, c("infection_detected", "sars_cov2_detected", "h1n1_detected")) %>% 
  as.data.frame() 

hits = rownames(infection_type_results)[!is.na(infection_type_results$padj) & infection_type_results$padj < 0.05]

goi = c(subset(module_assignments, module == "tan" & ensembl_gene_id %in% hits)$gene_name, non_human_genes$gene_name) %>% 
  unique()
infection_type_ma = pretty_MA_plot(infection_type_results, custom_annotation = gene_conv_formatted, genes = goi)
infection_type_ma = infection_type_ma +
  theme_bw(base_family = "Arial") +
  theme(legend.position = "none")

infection_type_ma
```

# Sharing   
## Clarissa
```{r}
iav_counts = counts(mp_des, normalized = T)[iav_genes$ensembl_gene_id, ]
mp_des$iav_counts = colSums(iav_counts)
mp_des$antisense_detected = antisense_counts > 0

md = colData(mp_des) %>% 
  as.data.frame()

cov2_hi = md %>% 
  dplyr::filter(replicating == "replication_detected" & covid_confirmed == T) %>% 
  arrange(desc(cov2_counts)) %>% 
  dplyr::select(sample, cov2_counts) %>% 
  dplyr::slice(1:3) %>% 
  .$sample

iav_detection = md %>% 
  dplyr::filter(iav_h1n1_detected == T & any_iav == T) %>% 
  arrange(desc(iav_counts)) %>% 
  dplyr::select(sample, iav_counts) %>% 
  dplyr::slice(1:2) %>% #only 2 strong ones
  .$sample

only_antisense = md %>% 
  dplyr::filter(antisense_detected == T & covid_confirmed == T) %>% 
  arrange(cov2_counts) %>% 
  dplyr::select(sample, cov2_counts) %>% 
  dplyr::slice(1:3) %>% 
  .$sample

healthy_controls = md %>% 
  dplyr::filter(pna_type == "Healthy Control" & cov2_counts == 0) %>% 
  dplyr::slice(1:3) %>% 
  .$sample

to_share = c(cov2_hi, iav_detection, only_antisense, healthy_controls)
to_share_regex = paste(to_share, collapse = "|")

#find sample locations
seq_dirs = list.dirs("/projects/b1038/Pulmonary/rgrant/script_bulk",
                     full.names = T, 
                     recursive = F)
seq_dirs = seq_dirs[grepl("^/projects/b1038/Pulmonary/rgrant/script_bulk/\\d{6}_|SCRIPT_BATCH_07", seq_dirs)]
fastqs = lapply(seq_dirs, function(x){
  fastq_syms = list.files(paste0(x, "/fastq"),
                          pattern = to_share_regex,
                          full.names = T)
  #get real locations
  real_paths = Sys.readlink(fastq_syms)
  return(real_paths) })
fastqs = unlist(fastqs)

#copy to a new folder
# dir.create("/projects/b1038/Pulmonary/rgrant/script_bulk/data_for_clarissa")
# file.copy(from = fastqs,
#           to = "/projects/b1038/Pulmonary/rgrant/script_bulk/data_for_clarissa",
#           copy.date = T, 
#           overwrite = F)
# copy over metadata
# clarissa_md = md %>%
#   dplyr::filter(sample %in% to_share) %>%
#   dplyr::select(sample, covid_confirmed,
#                 replicating, cov2_counts,
#                 antisense_detected, iav_h1n1_detected,
#                 any_iav, iav_counts,
#                 pna_type)
# write.csv(clarissa_md, "/projects/b1038/Pulmonary/rgrant/script_bulk/data_for_clarissa/metadata_for_clarissa.csv")
# our_counts = counts(mp_des, normalized = T)[, rownames(clarissa_md)]
# write.csv(our_counts, "/projects/b1038/Pulmonary/rgrant/script_bulk/data_for_clarissa/counts_for_clarissa.csv")
```   
   
## Internal   
```{r}
saveRDS(mp_des, "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_complete_bulk.rds")
colData(mp_des) %>% 
  as_tibble() %>% 
  dplyr::select(sample, study_id, tc_pt_study_id) %>% 
  write.csv(., "/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_bulk_md_for_counting.csv")
```

   
# Figure 4 (all time-series data)   
## Arrange in patchwork   
```{r}
cov2_kmeans_clustered_gg = readRDS("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_k_means_clustered.rds")$plot %>% 
  as.ggplot()
module_cor_heatmap_vent_gg = as.ggplot(module_cor_heatmap_vent)

layout = "AAABBB\nAAABBB\nAAABBB\nAAABBB\nAAABBB\nAAABBB\nAAABBB\nAAABBB\nAAABBB\nCCFFFF\nDDFFFF\nEEFFFF"
fig4 = cov2_kmeans_clustered_gg +
  module_cor_heatmap_vent_gg +
  (typei_corplot + ylab("")) +
  (typeii_corplot + ylab("Average Expression")) +
  (module_15_corplot + ylab("") + xlab("Days from First Intubation")) +
  covid_progressions +
    plot_layout(design = layout) +
    plot_annotation(tag_levels = 'a') & theme(plot.tag = element_text(size = 20, family = "Arial"))

CairoPDF("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_fig4_rough_quest.pdf",
    width = max_width_nature * 3,
    height = max_height_nature * 1.5, family = "Arial")
fig4
dev.off()

layout = "ADD\nBDD\nCDD"
fig4_bottom = (typei_corplot + ylab("")) +
  (typeii_corplot + ylab("Average Expression")) +
  (module_15_corplot + ylab("") + xlab("Days from First Intubation")) +
  covid_progressions +
    plot_layout(design = layout) +
    plot_annotation(tag_levels = 'a') & theme(plot.tag = element_text(size = 20, family = "Arial"))
CairoPDF("/projects/b1038/Pulmonary/rgrant/script_bulk/Analysis/201023_fig4_bottom_quest.pdf",
    width = max_width_nature * 3,
    height = max_height_nature * 0.75, family = "Arial")
fig4_bottom
dev.off()
```

